// tdesign-vue-next/cjs/upload/components/image-card.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var loading_index = require('../../loading/index.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-449273df.js');
var index$1 = require('../../_chunks/dep-c0cb4d13.js');
require('../../config-provider/hooks/useConfig.js');
var index = require('../../_chunks/dep-88ae5d8a.js');
require('@babel/runtime/helpers/slicedToArray');
var utils = require('../../_chunks/dep-2c1cb23b.js');
require('../../_chunks/dep-497fc9a5.js');
var imageViewer_index = require('../../image-viewer/index.js');
var upload_constants_index = require('../constants/index.js');
var link_index = require('../../link/index.js');
var image_index = require('../../image/index.js');
require('../../_chunks/dep-203a5570.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-3231adce.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-95cb1381.js');
require('../../_chunks/dep-e3eb5659.js');
require('../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6130f266.js');
require('../../_chunks/dep-3ffe18b0.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-e41b3434.js');
require('../../_chunks/dep-f9b59444.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-80827572.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-6e64ef0e.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-51846987.js');
require('../../_chunks/dep-be760bc2.js');
require('../../_chunks/dep-9ada3ec1.js');
require('../../loading/props.js');
require('../../_chunks/dep-9dfd682d.js');
require('../../_chunks/dep-2db31625.js');
require('../../_chunks/dep-0a17438d.js');
require('../../_chunks/dep-5090b515.js');
require('../../_chunks/dep-39313d05.js');
require('../../_chunks/dep-f9bc7c82.js');
require('../../_chunks/dep-1c6213b5.js');
require('../../_chunks/dep-cfdee5e7.js');
require('../../_chunks/dep-e46285c0.js');
require('../../_chunks/dep-54801354.js');
require('../../_chunks/dep-bd59ca66.js');
require('../../_chunks/dep-f7508ffa.js');
require('../../_chunks/dep-470f211b.js');
require('../../_chunks/dep-9b549c85.js');
require('../../_chunks/dep-e25ae470.js');
require('../../_chunks/dep-bed84553.js');
require('../../_chunks/dep-a4b963ab.js');
require('../../_chunks/dep-abc005aa.js');
require('../../_chunks/dep-b4528c21.js');
require('../../image-viewer/image-viewer.js');
require('../../_chunks/dep-4b52068a.js');
require('../../_chunks/dep-d0227fbe.js');
require('../../_chunks/dep-0f015f65.js');
require('../../_chunks/dep-74a0ebc2.js');
require('../../_chunks/dep-94323f13.js');
require('../../_chunks/dep-75d3928f.js');
require('../../_chunks/dep-801a06fe.js');
require('../../_chunks/dep-f3377589.js');
require('../../image-viewer/base/ImageItem.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../_chunks/dep-e04a8c18.js');
require('../../image-viewer/base/ImageModalIcon.js');
require('../../image-viewer/base/ImageViewerModal.js');
require('../../dialog/index.js');
require('../../dialog/dialog.js');
require('../../dialog/props.js');
require('../../_chunks/dep-e0e6a4e1.js');
require('../../button/index.js');
require('../../button/button.js');
require('../../button/props.js');
require('../../_chunks/dep-a2e46896.js');
require('../../_chunks/dep-6512c9d2.js');
require('../../_chunks/dep-3cd26986.js');
require('../../dialog/hooks/useSameTarget.js');
require('../../_chunks/dep-6530011a.js');
require('../../dialog/utils/index.js');
require('../../dialog/dialog-card.js');
require('../../dialog/dialog-card-props.js');
require('../../dialog/hooks/useAction.js');
require('../../_chunks/dep-3a800699.js');
require('../../_chunks/dep-648bfb84.js');
require('../../_chunks/dep-4e2907e0.js');
require('../../dialog/plugin.js');
require('../../image-viewer/props.js');
require('../../image-viewer/base/ImageViewerUtils.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../popup/props.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../popup/container.js');
require('../../_chunks/dep-ba2511b0.js');
require('../../_chunks/dep-4e2a4791.js');
require('../../_chunks/dep-71d9daf0.js');
require('../../tooltip/utils/index.js');
require('../../image-viewer/constants/index.js');
require('../../image-viewer/utils/index.js');
require('../../_chunks/dep-d20c4e5e.js');
require('../../image/image.js');
require('../../_chunks/dep-12e6ec24.js');
require('../../image/props.js');
require('../../space/index.js');
require('../../space/space.js');
require('../../space/props.js');
require('../../_chunks/dep-cec24bbf.js');
require('../../link/link.js');
require('../../link/props.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var ImageCard = Vue.defineComponent({
  name: "UploadImageCard",
  props: _objectSpread(_objectSpread({}, upload_constants_index.commonProps), {}, {
    multiple: Boolean,
    max: Number,
    disabled: Boolean,
    showUploadProgress: Boolean,
    triggerUpload: Function,
    uploadFiles: Function,
    cancelUpload: Function,
    onPreview: Function,
    showImageFileName: Boolean
  }),
  setup: function setup(props) {
    var _toRefs = Vue.toRefs(props),
      displayFiles = _toRefs.displayFiles,
      classPrefix = _toRefs.classPrefix,
      multiple = _toRefs.multiple,
      max = _toRefs.max;
    var locale = Vue.computed(function () {
      return props.locale;
    });
    var _useGlobalIcon = index.useGlobalIcon({
        AddIcon: tdesignIconsVueNext.AddIcon,
        BrowseIcon: tdesignIconsVueNext.BrowseIcon,
        DeleteIcon: tdesignIconsVueNext.DeleteIcon,
        ErrorCircleFilledIcon: tdesignIconsVueNext.ErrorCircleFilledIcon
      }),
      BrowseIcon = _useGlobalIcon.BrowseIcon,
      DeleteIcon = _useGlobalIcon.DeleteIcon,
      AddIcon = _useGlobalIcon.AddIcon,
      ErrorCircleFilledIcon = _useGlobalIcon.ErrorCircleFilledIcon;
    var renderTNodeJSX = index$1.useTNodeJSX();
    var showTrigger = Vue.computed(function () {
      var _displayFiles$value;
      if (multiple.value) {
        return !max.value || displayFiles.value.length < max.value;
      }
      return !((_displayFiles$value = displayFiles.value) !== null && _displayFiles$value !== void 0 && _displayFiles$value[0]);
    });
    var renderMainContent = function renderMainContent(file, index) {
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-upload__card-content ").concat(classPrefix.value, "-upload__card-box")
      }, [Vue.createVNode(image_index.Image, {
        "class": "".concat(classPrefix.value, "-upload__card-image"),
        "src": file.url || file.raw,
        "error": "",
        "fit": "contain"
      }, null), Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-upload__card-mask")
      }, [Vue.createVNode("span", {
        "class": "".concat(classPrefix.value, "-upload__card-mask-item"),
        "onClick": function onClick(e) {
          return e.stopPropagation();
        }
      }, [Vue.createVNode(imageViewer_index.ImageViewer, Vue.mergeProps({
        "images": displayFiles.value.map(function (t) {
          return t.url || t.raw;
        }),
        "defaultIndex": index,
        "trigger": function trigger(h, _ref) {
          var open = _ref.open;
          return Vue.createVNode(BrowseIcon, {
            "onClick": function onClick(_ref2) {
              var _props$onPreview;
              var e = _ref2.e;
              (_props$onPreview = props.onPreview) === null || _props$onPreview === void 0 || _props$onPreview.call(props, {
                file: file,
                index: index,
                e: e
              });
              open();
            }
          }, null);
        }
      }, props.imageViewerProps), null)]), !props.disabled && Vue.createVNode(Vue.Fragment, null, [Vue.createVNode("span", {
        "class": "".concat(classPrefix.value, "-upload__card-mask-item-divider")
      }, null), Vue.createVNode("span", {
        "class": "".concat(classPrefix.value, "-upload__card-mask-item"),
        "onClick": function onClick(e) {
          return e.stopPropagation();
        }
      }, [Vue.createVNode(DeleteIcon, {
        "onClick": function onClick(_ref3) {
          var _props$onRemove;
          var e = _ref3.e;
          return props === null || props === void 0 || (_props$onRemove = props.onRemove) === null || _props$onRemove === void 0 ? void 0 : _props$onRemove.call(props, {
            e: e,
            file: file,
            index: index
          });
        }
      }, null)])])])]);
    };
    var renderProgressFile = function renderProgressFile(file, loadCard) {
      var _locale$value;
      return Vue.createVNode("div", {
        "class": [loadCard, "".concat(classPrefix.value, "-upload__").concat(props.theme, "-").concat(file.status)]
      }, [Vue.createVNode(loading_index.Loading, {
        "loading": true,
        "size": "medium"
      }, null), Vue.createVNode("p", null, [(_locale$value = locale.value) === null || _locale$value === void 0 || (_locale$value = _locale$value.progress) === null || _locale$value === void 0 ? void 0 : _locale$value.uploadingText, props.showUploadProgress ? " ".concat(file.percent, "%") : ""])]);
    };
    var renderFailFile = function renderFailFile(file, index, loadCard) {
      var _file$response, _locale$value2;
      return Vue.createVNode("div", {
        "class": loadCard
      }, [Vue.createVNode(ErrorCircleFilledIcon, null, null), Vue.createVNode("p", null, [((_file$response = file.response) === null || _file$response === void 0 ? void 0 : _file$response.error) || ((_locale$value2 = locale.value) === null || _locale$value2 === void 0 || (_locale$value2 = _locale$value2.progress) === null || _locale$value2 === void 0 ? void 0 : _locale$value2.failText)]), Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-upload__card-mask")
      }, [Vue.createVNode("span", {
        "class": "".concat(classPrefix.value, "-upload__card-mask-item"),
        "onClick": function onClick(e) {
          return e.stopPropagation();
        }
      }, [Vue.createVNode(DeleteIcon, {
        "onClick": function onClick(_ref4) {
          var _props$onRemove2;
          var e = _ref4.e;
          return props === null || props === void 0 || (_props$onRemove2 = props.onRemove) === null || _props$onRemove2 === void 0 ? void 0 : _props$onRemove2.call(props, {
            e: e,
            file: file,
            index: index
          });
        }
      }, null)])])]);
    };
    return function () {
      var _displayFiles$value2;
      var customList = renderTNodeJSX("fileListDisplay", {
        params: {
          triggerUpload: props.triggerUpload,
          uploadFiles: props.uploadFiles,
          cancelUpload: props.cancelUpload,
          onPreview: props.onPreview,
          onRemove: props.onRemove,
          toUploadFiles: props.toUploadFiles,
          sizeOverLimitMessage: props.sizeOverLimitMessage,
          locale: props.locale,
          files: displayFiles.value
        }
      });
      if (customList) return customList;
      var cardItemClasses = "".concat(classPrefix.value, "-upload__card-item ").concat(classPrefix.value, "-is-background");
      var renderTrigger = function renderTrigger() {
        var _locale$value3;
        return renderTNodeJSX("trigger", {
          params: {
            triggerUpload: props.triggerUpload
          }
        }) || Vue.createVNode("li", {
          "class": cardItemClasses,
          "onClick": props.triggerUpload
        }, [Vue.createVNode("div", {
          "class": ["".concat(classPrefix.value, "-upload__image-add"), "".concat(classPrefix.value, "-upload__card-container"), "".concat(classPrefix.value, "-upload__card-box"), _defineProperty__default["default"]({}, "".concat(classPrefix.value, "-is-disabled"), props.disabled)]
        }, [Vue.createVNode(AddIcon, null, null), Vue.createVNode("p", {
          "class": ["".concat(classPrefix.value, "-size-s"), "".concat(classPrefix.value, "-upload__add-text")]
        }, [(_locale$value3 = locale.value) === null || _locale$value3 === void 0 || (_locale$value3 = _locale$value3.triggerUploadText) === null || _locale$value3 === void 0 ? void 0 : _locale$value3.image])])]);
      };
      return Vue.createVNode("div", null, [Vue.createVNode("ul", {
        "class": "".concat(classPrefix.value, "-upload__card")
      }, [(_displayFiles$value2 = displayFiles.value) === null || _displayFiles$value2 === void 0 ? void 0 : _displayFiles$value2.map(function (file, index) {
        var fileNameClassName = "".concat(classPrefix.value, "-upload__card-name");
        var loadCard = "".concat(classPrefix.value, "-upload__card-container ").concat(classPrefix.value, "-upload__card-box");
        var fileName = props.abridgeName ? utils.abridgeName.apply(void 0, [file.name].concat(_toConsumableArray__default["default"](props.abridgeName))) : file.name;
        return Vue.createVNode("li", {
          "class": cardItemClasses,
          "key": index
        }, [file.status === "progress" && renderProgressFile(file, loadCard), file.status === "fail" && renderFailFile(file, index, loadCard), !["progress", "fail"].includes(file.status) && renderMainContent(file, index), Boolean(fileName && props.showImageFileName) && (file.url ? Vue.createVNode(link_index.Link, {
          "href": file.url,
          "class": fileNameClassName,
          "target": "_blank",
          "hover": "color",
          "size": "small",
          "disabled": false
        }, _isSlot(fileName) ? fileName : {
          "default": function _default() {
            return [fileName];
          }
        }) : Vue.createVNode("span", {
          "class": fileNameClassName
        }, [fileName]))]);
      }), showTrigger.value && renderTrigger()])]);
    };
  }
});

exports["default"] = ImageCard;
