// tdesign-vue-next/cjs/date-picker/components/base/Cell.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-449273df.js');
var index$1 = require('../../../_chunks/dep-c0cb4d13.js');
var index = require('../../../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../../_chunks/dep-497fc9a5.js');
var utils = require('../../../_chunks/dep-abacafd8.js');
var dayjs = require('dayjs');
require('../../../_chunks/dep-f9b59444.js');
require('../../../_chunks/dep-51846987.js');
require('../../../_chunks/dep-80827572.js');
require('../../../_chunks/dep-2610df9f.js');
require('../../../_chunks/dep-2c7a67b8.js');
require('../../../_chunks/dep-be760bc2.js');
require('../../../_chunks/dep-959f4847.js');
require('../../../_chunks/dep-6e64ef0e.js');
require('../../../_chunks/dep-9dfd682d.js');
require('../../../_chunks/dep-2db31625.js');
require('../../../_chunks/dep-0a17438d.js');
require('../../../_chunks/dep-5090b515.js');
require('../../../_chunks/dep-39313d05.js');
require('../../../_chunks/dep-f9bc7c82.js');
require('../../../_chunks/dep-1c6213b5.js');
require('../../../_chunks/dep-cfdee5e7.js');
require('../../../config-provider/hooks/useConfig.js');
require('../../../_chunks/dep-31abb282.js');
require('../../../_chunks/dep-c71a9cd7.js');
require('../../../_chunks/dep-39529ef1.js');
require('../../../_chunks/dep-f4c1cb5b.js');
require('../../../_chunks/dep-4896d30f.js');
require('../../../_chunks/dep-db6459dd.js');
require('../../../_chunks/dep-4d7a3a91.js');
require('../../../_chunks/dep-ae380218.js');
require('../../../_chunks/dep-fdac7521.js');
require('../../../_chunks/dep-df5442c5.js');
require('../../../_chunks/dep-141cbae1.js');
require('../../../_chunks/dep-c277c7cb.js');
require('../../../_chunks/dep-7653f80f.js');
require('../../../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');
require('../../../_chunks/dep-2a84fc5a.js');
require('../../../_chunks/dep-f12e600d.js');
require('../../../_chunks/dep-d303c019.js');
require('../../../_chunks/dep-b4528c21.js');
require('../../../_chunks/dep-75d3928f.js');
require('../../../_chunks/dep-ed620b03.js');
require('../../../_chunks/dep-64b2629b.js');
require('../../../_chunks/dep-ceabf07b.js');
require('../../../_chunks/dep-e600142f.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var TDatePickerCell = Vue.defineComponent({
  name: "TDatePickerCell",
  props: {
    time: String,
    text: [String, Number],
    value: Date,
    active: Boolean,
    highlight: Boolean,
    disabled: Boolean,
    startOfRange: Boolean,
    endOfRange: Boolean,
    hoverHighlight: Boolean,
    hoverStartOfRange: Boolean,
    hoverEndOfRange: Boolean,
    additional: Boolean,
    now: Boolean,
    firstDayOfMonth: Boolean,
    lastDayOfMonth: Boolean,
    onClick: Function,
    onMouseEnter: Function,
    cell: Function,
    dayjsObj: dayjs.Dayjs
  },
  setup: function setup(props) {
    var COMPONENT_NAME = index.usePrefixClass("date-picker__cell");
    var renderTNodeJSX = index$1.useTNodeJSX();
    var cellClass = Vue.computed(function () {
      var _ref;
      return [COMPONENT_NAME.value, (_ref = {}, _defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_ref, "".concat(COMPONENT_NAME.value, "--now"), props.now), "".concat(COMPONENT_NAME.value, "--active"), props.active), "".concat(COMPONENT_NAME.value, "--disabled"), props.disabled), "".concat(COMPONENT_NAME.value, "--highlight"), props.highlight), "".concat(COMPONENT_NAME.value, "--hover-highlight"), props.hoverHighlight), "".concat(COMPONENT_NAME.value, "--active-start"), props.startOfRange), "".concat(COMPONENT_NAME.value, "--active-end"), props.endOfRange), "".concat(COMPONENT_NAME.value, "--hover-start"), props.hoverStartOfRange), "".concat(COMPONENT_NAME.value, "--hover-end"), props.hoverEndOfRange), "".concat(COMPONENT_NAME.value, "--additional"), props.additional), _defineProperty__default["default"](_defineProperty__default["default"](_ref, "".concat(COMPONENT_NAME.value, "--first-day-of-month"), props.firstDayOfMonth), "".concat(COMPONENT_NAME.value, "--last-day-of-month"), props.lastDayOfMonth))];
    });
    function handleClick(e) {
      var _props$onClick;
      if (props.disabled) return;
      if (props.time) {
        var _extractTimeObj = utils.extractTimeObj(props.time),
          hours = _extractTimeObj.hours,
          minutes = _extractTimeObj.minutes,
          seconds = _extractTimeObj.seconds,
          milliseconds = _extractTimeObj.milliseconds,
          meridiem = _extractTimeObj.meridiem;
        var nextHours = hours;
        if (/am/i.test(meridiem) && nextHours === 12) nextHours -= 12;
        if (/pm/i.test(meridiem) && nextHours < 12) nextHours += 12;
        props.value.setHours(nextHours);
        props.value.setMinutes(minutes);
        props.value.setSeconds(seconds);
        props.value.setMilliseconds(milliseconds);
      }
      (_props$onClick = props.onClick) === null || _props$onClick === void 0 || _props$onClick.call(props, props.value, {
        e: e
      });
    }
    function handleMouseEnter() {
      var _props$onMouseEnter;
      if (props.disabled) return;
      if (props.time) {
        var _extractTimeObj2 = utils.extractTimeObj(props.time),
          hours = _extractTimeObj2.hours,
          minutes = _extractTimeObj2.minutes,
          seconds = _extractTimeObj2.seconds,
          milliseconds = _extractTimeObj2.milliseconds,
          meridiem = _extractTimeObj2.meridiem;
        var nextHours = hours;
        if (/am/i.test(meridiem) && nextHours === 12) nextHours -= 12;
        if (/pm/i.test(meridiem) && nextHours < 12) nextHours += 12;
        props.value.setHours(nextHours);
        props.value.setMinutes(minutes);
        props.value.setSeconds(seconds);
        props.value.setMilliseconds(milliseconds);
      }
      (_props$onMouseEnter = props.onMouseEnter) === null || _props$onMouseEnter === void 0 || _props$onMouseEnter.call(props, props.value);
    }
    return function () {
      var _renderTNodeJSX;
      return Vue.createVNode("td", {
        "class": cellClass.value,
        "onClick": handleClick,
        "onMouseenter": handleMouseEnter
      }, [Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "-inner")
      }, [(_renderTNodeJSX = renderTNodeJSX("cell", {
        params: {
          value: props.value
        }
      })) !== null && _renderTNodeJSX !== void 0 ? _renderTNodeJSX : props.text])]);
    };
  }
});

exports["default"] = TDatePickerCell;
