// tdesign-vue-next/cjs/date-picker/components/base/Footer.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-449273df.js');
var index = require('../../../_chunks/dep-c0cb4d13.js');
var index$1 = require('../../../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../../_chunks/dep-497fc9a5.js');
var button_index = require('../../../button/index.js');
var configProvider_utils_context = require('../../../_chunks/dep-31abb282.js');
var configProvider_hooks_useConfig = require('../../../config-provider/hooks/useConfig.js');
require('../../../_chunks/dep-f9b59444.js');
require('../../../_chunks/dep-51846987.js');
require('../../../_chunks/dep-80827572.js');
require('../../../_chunks/dep-2610df9f.js');
require('../../../_chunks/dep-2c7a67b8.js');
require('../../../_chunks/dep-be760bc2.js');
require('../../../_chunks/dep-959f4847.js');
require('../../../_chunks/dep-6e64ef0e.js');
require('../../../_chunks/dep-9dfd682d.js');
require('../../../_chunks/dep-2db31625.js');
require('../../../_chunks/dep-0a17438d.js');
require('../../../_chunks/dep-5090b515.js');
require('../../../_chunks/dep-39313d05.js');
require('../../../_chunks/dep-f9bc7c82.js');
require('../../../_chunks/dep-1c6213b5.js');
require('../../../_chunks/dep-cfdee5e7.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../../button/button.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-203a5570.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-3231adce.js');
require('../../../loading/icon/gradient.js');
require('../../../_chunks/dep-e3eb5659.js');
require('../../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../../_chunks/dep-6130f266.js');
require('../../../_chunks/dep-3ffe18b0.js');
require('../../../_chunks/dep-df5442c5.js');
require('../../../_chunks/dep-e41b3434.js');
require('../../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../../_chunks/dep-39529ef1.js');
require('../../../_chunks/dep-f4c1cb5b.js');
require('../../../_chunks/dep-4896d30f.js');
require('../../../_chunks/dep-db6459dd.js');
require('../../../_chunks/dep-4d7a3a91.js');
require('../../../_chunks/dep-ae380218.js');
require('../../../_chunks/dep-fdac7521.js');
require('../../../_chunks/dep-141cbae1.js');
require('../../../_chunks/dep-c277c7cb.js');
require('../../../_chunks/dep-7653f80f.js');
require('../../../_chunks/dep-6e7a2100.js');
require('../../../_chunks/dep-9ada3ec1.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-e46285c0.js');
require('../../../_chunks/dep-54801354.js');
require('../../../_chunks/dep-bd59ca66.js');
require('../../../_chunks/dep-f7508ffa.js');
require('../../../_chunks/dep-470f211b.js');
require('../../../_chunks/dep-9b549c85.js');
require('../../../_chunks/dep-e25ae470.js');
require('../../../_chunks/dep-bed84553.js');
require('../../../_chunks/dep-a4b963ab.js');
require('../../../_chunks/dep-abc005aa.js');
require('../../../button/props.js');
require('../../../_chunks/dep-a2e46896.js');
require('../../../_chunks/dep-6512c9d2.js');
require('../../../_chunks/dep-3cd26986.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var TDateFooter = Vue.defineComponent({
  name: "TDatePickerFooter",
  props: {
    enableTimePicker: Boolean,
    presetsPlacement: String,
    presets: Object,
    needConfirm: Boolean,
    selectedValue: [String, Number, Array, Date],
    onPresetClick: Function,
    onConfirmClick: Function
  },
  setup: function setup(props) {
    var renderTNodeJSX = index.useTNodeJSX();
    var COMPONENT_NAME = index$1.usePrefixClass("date-picker__footer");
    var presetsClass = index$1.usePrefixClass("date-picker__presets");
    var _useConfig = configProvider_hooks_useConfig.useConfig("datePicker"),
      t = _useConfig.t,
      globalConfig = _useConfig.globalConfig;
    var footerClass = Vue.computed(function () {
      return [COMPONENT_NAME.value, "".concat(COMPONENT_NAME.value, "--").concat(props.presetsPlacement)];
    });
    var renderPresets = function renderPresets() {
      if (configProvider_utils_context.isPlainObject(props.presets)) return Object.keys(props.presets).map(function (key) {
        return Vue.createVNode(button_index.Button, {
          "key": key,
          "size": "small",
          "variant": "text",
          "onClick": function onClick(e) {
            var _props$onPresetClick;
            return (_props$onPresetClick = props.onPresetClick) === null || _props$onPresetClick === void 0 ? void 0 : _props$onPresetClick.call(props, props.presets[key], {
              e: e,
              preset: _defineProperty__default["default"]({}, key, props.presets[key])
            });
          }
        }, _isSlot(key) ? key : {
          "default": function _default() {
            return [key];
          }
        });
      });
      var presetsNode = renderTNodeJSX("presets");
      return presetsNode !== null && presetsNode !== void 0 ? presetsNode : null;
    };
    return function () {
      var _slot;
      var presetsContent = renderPresets();
      var hasPresetsContent = Array.isArray(presetsContent) ? presetsContent.length > 0 : !!presetsContent;
      return Vue.createVNode("div", {
        "class": footerClass.value
      }, [hasPresetsContent ? Vue.createVNode("div", {
        "class": presetsClass.value
      }, [presetsContent]) : null, props.enableTimePicker && props.needConfirm && Vue.createVNode(button_index.Button, {
        "disabled": !props.selectedValue,
        "size": "small",
        "theme": "primary",
        "onClick": function onClick(e) {
          var _props$onConfirmClick;
          return (_props$onConfirmClick = props.onConfirmClick) === null || _props$onConfirmClick === void 0 ? void 0 : _props$onConfirmClick.call(props, {
            e: e
          });
        }
      }, _isSlot(_slot = t(globalConfig.value.confirm)) ? _slot : {
        "default": function _default() {
          return [_slot];
        }
      })]);
    };
  }
});

exports["default"] = TDateFooter;
