// tdesign-vue-next/cjs/date-picker/components/base/Header.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var pagination_index = require('../../../pagination/index.js');
var select_index = require('../../../select/index.js');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-449273df.js');
var index = require('../../../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../../_chunks/dep-497fc9a5.js');
require('@babel/runtime/helpers/defineProperty');
require('dayjs');
require('../../../_chunks/dep-abacafd8.js');
require('tdesign-icons-vue-next');
require('../../../_chunks/dep-d303c019.js');
var datePicker_hooks_useSelectRange = require('../../hooks/useSelectRange.js');
var configProvider_hooks_useConfig = require('../../../config-provider/hooks/useConfig.js');
require('../../../pagination/pagination.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../../_chunks/dep-c0cb4d13.js');
require('../../../_chunks/dep-51846987.js');
require('../../../_chunks/dep-80827572.js');
require('../../../_chunks/dep-2610df9f.js');
require('../../../_chunks/dep-2c7a67b8.js');
require('../../../_chunks/dep-be760bc2.js');
require('../../../_chunks/dep-959f4847.js');
require('../../../_chunks/dep-6e64ef0e.js');
require('../../../_chunks/dep-9dfd682d.js');
require('../../../_chunks/dep-2db31625.js');
require('../../../_chunks/dep-0a17438d.js');
require('../../../_chunks/dep-5090b515.js');
require('../../../_chunks/dep-39313d05.js');
require('../../../_chunks/dep-f9bc7c82.js');
require('../../../_chunks/dep-1c6213b5.js');
require('../../../_chunks/dep-cfdee5e7.js');
require('../../../_chunks/dep-4b52068a.js');
require('../../../_chunks/dep-88ae5d8a.js');
require('../../../_chunks/dep-d0227fbe.js');
require('../../../input-number/index.js');
require('../../../input-number/input-number.js');
require('../../../button/index.js');
require('../../../button/button.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-203a5570.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-3231adce.js');
require('../../../loading/icon/gradient.js');
require('../../../_chunks/dep-e3eb5659.js');
require('../../../_chunks/dep-cce7af6a.js');
require('../../../_chunks/dep-6130f266.js');
require('../../../_chunks/dep-3ffe18b0.js');
require('../../../_chunks/dep-df5442c5.js');
require('../../../_chunks/dep-e41b3434.js');
require('../../../_chunks/dep-f9b59444.js');
require('../../../_chunks/dep-31abb282.js');
require('../../../_chunks/dep-c71a9cd7.js');
require('../../../_chunks/dep-39529ef1.js');
require('../../../_chunks/dep-f4c1cb5b.js');
require('../../../_chunks/dep-4896d30f.js');
require('../../../_chunks/dep-db6459dd.js');
require('../../../_chunks/dep-4d7a3a91.js');
require('../../../_chunks/dep-ae380218.js');
require('../../../_chunks/dep-fdac7521.js');
require('../../../_chunks/dep-141cbae1.js');
require('../../../_chunks/dep-c277c7cb.js');
require('../../../_chunks/dep-7653f80f.js');
require('../../../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../../_chunks/dep-9ada3ec1.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-e46285c0.js');
require('../../../_chunks/dep-54801354.js');
require('../../../_chunks/dep-bd59ca66.js');
require('../../../_chunks/dep-f7508ffa.js');
require('../../../_chunks/dep-470f211b.js');
require('../../../_chunks/dep-9b549c85.js');
require('../../../_chunks/dep-e25ae470.js');
require('../../../_chunks/dep-bed84553.js');
require('../../../_chunks/dep-a4b963ab.js');
require('../../../_chunks/dep-abc005aa.js');
require('../../../button/props.js');
require('../../../_chunks/dep-a2e46896.js');
require('../../../_chunks/dep-6512c9d2.js');
require('../../../_chunks/dep-3cd26986.js');
require('../../../input/index.js');
require('../../../input/input.js');
require('../../../input/props.js');
require('../../../_chunks/dep-f02d9258.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../_chunks/dep-b4528c21.js');
require('../../../_chunks/dep-d20c4e5e.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../_chunks/dep-ba2511b0.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../input-number/props.js');
require('../../../_chunks/dep-5ea73cce.js');
require('../../../input-number/hooks/useInputNumber.js');
require('../../../_chunks/dep-74a0ebc2.js');
require('../../../_chunks/dep-94323f13.js');
require('../../../_chunks/dep-75d3928f.js');
require('../../../input-adornment/index.js');
require('../../../input-adornment/input-adornment.js');
require('../../../input-adornment/props.js');
require('../../../pagination/props.js');
require('../../../pagination/hooks/useMoreAction.js');
require('../../../pagination/hooks/usePaginationClasses.js');
require('../../../select/select.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../../common-components/fake-arrow.js');
require('../../../select-input/index.js');
require('../../../select-input/select-input.js');
require('../../../popup/index.js');
require('../../../popup/popup.js');
require('@popperjs/core');
require('../../../popup/container.js');
require('../../../popup/props.js');
require('../../../_chunks/dep-4e2a4791.js');
require('../../../_chunks/dep-f3377589.js');
require('../../../_chunks/dep-71d9daf0.js');
require('../../../select-input/props.js');
require('../../../select-input/hooks/useMultiple.js');
require('../../../tag-input/index.js');
require('../../../tag-input/tag-input.js');
require('../../../tag-input/props.js');
require('../../../tag-input/hooks/useDragSorter.js');
require('../../../tag-input/hooks/useHover.js');
require('../../../tag-input/hooks/useTagScroll.js');
require('../../../tag-input/hooks/useTagList.js');
require('../../../tag/index.js');
require('../../../tag/tag.js');
require('tinycolor2');
require('../../../tag/props.js');
require('../../../tag/check-tag.js');
require('../../../tag/check-tag-props.js');
require('../../../_chunks/dep-2e8277b1.js');
require('../../../tag/check-tag-group.js');
require('../../../tag/check-tag-group-props.js');
require('../../../select-input/hooks/useOverlayInnerStyle.js');
require('../../../select-input/hooks/useSingle.js');
require('../../../_chunks/dep-008d4a55.js');
require('../../../_chunks/dep-0e1751ee.js');
require('../../../_chunks/dep-4e2907e0.js');
require('../../../select/components/select-panel.js');
require('../../../select/option.js');
require('../../../select/option-props.js');
require('../../../checkbox/index.js');
require('../../../checkbox/checkbox.js');
require('../../../checkbox/props.js');
require('../../../checkbox/constants/index.js');
require('../../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../../_chunks/dep-12e6ec24.js');
require('../../../checkbox/hooks/useKeyboardEvent.js');
require('../../../checkbox/group.js');
require('../../../checkbox/checkbox-group-props.js');
require('../../../_chunks/dep-cec24bbf.js');
require('../../../_chunks/dep-f786e1df.js');
require('../../../_chunks/dep-618d56e7.js');
require('../../../_chunks/dep-2cf25d25.js');
require('../../../select/utils/index.js');
require('../../../select/constants/index.js');
require('../../../select/option-group.js');
require('../../../select/option-group-props.js');
require('../../../select/props.js');
require('../../../select/hooks/usePanelVirtualScroll.js');
require('../../../_chunks/dep-3a800699.js');
require('../../../_chunks/dep-648bfb84.js');
require('../../../select/hooks/useKeyboardControl.js');
require('../../../select/hooks/useSelectOptions.js');
require('../../../_chunks/dep-8f2069e9.js');
require('../../../pagination/pagination-mini.js');
require('../../../pagination/pagination-mini-props.js');
require('../../../tooltip/index.js');
require('../../../tooltip/tooltip.js');
require('../../../tooltip/props.js');
require('../../../tooltip/utils/index.js');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');
require('../../../_chunks/dep-2a84fc5a.js');
require('../../../_chunks/dep-f12e600d.js');
require('../../../_chunks/dep-ed620b03.js');
require('../../../_chunks/dep-64b2629b.js');
require('../../../_chunks/dep-ceabf07b.js');
require('../../../_chunks/dep-e600142f.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);

var TDateHeader = Vue.defineComponent({
  name: "TDatePickerHeader",
  props: {
    mode: {
      type: String,
      "default": "date"
    },
    range: [Array, Function],
    year: Number,
    month: Number,
    internalYear: Array,
    partial: String,
    onMonthChange: Function,
    onYearChange: Function,
    onJumperClick: Function
  },
  setup: function setup(props) {
    var _toRefs = Vue.toRefs(props),
      year = _toRefs.year,
      month = _toRefs.month;
    var _useConfig = configProvider_hooks_useConfig.useConfig("classPrefix"),
      classPrefix = _useConfig.classPrefix;
    var COMPONENT_NAME = index.usePrefixClass("date-picker__header");
    var _useConfig2 = configProvider_hooks_useConfig.useConfig("datePicker"),
      globalConfig = _useConfig2.globalConfig;
    var _useSelectRange = datePicker_hooks_useSelectRange.useSelectRange({
        range: props.range,
        mode: props.mode,
        year: year,
        month: month
      }),
      monthHasAnyAllowed = _useSelectRange.monthHasAnyAllowed,
      yearHasAnyAllowed = _useSelectRange.yearHasAnyAllowed,
      decadeHasAnyAllowed = _useSelectRange.decadeHasAnyAllowed,
      paginationDisabled = _useSelectRange.paginationDisabled,
      canLoadMoreTop = _useSelectRange.canLoadMoreTop,
      canLoadMoreBottom = _useSelectRange.canLoadMoreBottom;
    var yearOptions = Vue.ref(initOptions(props.year));
    var showMonthPicker = Vue.computed(function () {
      return props.mode === "date" || props.mode === "week";
    });
    var nearestYear = Vue.computed(function () {
      var _yearOptions$value$fi;
      var extraYear = props.partial === "end" && props.mode === "year" && Number(props.internalYear[1]) - Number(props.internalYear[0]) <= 9 ? 9 : 0;
      return ((_yearOptions$value$fi = yearOptions.value.find(function (option) {
        return option.value - (props.year + extraYear) <= 9 && option.value - (props.year + extraYear) >= 0;
      })) === null || _yearOptions$value$fi === void 0 ? void 0 : _yearOptions$value$fi.value) || props.year;
    });
    var monthOptions = Vue.computed(function () {
      return globalConfig.value.months.map(function (item, index) {
        return {
          label: item,
          value: index,
          disabled: !monthHasAnyAllowed(props.year, index)
        };
      });
    });
    var showPanelTop = Vue.computed(function () {
      var options = yearOptions.value;
      if (!options.length) return false;
      var first = options[0].value;
      return canLoadMoreTop(first);
    });
    var showPanelBottom = Vue.computed(function () {
      var options = yearOptions.value;
      if (!options.length) return false;
      var last = options[options.length - 1].value;
      return canLoadMoreBottom(last);
    });
    function initOptions(year2) {
      var options = [];
      if (props.mode === "year") {
        var extraYear = year2 % 10;
        var minYear = year2 - extraYear - 100;
        var maxYear = year2 - extraYear + 100;
        for (var i = minYear; i <= maxYear; i += 10) {
          var end = i + 9;
          if (decadeHasAnyAllowed(end)) {
            options.push({
              label: "".concat(i, " - ").concat(end),
              value: end,
              disabled: false
            });
          }
        }
      } else {
        if (yearHasAnyAllowed(year2)) options.push({
          label: "".concat(year2),
          value: year2,
          disabled: false
        });
        for (var _i = 1; _i <= 10; _i++) {
          if (yearHasAnyAllowed(year2 + _i)) options.push({
            label: "".concat(year2 + _i),
            value: year2 + _i,
            disabled: false
          });
          if (yearHasAnyAllowed(year2 - _i)) options.unshift({
            label: "".concat(year2 - _i),
            value: year2 - _i,
            disabled: false
          });
        }
      }
      return options;
    }
    function loadMoreYear(year2, type) {
      var options = [];
      if (props.mode === "year") {
        var extraYear = year2 % 10;
        if (type === "add") {
          for (var i = year2 - extraYear + 10; i <= year2 - extraYear + 50; i += 10) {
            var end = i + 9;
            if (decadeHasAnyAllowed(end)) options.push({
              label: "".concat(i, " - ").concat(end),
              value: end,
              disabled: false
            });
          }
        } else {
          for (var _i2 = year2 - extraYear - 1; _i2 > year2 - extraYear - 50; _i2 -= 10) {
            if (decadeHasAnyAllowed(_i2)) options.unshift({
              label: "".concat(_i2 - 9, " - ").concat(_i2),
              value: _i2,
              disabled: false
            });
          }
        }
      } else if (type === "add") {
        for (var _i3 = year2 + 1; _i3 <= year2 + 10; _i3++) {
          if (yearHasAnyAllowed(_i3)) options.push({
            label: "".concat(_i3),
            value: _i3,
            disabled: false
          });
        }
      } else {
        for (var _i4 = year2 - 1; _i4 > year2 - 10; _i4--) {
          if (yearHasAnyAllowed(_i4)) options.unshift({
            label: "".concat(_i4),
            value: _i4,
            disabled: false
          });
        }
      }
      return options;
    }
    var labelMap = Vue.computed(function () {
      return {
        year: {
          prev: globalConfig.value.preDecade,
          current: globalConfig.value.now,
          next: globalConfig.value.nextDecade
        },
        month: {
          prev: globalConfig.value.preYear,
          current: globalConfig.value.now,
          next: globalConfig.value.nextYear
        },
        date: {
          prev: globalConfig.value.preMonth,
          current: globalConfig.value.now,
          next: globalConfig.value.nextMonth
        },
        quarter: {
          prev: globalConfig.value.preYear,
          current: globalConfig.value.now,
          next: globalConfig.value.nextYear
        },
        week: {
          prev: globalConfig.value.preMonth,
          current: globalConfig.value.now,
          next: globalConfig.value.nextMonth
        }
      };
    });
    function handleScroll(_ref) {
      var e = _ref.e;
      if (e.target.scrollTop === 0) {
        if (showPanelTop.value) handlePanelTopClick(e);
      } else if (e.target.scrollTop === e.target.scrollHeight - e.target.clientHeight) {
        if (showPanelBottom.value) handlePanelBottomClick(e);
      }
    }
    function handlePanelTopClick(e) {
      var _e$stopPropagation;
      e === null || e === void 0 || (_e$stopPropagation = e.stopPropagation) === null || _e$stopPropagation === void 0 || _e$stopPropagation.call(e);
      var firstYear = yearOptions.value[0].value;
      var options = loadMoreYear(firstYear, "reduce");
      yearOptions.value = [].concat(_toConsumableArray__default["default"](options), _toConsumableArray__default["default"](yearOptions.value));
    }
    function handlePanelBottomClick(e) {
      var _e$stopPropagation2;
      e === null || e === void 0 || (_e$stopPropagation2 = e.stopPropagation) === null || _e$stopPropagation2 === void 0 || _e$stopPropagation2.call(e);
      var lastYear = yearOptions.value.slice(-1)[0].value;
      var options = loadMoreYear(lastYear, "add");
      yearOptions.value = [].concat(_toConsumableArray__default["default"](yearOptions.value), _toConsumableArray__default["default"](options));
    }
    Vue.watch(function () {
      return props.mode;
    }, function () {
      yearOptions.value = initOptions(props.year);
    });
    return function () {
      return Vue.createVNode("div", {
        "class": COMPONENT_NAME.value
      }, [Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "-controller")
      }, [showMonthPicker.value && Vue.createVNode(select_index.Select, {
        "class": "".concat(COMPONENT_NAME.value, "-controller-month"),
        "value": props.month,
        "options": monthOptions.value,
        "onChange": function onChange(val) {
          var _props$onMonthChange;
          (_props$onMonthChange = props.onMonthChange) === null || _props$onMonthChange === void 0 || _props$onMonthChange.call(props, val);
        },
        "popupProps": {
          attach: function attach(triggerElement) {
            return triggerElement.parentNode;
          },
          overlayClassName: "".concat(COMPONENT_NAME.value, "-controller-month-popup")
        }
      }, null), Vue.createVNode(select_index.Select, {
        "class": "".concat(COMPONENT_NAME.value, "-controller-year"),
        "value": props.mode === "year" ? nearestYear.value : props.year,
        "options": yearOptions.value,
        "onChange": function onChange(val) {
          var _props$onYearChange;
          (_props$onYearChange = props.onYearChange) === null || _props$onYearChange === void 0 || _props$onYearChange.call(props, val);
        },
        "popupProps": {
          onScroll: handleScroll,
          attach: function attach(triggerElement) {
            return triggerElement.parentNode;
          },
          overlayClassName: "".concat(COMPONENT_NAME.value, "-controller-year-popup")
        },
        "panelTopContent": function panelTopContent() {
          return showPanelTop.value && Vue.createVNode("div", {
            "class": "".concat(classPrefix.value, "-select-option"),
            "onClick": handlePanelTopClick
          }, [Vue.createTextVNode("...")]);
        },
        "panelBottomContent": function panelBottomContent() {
          return showPanelBottom.value && Vue.createVNode("div", {
            "class": "".concat(classPrefix.value, "-select-option"),
            "onClick": handlePanelBottomClick
          }, [Vue.createTextVNode("...")]);
        }
      }, null)]), Vue.createVNode(pagination_index.PaginationMini, {
        "tips": labelMap.value[props.mode],
        "size": "small",
        "onChange": props.onJumperClick,
        "disabled": paginationDisabled.value
      }, null)]);
    };
  }
});

exports["default"] = TDateHeader;
