// tdesign-vue-next/cjs/cascader/components/Panel.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var cascader_components_Item = require('./Item.js');
var cascader_props = require('../props.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-449273df.js');
var renderTnode = require('../../_chunks/dep-9dfd682d.js');
var index = require('../../_chunks/dep-c0cb4d13.js');
var index$1 = require('../../_chunks/dep-95cb1381.js');
require('../../_chunks/dep-497fc9a5.js');
var cascader_utils_effect = require('../utils/effect.js');
var cascader_utils_helper = require('../utils/helper.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
require('tdesign-icons-vue-next');
require('../utils/className.js');
require('../../checkbox/index.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../_chunks/dep-9ada3ec1.js');
require('../../_chunks/dep-a2e46896.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-6512c9d2.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-6e64ef0e.js');
require('../../_chunks/dep-f02d9258.js');
require('../../_chunks/dep-3cd26986.js');
require('../../_chunks/dep-e3eb5659.js');
require('../../_chunks/dep-d0227fbe.js');
require('../../_chunks/dep-cfdee5e7.js');
require('../../_chunks/dep-1c6213b5.js');
require('../../_chunks/dep-0a17438d.js');
require('../../_chunks/dep-5090b515.js');
require('../../checkbox/constants/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../_chunks/dep-12e6ec24.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../_chunks/dep-2e8277b1.js');
require('../../_chunks/dep-be760bc2.js');
require('../../_chunks/dep-f9b59444.js');
require('../../_chunks/dep-51846987.js');
require('../../_chunks/dep-80827572.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../../_chunks/dep-2db31625.js');
require('../../_chunks/dep-39313d05.js');
require('../../_chunks/dep-f9bc7c82.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../_chunks/dep-cec24bbf.js');
require('../../_chunks/dep-f786e1df.js');
require('../../_chunks/dep-bd59ca66.js');
require('../../_chunks/dep-618d56e7.js');
require('../../_chunks/dep-2cf25d25.js');
require('../../_chunks/dep-abc005aa.js');
require('../../loading/index.js');
require('../../_chunks/dep-203a5570.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-3231adce.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6130f266.js');
require('../../_chunks/dep-3ffe18b0.js');
require('../../_chunks/dep-e41b3434.js');
require('../../loading/props.js');
require('../../_chunks/dep-e46285c0.js');
require('../../_chunks/dep-54801354.js');
require('../../_chunks/dep-f7508ffa.js');
require('../../_chunks/dep-470f211b.js');
require('../../_chunks/dep-9b549c85.js');
require('../../_chunks/dep-e25ae470.js');
require('../../_chunks/dep-bed84553.js');
require('../../_chunks/dep-a4b963ab.js');
require('../../_chunks/dep-88ae5d8a.js');
require('../../_chunks/dep-d20c4e5e.js');
require('../../_chunks/dep-a73f6bfc.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TCascaderSubPanel = Vue.defineComponent({
  name: "TCascaderSubPanel",
  props: {
    option: cascader_props["default"].option,
    options: cascader_props["default"].options,
    empty: cascader_props["default"].empty,
    trigger: cascader_props["default"].trigger,
    onChange: cascader_props["default"].onChange,
    loading: cascader_props["default"].loading,
    loadingText: cascader_props["default"].loadingText,
    cascaderContext: {
      type: Object
    }
  },
  setup: function setup(props) {
    var renderTNodeJSXDefault = index.useTNodeDefault();
    var renderTNodeJSX = index.useTNodeJSX();
    var COMPONENT_NAME = index$1.usePrefixClass("cascader");
    var _useConfig = configProvider_hooks_useConfig.useConfig("cascader"),
      globalConfig = _useConfig.globalConfig;
    var panels = Vue.computed(function () {
      return cascader_utils_helper.getPanels(props.cascaderContext.treeNodes);
    });
    var filterState = Vue.shallowRef(null);
    var hasActiveFilter = Vue.computed(function () {
      var state = filterState.value;
      return state && hasAnyActiveFilter(state.filters);
    });
    var getFilteredNodes = function getFilteredNodes(nodes, index) {
      var state = filterState.value;
      if (!state) return nodes;
      var filter = state.filters[index];
      if (!filter) return nodes;
      return cascader_utils_helper.filterOptions(nodes, filter, index);
    };
    var hasAnyActiveFilter = function hasAnyActiveFilter(filters) {
      return Object.values(filters).some(function (f) {
        return cascader_utils_helper.isFilterActive(f);
      });
    };
    var clearExpiredFilters = function clearExpiredFilters(filters, maxLevel) {
      return Object.fromEntries(Object.entries(filters).filter(function (_ref) {
        var _ref2 = _slicedToArray__default["default"](_ref, 1),
          panelIndex = _ref2[0];
        return Number(panelIndex) <= maxLevel;
      }));
    };
    var calculateCascadeMaxLevel = function calculateCascadeMaxLevel(panelIndex, filteredNodes, currentMaxLevel) {
      if (filteredNodes.length === 0) {
        return panelIndex;
      }
      return Math.max(panelIndex, currentMaxLevel);
    };
    var handleFilter = function handleFilter(index, filter) {
      var _prev$maxLevel, _prev$maxLevel2;
      var prev = filterState.value;
      var filters = _objectSpread({}, prev === null || prev === void 0 ? void 0 : prev.filters);
      if (cascader_utils_helper.isFilterActive(filter)) {
        filters[index] = filter;
      } else {
        delete filters[index];
      }
      var maxLevel = (_prev$maxLevel = prev === null || prev === void 0 ? void 0 : prev.maxLevel) !== null && _prev$maxLevel !== void 0 ? _prev$maxLevel : cascader_utils_helper.FILTER_INACTIVE_LEVEL;
      if (cascader_utils_helper.isFilterActive(filter)) {
        var currentNodes = panels.value[index] || [];
        var filteredNodes = cascader_utils_helper.filterOptions(currentNodes, filter, index);
        maxLevel = calculateCascadeMaxLevel(index, filteredNodes, maxLevel);
      } else if (!hasAnyActiveFilter(filters)) {
        maxLevel = cascader_utils_helper.FILTER_INACTIVE_LEVEL;
      }
      if (maxLevel < ((_prev$maxLevel2 = prev === null || prev === void 0 ? void 0 : prev.maxLevel) !== null && _prev$maxLevel2 !== void 0 ? _prev$maxLevel2 : cascader_utils_helper.FILTER_INACTIVE_LEVEL)) {
        filters = clearExpiredFilters(filters, maxLevel);
      }
      filterState.value = {
        filters: filters,
        maxLevel: maxLevel
      };
    };
    var shouldShowPanel = function shouldShowPanel(index) {
      var state = filterState.value;
      if (!hasActiveFilter.value || !state || !cascader_utils_helper.isFilterLevelActive(state.maxLevel)) {
        return true;
      }
      return index <= state.maxLevel;
    };
    var handleExpand = function handleExpand(node, trigger, level) {
      var state = filterState.value;
      var children = node.children;
      if (state && cascader_utils_helper.isFilterLevelActive(state.maxLevel) && props.trigger === trigger && Array.isArray(children) && children.length) {
        var childLevel = level + 1;
        if (childLevel > state.maxLevel) {
          var cleanedFilters = clearExpiredFilters(state.filters, childLevel);
          filterState.value = {
            filters: cleanedFilters,
            maxLevel: childLevel
          };
        }
      }
      cascader_utils_effect.expandClickEffect(props.trigger, trigger, node, props.cascaderContext);
    };
    var onFilterCallbacks = /* @__PURE__ */new Map();
    var getOnFilterCallback = function getOnFilterCallback(index) {
      var callback = onFilterCallbacks.get(index);
      if (!callback) {
        callback = function callback(filter) {
          return handleFilter(index, filter);
        };
        onFilterCallbacks.set(index, callback);
      }
      return callback;
    };
    Vue.watch(panels, function (newPanels) {
      var maxIndex = newPanels.length - 1;
      var _iterator = _createForOfIteratorHelper(onFilterCallbacks),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var _step$value = _slicedToArray__default["default"](_step.value, 1),
            index = _step$value[0];
          if (index > maxIndex) {
            onFilterCallbacks["delete"](index);
          }
        }
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
    }, {
      flush: "post"
    });
    Vue.onUnmounted(function () {
      onFilterCallbacks.clear();
    });
    var renderItem = function renderItem(node, index) {
      var optionChild = node.data.content ? renderTnode.getDefaultNode(node.data.content(Vue.h)) : renderTNodeJSXDefault("option", {
        params: {
          item: node.data,
          index: index,
          onExpand: function onExpand() {
            return handleExpand(node, "click", index);
          },
          onChange: function onChange() {
            return cascader_utils_effect.valueChangeEffect(node, props.cascaderContext);
          }
        }
      });
      return Vue.createVNode(cascader_components_Item["default"], {
        "key": node.value,
        "node": node,
        "optionChild": optionChild,
        "cascaderContext": props.cascaderContext,
        "onClick": function onClick() {
          return handleExpand(node, "click", index);
        },
        "onMouseenter": function onMouseenter() {
          return handleExpand(node, "hover", index);
        },
        "onChange": function onChange() {
          return cascader_utils_effect.valueChangeEffect(node, props.cascaderContext);
        }
      }, null);
    };
    var renderList = function renderList(treeNodes) {
      var segment = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : true;
      var index = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
      var displayNodes = hasActiveFilter.value ? getFilteredNodes(treeNodes, index) : treeNodes;
      var columnParams = {
        panelIndex: index,
        options: treeNodes.map(function (node) {
          return node.data;
        }),
        filteredOptions: displayNodes.map(function (node) {
          return node.data;
        })
      };
      return Vue.createVNode("ul", {
        "class": ["".concat(COMPONENT_NAME.value, "__menu"), "narrow-scrollbar", _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "__menu--segment"), segment)],
        "key": "".concat(COMPONENT_NAME, "__menu").concat(index)
      }, [renderTNodeJSX("columnHeader", {
        params: _objectSpread(_objectSpread({}, columnParams), {}, {
          onFilter: getOnFilterCallback(index)
        })
      }), displayNodes.map(function (node) {
        return renderItem(node, index);
      }), renderTNodeJSX("columnFooter", {
        params: _objectSpread(_objectSpread({}, columnParams), {}, {
          onFilter: getOnFilterCallback(index)
        })
      })]);
    };
    var noop = function noop() {};
    var renderFilteredList = function renderFilteredList(treeNodes) {
      var columnParams = {
        panelIndex: 0,
        options: treeNodes.map(function (node) {
          return node.data;
        }),
        filteredOptions: treeNodes.map(function (node) {
          return node.data;
        }),
        onFilter: noop
      };
      return Vue.createVNode("ul", {
        "class": ["".concat(COMPONENT_NAME.value, "__menu"), "narrow-scrollbar", "".concat(COMPONENT_NAME.value, "__menu--filter")],
        "key": "".concat(COMPONENT_NAME, "__menu--filtered")
      }, [renderTNodeJSX("columnHeader", {
        params: columnParams
      }), treeNodes.map(function (node) {
        return renderItem(node, 0);
      }), renderTNodeJSX("columnFooter", {
        params: columnParams
      })]);
    };
    var renderPanels = function renderPanels() {
      var _props$cascaderContex = props.cascaderContext,
        inputVal = _props$cascaderContex.inputVal,
        treeNodes = _props$cascaderContex.treeNodes;
      if (inputVal) return renderFilteredList(treeNodes);
      var result = [];
      var len = panels.value.length;
      for (var i = 0; i < len; i++) {
        if (shouldShowPanel(i)) {
          result.push(renderList(panels.value[i], i !== len - 1, i));
        }
      }
      return result;
    };
    return function () {
      var content;
      if (props.loading) {
        content = renderTNodeJSXDefault("loadingText", Vue.createVNode("div", {
          "class": "".concat(COMPONENT_NAME.value, "__panel--empty")
        }, [globalConfig.value.loadingText]));
      } else {
        content = panels.value.length ? renderPanels() : renderTNodeJSXDefault("empty", Vue.createVNode("div", {
          "class": "".concat(COMPONENT_NAME.value, "__panel--empty")
        }, [globalConfig.value.empty]));
      }
      return Vue.createVNode("div", {
        "class": ["".concat(COMPONENT_NAME.value, "__panel"), _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--normal"), panels.value.length && !props.loading)]
      }, [content]);
    };
  }
});

exports["default"] = TCascaderSubPanel;
