// tdesign-vue-next/lib/range-input/range-input-popup.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode, mergeProps } from 'vue';
import _defineProperty from '@babel/runtime/helpers/defineProperty';
import { isArray } from 'lodash-es';
import { Popup } from '../popup/index.js';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../_chunks/dep-d9440b5f.js';
import { u as usePrefixClass } from '../_chunks/dep-4d7f902f.js';
import '@babel/runtime/helpers/slicedToArray';
import '../_chunks/dep-dfeea6f5.js';
import _RangeInput from './range-input.js';
import props from './range-input-popup-props.js';
import { useOverlayInnerStyle } from '../select-input/hooks/useOverlayInnerStyle.js';
import '../popup/popup.js';
import '@popperjs/core';
import '../_chunks/dep-41cafcc6.js';
import '../_chunks/dep-b3575781.js';
import '../_chunks/dep-43a3baa0.js';
import '../_chunks/dep-9e2a2dff.js';
import '../config-provider/hooks/useConfig.js';
import '../config-provider/utils/context.js';
import '../_chunks/dep-854d2777.js';
import '../_chunks/dep-a6042a25.js';
import 'dayjs';
import '../_chunks/dep-cdc7f78f.js';
import '../_chunks/dep-cc2cf248.js';
import '../popup/container.js';
import '../popup/props.js';
import '../_chunks/dep-b239e51b.js';
import '../_chunks/dep-c53cb022.js';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import 'tdesign-icons-vue-next';
import '../input/index.js';
import '../input/input.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../input/props.js';
import '../_chunks/dep-a7c4f6fa.js';
import '../_chunks/dep-b4a20622.js';
import '../_chunks/dep-8079b4fa.js';
import '../input/hooks/useInput.js';
import '../form/constants/index.js';
import '../input/hooks/useLengthLimit.js';
import '../_chunks/dep-2721b80a.js';
import '../_chunks/dep-9e860115.js';
import '../input/hooks/useInputEventHandler.js';
import '../input/hooks/useInputWidth.js';
import '../input/input-group.js';
import '../input/input-group-props.js';
import './props.js';
import '../_chunks/dep-03d84be1.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var _RangeInputPopup = defineComponent({
  name: "TRangeInputPopup",
  props: props,
  setup: function setup(props2) {
    var COMPONENT_NAME = usePrefixClass("range-input-popup");
    var _useOverlayInnerStyle = useOverlayInnerStyle(props2),
      tOverlayInnerStyle = _useOverlayInnerStyle.tOverlayInnerStyle,
      innerPopupVisible = _useOverlayInnerStyle.innerPopupVisible,
      onInnerPopupVisibleChange = _useOverlayInnerStyle.onInnerPopupVisibleChange;
    var popupClasses = computed(function () {
      return [COMPONENT_NAME.value, _defineProperty({}, "".concat(COMPONENT_NAME.value, "--visible"), props2.popupVisible || innerPopupVisible.value)];
    });
    return function () {
      return createVNode("div", {
        "class": popupClasses.value
      }, [createVNode(Popup, mergeProps({
        "hideEmptyPopup": true,
        "content": props2.panel,
        "trigger": "click",
        "placement": "bottom-left",
        "disabled": isArray(props2.disabled) ? props2.disabled.every(Boolean) : props2.disabled,
        "visible": props2.popupVisible || innerPopupVisible.value
      }, _objectSpread({
        overlayInnerStyle: tOverlayInnerStyle.value,
        onVisibleChange: onInnerPopupVisibleChange
      }, props2.popupProps)), {
        "default": function _default() {
          return [createVNode(_RangeInput, _objectSpread({
            value: props2.inputValue,
            onChange: props2.onInputChange,
            disabled: props2.disabled,
            label: props2.label,
            status: props2.status,
            tips: props2.tips
          }, props2.rangeInputProps), null)];
        }
      })]);
    };
  }
});

export { _RangeInputPopup as default };
