// tdesign-vue-next/es/color-picker/utils/index.mjs
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

export { C as Color, D as Draggable, d as cmyk2rgb, e as cmykInputToColor, b as genGradientPoint, h as genId, k as getColorFormatInputs, j as getColorFormatMap, a as getColorFormatOptions, g as getColorObject, f as getColorWithoutAlpha, c as gradientColors2string, i as initColorFormat, l as isGradientColor, p as parseGradientString, r as rgb2cmyk } from '../../_chunks/dep-6c281967.mjs';
export { A as ALPHA_FORMAT_MAP, C as COLOR_FORMAT_INPUTS, c as COLOR_MODES, a as DEFAULT_COLOR, D as DEFAULT_LINEAR_GRADIENT, b as DEFAULT_SYSTEM_SWATCH_COLORS, F as FORMATS, G as GRADIENT_SLIDER_DEFAULT_WIDTH, d as SATURATION_PANEL_DEFAULT_HEIGHT, S as SATURATION_PANEL_DEFAULT_WIDTH, e as SLIDER_DEFAULT_WIDTH, T as TD_COLOR_USED_COLORS_MAX_SIZE } from '../../_chunks/dep-863731bb.mjs';
import '../../_chunks/dep-ea5fbe21.mjs';
import '../../_chunks/dep-26d65784.mjs';
import '../../_chunks/dep-a4f55e3b.mjs';
import '../../_chunks/dep-98dac781.mjs';
import '../../_chunks/dep-f038e2b4.mjs';
import '../../_chunks/dep-a8b40385.mjs';
import 'lodash-es';
import '../../_chunks/dep-557b68dd.mjs';
