// tdesign-vue-next/es/color-picker/components/panel/alpha.mjs
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode } from 'vue';
import ColorSlider from './slider.mjs';
import { useBaseClassName } from '../../hooks/index.mjs';
import '../../../_chunks/dep-98dac781.mjs';
import '../../../_chunks/dep-f038e2b4.mjs';
import '../../../_chunks/dep-863731bb.mjs';
import '../../utils/index.mjs';
import '../../../_chunks/dep-6c281967.mjs';
import '../../../_chunks/dep-ea5fbe21.mjs';
import '../../../_chunks/dep-26d65784.mjs';
import '../../../_chunks/dep-a4f55e3b.mjs';
import '../../../_chunks/dep-a8b40385.mjs';
import 'lodash-es';
import '../../../_chunks/dep-557b68dd.mjs';
import './base-props.mjs';
import '../../../_chunks/dep-9ce910b6.mjs';
import '../../../_chunks/dep-05bddb06.mjs';
import '../../../_chunks/dep-57027bda.mjs';
import '../../../config-provider/hooks/useConfig.mjs';
import '../../../config-provider/utils/context.mjs';
import '../../../_chunks/dep-57949dcd.mjs';
import '../../../_chunks/dep-cafb5eac.mjs';
import '../../../_chunks/dep-f1da7269.mjs';
import '../../../_chunks/dep-192d0c88.mjs';
import '../../../_chunks/dep-652c99fe.mjs';
import '../../../_chunks/dep-5b369203.mjs';
import '../../../_chunks/dep-920eff5b.mjs';

var AlphaSlider = defineComponent({
  name: "AlphaSlider",
  inheritAttrs: false,
  props: {
    color: {
      type: Object
    },
    disabled: {
      type: Boolean,
      "default": false
    },
    onChange: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    }
  },
  setup: function setup(props) {
    var baseClassName = useBaseClassName();
    var handleChange = function handleChange(v, isDragEnd) {
      props.onChange(v / 100, isDragEnd);
    };
    var railStyle = computed(function () {
      return {
        background: "linear-gradient(to right, rgba(0, 0, 0, 0), ".concat(props.color.rgb, ")")
      };
    });
    return function () {
      return createVNode(ColorSlider, {
        "class": ["".concat(baseClassName.value, "__alpha"), "".concat(baseClassName.value, "--bg-alpha")],
        "color": props.color,
        "value": props.color.alpha * 100,
        "onChange": handleChange,
        "rail-style": railStyle.value,
        "max-value": 100,
        "disabled": props.disabled,
        "type": "alpha"
      }, null);
    };
  }
});

export { AlphaSlider as default };
