// tdesign-vue-next/cjs/empty/empty.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-449273df.js');
var index$2 = require('../_chunks/dep-c0cb4d13.js');
var index$1 = require('../_chunks/dep-9ada3ec1.js');
var index = require('../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-497fc9a5.js');
require('@babel/runtime/helpers/defineProperty');
var empty_props = require('./props.js');
var image_index = require('../image/index.js');
var empty_components_MaintenanceSvg = require('./components/MaintenanceSvg.js');
var empty_components_NetworkErrorSvg = require('./components/NetworkErrorSvg.js');
var empty_components_EmptySvg = require('./components/EmptySvg.js');
var empty_components_FailSvg = require('./components/FailSvg.js');
var empty_components_SuccessSvg = require('./components/SuccessSvg.js');
var isString = require('../_chunks/dep-be760bc2.js');
var configProvider_utils_context = require('../_chunks/dep-31abb282.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-f9b59444.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-80827572.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-cfdee5e7.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../image/image.js');
require('tdesign-icons-vue-next');
require('../_chunks/dep-12e6ec24.js');
require('../_chunks/dep-e04a8c18.js');
require('../_chunks/dep-2c1cb23b.js');
require('../_chunks/dep-b4528c21.js');
require('../image/props.js');
require('../space/index.js');
require('../space/space.js');
require('../space/props.js');
require('../_chunks/dep-cec24bbf.js');
require('../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-6130f266.js');
require('../_chunks/dep-3ffe18b0.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-e41b3434.js');
require('../_chunks/dep-d20c4e5e.js');
require('../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('../_chunks/dep-abc005aa.js');
require('../_chunks/dep-3a800699.js');
require('../_chunks/dep-e25ae470.js');
require('../_chunks/dep-648bfb84.js');
require('../_chunks/dep-4e2907e0.js');

var _Empty = Vue.defineComponent({
  name: "TEmpty",
  components: {
    TImage: image_index.Image
  },
  props: empty_props["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var _toRefs = Vue.toRefs(props2),
      size = _toRefs.size,
      propsImage = _toRefs.image,
      propsDescription = _toRefs.description,
      propsTitle = _toRefs.title,
      type = _toRefs.type;
    var _useConfig = configProvider_hooks_useConfig.useConfig("empty"),
      globalConfig = _useConfig.globalConfig;
    var classPrefix = index.usePrefixClass("empty");
    var showAction = Vue.computed(function () {
      return props2.action || slots.action;
    });
    var _useCommonClassName = index$1.useCommonClassName(),
      SIZE = _useCommonClassName.SIZE;
    var renderTNodeJSX = index$2.useTNodeJSX();
    var defaultMaps = Vue.computed(function () {
      return {
        maintenance: {
          image: globalConfig.value.image.maintenance || empty_components_MaintenanceSvg["default"],
          title: globalConfig.value.titleText.maintenance
        },
        success: {
          image: globalConfig.value.image.success || empty_components_SuccessSvg["default"],
          title: globalConfig.value.titleText.success
        },
        fail: {
          image: globalConfig.value.image.fail || empty_components_FailSvg["default"],
          title: globalConfig.value.titleText.fail
        },
        "network-error": {
          image: globalConfig.value.image.networkError || empty_components_NetworkErrorSvg["default"],
          title: globalConfig.value.titleText.networkError
        },
        empty: {
          image: globalConfig.value.image.empty || empty_components_EmptySvg["default"],
          title: globalConfig.value.titleText.empty
        }
      };
    });
    var emptyClasses = Vue.computed(function () {
      return [classPrefix.value, SIZE.value[size.value]];
    });
    var titleClasses = ["".concat(classPrefix.value, "__title")];
    var imageClasses = ["".concat(classPrefix.value, "__image")];
    var descriptionClasses = ["".concat(classPrefix.value, "__description")];
    var actionClass = ["".concat(classPrefix.value, "__action")];
    var typeImageProps = Vue.computed(function () {
      var _defaultMaps$value$ty;
      return (_defaultMaps$value$ty = defaultMaps.value[type.value]) !== null && _defaultMaps$value$ty !== void 0 ? _defaultMaps$value$ty : null;
    });
    var showImage = Vue.computed(function () {
      var _slots$image, _typeImageProps$value;
      return propsImage.value || (slots === null || slots === void 0 || (_slots$image = slots.image) === null || _slots$image === void 0 ? void 0 : _slots$image.call(slots)) || ((_typeImageProps$value = typeImageProps.value) === null || _typeImageProps$value === void 0 ? void 0 : _typeImageProps$value.image);
    });
    var showTitle = Vue.computed(function () {
      var _slots$title, _typeImageProps$value2;
      return propsTitle.value || (slots === null || slots === void 0 || (_slots$title = slots.title) === null || _slots$title === void 0 ? void 0 : _slots$title.call(slots)) || ((_typeImageProps$value2 = typeImageProps.value) === null || _typeImageProps$value2 === void 0 ? void 0 : _typeImageProps$value2.title);
    });
    var showDescription = Vue.computed(function () {
      var _slots$description;
      return propsDescription.value || (slots === null || slots === void 0 || (_slots$description = slots.description) === null || _slots$description === void 0 ? void 0 : _slots$description.call(slots));
    });
    var renderTitle = function renderTitle() {
      if (!showTitle.value) {
        return null;
      }
      return Vue.createVNode("div", {
        "class": titleClasses
      }, [showTitle.value]);
    };
    var renderDescription = function renderDescription() {
      if (!showDescription.value) {
        return null;
      }
      return Vue.createVNode("div", {
        "class": descriptionClasses
      }, [showDescription.value]);
    };
    var getImageIns = function getImageIns() {
      var data = showImage.value;
      var result = null;
      if (isString.isString(data)) {
        result = Vue.createVNode(image_index.Image, {
          "src": data
        }, null);
      } else if (data && Reflect.has(data, "setup")) {
        result = Vue.h(data);
      } else if (configProvider_utils_context.isPlainObject(data)) {
        result = Vue.createVNode(image_index.Image, data, null);
      }
      return result;
    };
    return function () {
      return Vue.createVNode("div", {
        "class": emptyClasses.value
      }, [showImage.value ? Vue.createVNode("div", {
        "class": imageClasses,
        "style": props2.imageStyle
      }, [slots !== null && slots !== void 0 && slots.image ? renderTNodeJSX("image") : getImageIns()]) : null, renderTitle(), renderDescription(), showAction.value ? Vue.createVNode("div", {
        "class": actionClass
      }, [renderTNodeJSX("action")]) : null]);
    };
  }
});

exports["default"] = _Empty;
