// tdesign-vue-next/cjs/descriptions/descriptions.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var index$2 = require('../_chunks/dep-cec24bbf.js');
var index$3 = require('../_chunks/dep-c0cb4d13.js');
var index$1 = require('../_chunks/dep-9ada3ec1.js');
var index = require('../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-497fc9a5.js');
require('../_chunks/dep-449273df.js');
require('@babel/runtime/helpers/toConsumableArray');
var descriptions_props = require('./props.js');
var descriptions_constants_index = require('./constants/index.js');
var descriptions_descriptionsRow = require('./descriptions-row.js');
var descriptions_utils_index = require('../_chunks/dep-05153078.js');
var isArray = require('../_chunks/dep-959f4847.js');
var isNil = require('../_chunks/dep-59ce9497.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-be760bc2.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-80827572.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-cfdee5e7.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('../_chunks/dep-f9b59444.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var _Descriptions = Vue.defineComponent({
  name: "TDescriptions",
  props: descriptions_props["default"],
  setup: function setup(props2) {
    var COMPONENT_NAME = index.usePrefixClass("descriptions");
    var _useCommonClassName = index$1.useCommonClassName(),
      SIZE = _useCommonClassName.SIZE;
    var getChildByName = index$2.useChildComponentSlots();
    var renderTNodeJSX = index$3.useTNodeJSX();
    var itemsType = Vue.ref(descriptions_utils_index.ItemsType.props);
    var getRows = function getRows() {
      var column = props2.column,
        layout = props2.layout;
      var items = [];
      if (isArray.isArray(props2.items)) {
        items = props2.items.map(function (item) {
          return {
            label: descriptions_utils_index.renderCustomNode(item.label),
            content: descriptions_utils_index.renderCustomNode(item.content),
            span: item.span || 1
          };
        });
        itemsType.value = descriptions_utils_index.ItemsType.props;
      } else {
        var slots = getChildByName("TDescriptionsItem");
        if (slots.length !== 0) {
          items = slots;
          itemsType.value = descriptions_utils_index.ItemsType.slots;
        }
      }
      if (layout === "vertical") {
        return [items];
      }
      var temp = [];
      var reset = column;
      var res = [];
      items.forEach(function (item, index) {
        var span = 1;
        if (descriptions_utils_index.itemTypeIsProps(itemsType.value)) {
          span = isNil.isNil(item.span) ? span : item.span;
          span = span > column ? column : span;
        } else {
          var _item$props;
          item.props = item.props || {};
          span = isNil.isNil((_item$props = item.props) === null || _item$props === void 0 ? void 0 : _item$props.span) ? span : item.props.span;
          span = span > column ? column : span;
          item.props.span = span;
        }
        if (reset >= span) {
          temp.push(item);
          reset -= span;
        } else {
          res.push(temp);
          temp = [item];
          reset = column - span;
        }
        if (index === items.length - 1) {
          if (descriptions_utils_index.itemTypeIsProps(itemsType.value)) {
            item.span += reset;
          } else {
            item.props.span += reset;
          }
          res.push(temp);
        }
      });
      return res;
    };
    Vue.provide(descriptions_constants_index.descriptionsKey, props2);
    var renderBody = function renderBody() {
      var tableClass = ["".concat(COMPONENT_NAME.value, "__body"), SIZE.value[props2.size], _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "__body--fixed"), props2.tableLayout === "fixed"), _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "__body--border"), props2.bordered)];
      return Vue.createVNode("table", {
        "class": tableClass
      }, [Vue.createVNode("tbody", null, [getRows().map(function (row) {
        return Vue.createVNode(descriptions_descriptionsRow["default"], {
          "item-type": itemsType.value,
          "row": row
        }, null);
      })])]);
    };
    var renderHeader = function renderHeader() {
      var title = renderTNodeJSX("title");
      return title ? Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__header")
      }, [title]) : "";
    };
    return function () {
      return Vue.createVNode("div", {
        "class": COMPONENT_NAME.value
      }, [renderHeader(), renderBody()]);
    };
  }
});

exports["default"] = _Descriptions;
