// tdesign-vue-next/es/layout/aside.mjs
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, inject, onMounted, onUnmounted, createVNode } from 'vue';
import props from './aside-props.mjs';
import { u as usePrefixClass, a as useTNodeJSX } from '../_chunks/dep-9ce910b6.mjs';
import 'lodash-es';
import '../_chunks/dep-557b68dd.mjs';
import '../_chunks/dep-26d65784.mjs';
import '../_chunks/dep-f038e2b4.mjs';
import '../_chunks/dep-05bddb06.mjs';
import '../_chunks/dep-57027bda.mjs';
import '../config-provider/hooks/useConfig.mjs';
import '../config-provider/utils/context.mjs';
import '../_chunks/dep-57949dcd.mjs';
import '../_chunks/dep-cafb5eac.mjs';
import '../_chunks/dep-f1da7269.mjs';
import '../_chunks/dep-ea5fbe21.mjs';
import '../_chunks/dep-192d0c88.mjs';
import '../_chunks/dep-652c99fe.mjs';
import '../_chunks/dep-5b369203.mjs';
import '../_chunks/dep-a4f55e3b.mjs';
import '../_chunks/dep-98dac781.mjs';
import '../_chunks/dep-920eff5b.mjs';

var _Aside = defineComponent({
  name: "TAside",
  props: props,
  setup: function setup(props2) {
    var _inject = inject("layout", /* @__PURE__ */Object.create(null)),
      hasSide = _inject.hasSide;
    var COMPONENT_NAME = usePrefixClass("layout__sider");
    var renderTNodeJSX = useTNodeJSX();
    if (!hasSide) return;
    onMounted(function () {
      hasSide.value = true;
    });
    onUnmounted(function () {
      hasSide.value = false;
    });
    return function () {
      var styles = props2.width ? {
        width: props2.width
      } : {};
      return createVNode("aside", {
        "class": COMPONENT_NAME.value,
        "style": styles
      }, [renderTNodeJSX("default")]);
    };
  }
});

export { _Aside as default };
