// tdesign-vue-next/cjs/pagination/pagination-mini.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var pagination_paginationMiniProps = require('./pagination-mini-props.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-449273df.js');
var index = require('../_chunks/dep-95cb1381.js');
var index$1 = require('../_chunks/dep-88ae5d8a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-497fc9a5.js');
var button_index = require('../button/index.js');
var tooltip_index = require('../tooltip/index.js');
var isObject = require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-f9b59444.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-80827572.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../button/button.js');
require('../loading/index.js');
require('../_chunks/dep-203a5570.js');
require('../loading/plugin.js');
require('../_chunks/dep-3231adce.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-e3eb5659.js');
require('../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-6130f266.js');
require('../_chunks/dep-3ffe18b0.js');
require('../_chunks/dep-e41b3434.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-be760bc2.js');
require('../_chunks/dep-c0cb4d13.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-cfdee5e7.js');
require('../_chunks/dep-9ada3ec1.js');
require('../loading/props.js');
require('../_chunks/dep-e46285c0.js');
require('../_chunks/dep-54801354.js');
require('../_chunks/dep-bd59ca66.js');
require('../_chunks/dep-f7508ffa.js');
require('../_chunks/dep-470f211b.js');
require('../_chunks/dep-9b549c85.js');
require('../_chunks/dep-e25ae470.js');
require('../_chunks/dep-bed84553.js');
require('../_chunks/dep-a4b963ab.js');
require('../_chunks/dep-abc005aa.js');
require('../button/props.js');
require('../_chunks/dep-a2e46896.js');
require('../_chunks/dep-6512c9d2.js');
require('../_chunks/dep-3cd26986.js');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../popup/props.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../_chunks/dep-d0227fbe.js');
require('../popup/container.js');
require('../_chunks/dep-ba2511b0.js');
require('../_chunks/dep-4e2a4791.js');
require('../_chunks/dep-f3377589.js');
require('../_chunks/dep-71d9daf0.js');
require('../tooltip/utils/index.js');
require('../_chunks/dep-3a800699.js');
require('../_chunks/dep-648bfb84.js');
require('../_chunks/dep-4e2907e0.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var _PaginationMini = Vue.defineComponent({
  name: "TPaginationMini",
  props: pagination_paginationMiniProps["default"],
  setup: function setup(props2) {
    var COMPONENT_NAME = index.usePrefixClass("pagination-mini");
    var _useGlobalIcon = index$1.useGlobalIcon({
        ChevronLeftIcon: tdesignIconsVueNext.ChevronLeftIcon,
        RoundIcon: tdesignIconsVueNext.RoundIcon,
        ChevronRightIcon: tdesignIconsVueNext.ChevronRightIcon,
        ChevronUpIcon: tdesignIconsVueNext.ChevronUpIcon,
        ChevronDownIcon: tdesignIconsVueNext.ChevronDownIcon
      }),
      ChevronLeftIcon = _useGlobalIcon.ChevronLeftIcon,
      RoundIcon = _useGlobalIcon.RoundIcon,
      ChevronRightIcon = _useGlobalIcon.ChevronRightIcon,
      ChevronUpIcon = _useGlobalIcon.ChevronUpIcon,
      ChevronDownIcon = _useGlobalIcon.ChevronDownIcon;
    var titleConfig = Vue.computed(function () {
      if (isObject.isObject(props2.tips)) return props2.tips;
      if (props2.tips === true) return {
        prev: "\u4E0A\u4E00\u9875",
        current: "\u5F53\u524D",
        next: "\u4E0B\u4E00\u9875"
      };
      return {};
    });
    var disabledConfig = Vue.computed(function () {
      if (isObject.isObject(props2.disabled)) return props2.disabled;
      if (props2.disabled === true) return {
        prev: true,
        current: true,
        next: true
      };
      return {
        prev: false,
        current: false,
        next: false
      };
    });
    var renderWithTooltip = function renderWithTooltip(content, node) {
      if (!content) return node;
      return Vue.createVNode(tooltip_index.Tooltip, {
        "content": content,
        "showArrow": false
      }, _isSlot(node) ? node : {
        "default": function _default() {
          return [node];
        }
      });
    };
    return function () {
      var jumperClass = [COMPONENT_NAME.value, _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--outline"), props2.variant === "outline")];
      return Vue.createVNode("div", {
        "class": jumperClass
      }, [renderWithTooltip(titleConfig.value.prev, Vue.createVNode(button_index.Button, {
        "variant": props2.variant,
        "size": props2.size,
        "shape": "square",
        "onClick": function onClick(e) {
          var _props2$onChange;
          return (_props2$onChange = props2.onChange) === null || _props2$onChange === void 0 ? void 0 : _props2$onChange.call(props2, {
            e: e,
            trigger: "prev"
          });
        },
        "icon": props2.layout === "horizontal" ? function () {
          return Vue.createVNode(ChevronLeftIcon, null, null);
        } : function () {
          return Vue.createVNode(ChevronUpIcon, null, null);
        },
        "class": "".concat(COMPONENT_NAME.value, "__prev"),
        "disabled": disabledConfig.value.prev
      }, null)), props2.showCurrent && renderWithTooltip(titleConfig.value.current, Vue.createVNode(button_index.Button, {
        "variant": props2.variant,
        "size": props2.size,
        "shape": "square",
        "onClick": function onClick(e) {
          var _props2$onChange2;
          return (_props2$onChange2 = props2.onChange) === null || _props2$onChange2 === void 0 ? void 0 : _props2$onChange2.call(props2, {
            e: e,
            trigger: "current"
          });
        },
        "icon": function icon() {
          return Vue.createVNode(RoundIcon, null, null);
        },
        "class": "".concat(COMPONENT_NAME.value, "__current"),
        "disabled": disabledConfig.value.current
      }, null)), renderWithTooltip(titleConfig.value.next, Vue.createVNode(button_index.Button, {
        "variant": props2.variant,
        "size": props2.size,
        "shape": "square",
        "onClick": function onClick(e) {
          var _props2$onChange3;
          return (_props2$onChange3 = props2.onChange) === null || _props2$onChange3 === void 0 ? void 0 : _props2$onChange3.call(props2, {
            e: e,
            trigger: "next"
          });
        },
        "icon": props2.layout === "horizontal" ? function () {
          return Vue.createVNode(ChevronRightIcon, null, null);
        } : function () {
          return Vue.createVNode(ChevronDownIcon, null, null);
        },
        "class": "".concat(COMPONENT_NAME.value, "__next"),
        "disabled": disabledConfig.value.next
      }, null))]);
    };
  }
});

exports["default"] = _PaginationMini;
