// tdesign-vue-next/cjs/table/components/tbody.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var table_components_tr = require('./tr.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-449273df.js');
var index = require('../../_chunks/dep-c0cb4d13.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-497fc9a5.js');
var table_hooks_useClassName = require('../hooks/useClassName.js');
var table_baseTableProps = require('../base-table-props.js');
var table_hooks_useRowspanAndColspan = require('../hooks/useRowspanAndColspan.js');
var pick = require('../../_chunks/dep-008d4a55.js');
var get = require('../../_chunks/dep-9b549c85.js');
var camelCase = require('../../_chunks/dep-2db31625.js');
require('../utils/index.js');
require('../../_chunks/dep-80827572.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../hooks/useFixed.js');
require('../../_chunks/dep-b4528c21.js');
require('../../_chunks/dep-6530011a.js');
require('../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6130f266.js');
require('../../_chunks/dep-3ffe18b0.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-e41b3434.js');
require('../../_chunks/dep-51846987.js');
require('../../_chunks/dep-be760bc2.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-6e64ef0e.js');
require('../../_chunks/dep-6e7a2100.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-bd59ca66.js');
require('../../_chunks/dep-618d56e7.js');
require('../../_chunks/dep-0a17438d.js');
require('../../_chunks/dep-5090b515.js');
require('../../_chunks/dep-4e2907e0.js');
require('../../_chunks/dep-8f2069e9.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-f7508ffa.js');
require('../../_chunks/dep-648bfb84.js');
require('../../_chunks/dep-f3377589.js');
require('../../_chunks/dep-f9b59444.js');
require('../../_chunks/dep-e25ae470.js');
require('../../_chunks/dep-0e1751ee.js');
require('../../_chunks/dep-bed84553.js');
require('./ellipsis.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../popup/props.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../_chunks/dep-9ada3ec1.js');
require('../../_chunks/dep-95cb1381.js');
require('../../_chunks/dep-e3eb5659.js');
require('../../_chunks/dep-d0227fbe.js');
require('../../_chunks/dep-cfdee5e7.js');
require('../../_chunks/dep-1c6213b5.js');
require('../../popup/container.js');
require('../../_chunks/dep-ba2511b0.js');
require('../../_chunks/dep-4e2a4791.js');
require('../../_chunks/dep-71d9daf0.js');
require('../../_chunks/dep-9dfd682d.js');
require('../../_chunks/dep-39313d05.js');
require('../../_chunks/dep-f9bc7c82.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-abc005aa.js');
require('../../tooltip/utils/index.js');
require('../../_chunks/dep-3a800699.js');
require('../hooks/useLazyLoad.js');
require('../../_chunks/dep-12e6ec24.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var ROW_AND_TD_LISTENERS = [].concat(_toConsumableArray__default["default"](table_components_tr.ROW_LISTENERS), ["cell-click"]);
var extendTableProps = ["bordered", "rowKey", "rowClassName", "rowAttributes", "loading", "empty", "fixedRows", "firstFullRow", "lastFullRow", "rowspanAndColspan", "scroll", "cellEmptyContent", "pagination", "attach", "onCellClick", "onPageChange", "onRowClick", "onRowDblclick", "onRowMouseover", "onRowMousedown", "onRowMouseenter", "onRowMouseleave", "onRowMouseup", "onScroll", "onScrollX", "onScrollY"];
var TBody = Vue.defineComponent({
  name: "TBody",
  props: _objectSpread({
    classPrefix: String,
    data: Array,
    columns: Array,
    ellipsisOverlayClassName: String,
    rowAndColFixedPosition: Map,
    showColumnShadow: Object,
    tableElm: {},
    tableWidth: Number,
    isWidthOverflow: Boolean,
    virtualConfig: Object,
    tableContentElm: {},
    handleRowMounted: Function,
    renderExpandedRow: Function,
    firstFullRow: [String, Function],
    lastFullRow: [String, Function],
    activeRow: [Array],
    hoverRow: [String, Number]
  }, pick.pick(table_baseTableProps["default"], extendTableProps)),
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var renderTNode = index.useTNodeJSX();
    var _toRefs = Vue.toRefs(props),
      data = _toRefs.data,
      columns = _toRefs.columns,
      rowKey = _toRefs.rowKey,
      rowspanAndColspan = _toRefs.rowspanAndColspan;
    var _useConfig = configProvider_hooks_useConfig.useConfig("table", props.locale),
      t = _useConfig.t,
      globalConfig = _useConfig.globalConfig;
    var _useClassName = table_hooks_useClassName["default"](),
      tableFullRowClasses = _useClassName.tableFullRowClasses,
      tableBaseClass = _useClassName.tableBaseClass;
    var _useRowspanAndColspan = table_hooks_useRowspanAndColspan["default"](data, columns, rowKey, rowspanAndColspan),
      skipSpansMap = _useRowspanAndColspan.skipSpansMap;
    var tbodyClasses = Vue.computed(function () {
      return [tableBaseClass.body];
    });
    return function () {
      var _data$value, _data$value2, _props$virtualConfig, _props$virtualConfig2;
      var renderEmpty = function renderEmpty(columns2) {
        var tableWidth = props.bordered ? props.tableWidth - 2 : props.tableWidth;
        return Vue.createVNode("tr", {
          "class": [tableBaseClass.emptyRow, _defineProperty__default["default"]({}, tableFullRowClasses.base, props.isWidthOverflow)]
        }, [Vue.createVNode("td", {
          "colspan": columns2.length
        }, [Vue.createVNode("div", {
          "class": [tableBaseClass.empty, _defineProperty__default["default"]({}, tableFullRowClasses.innerFullRow, props.isWidthOverflow)],
          "style": props.isWidthOverflow ? {
            width: "".concat(tableWidth, "px")
          } : {}
        }, [renderTNode("empty") || t(globalConfig.value.empty)])])]);
      };
      var getFullRow = function getFullRow(columnLength2, type) {
        var tType = camelCase.camelCase(type);
        var fullRowNode = renderTNode(tType);
        if (["", null, void 0, false].includes(fullRowNode)) return null;
        var isFixedToLeft = props.isWidthOverflow && columns.value.find(function (col) {
          return col.fixed === "left";
        });
        var classes = [tableFullRowClasses.base, tableFullRowClasses[tType]];
        var tableWidth = props.bordered ? props.tableWidth - 2 : props.tableWidth;
        return Vue.createVNode("tr", {
          "class": classes,
          "key": "key-full-row-".concat(type)
        }, [Vue.createVNode("td", {
          "colspan": columnLength2
        }, [Vue.createVNode("div", {
          "class": _defineProperty__default["default"]({}, tableFullRowClasses.innerFullRow, isFixedToLeft),
          "style": isFixedToLeft ? {
            width: "".concat(tableWidth, "px")
          } : {}
        }, [Vue.createVNode("div", {
          "class": tableFullRowClasses.innerFullElement
        }, [fullRowNode])])])]);
      };
      var columnLength = columns.value.length;
      var dataLength = (_data$value = data.value) === null || _data$value === void 0 ? void 0 : _data$value.length;
      var trNodeList = [];
      var properties = ["classPrefix", "ellipsisOverlayClassName", "rowAndColFixedPosition", "scroll", "tableElm", "tableContentElm", "pagination", "attach"];
      var renderData = props.virtualConfig.isVirtualScroll.value ? props.virtualConfig.visibleData.value : data.value;
      renderData === null || renderData === void 0 || renderData.forEach(function (row, rowIndex) {
        var _props$activeRow;
        var rowKey2 = props.rowKey || "id";
        var rowValue = get.get(row, rowKey2);
        var trProps = _objectSpread(_objectSpread({}, pick.pick(props, table_components_tr.TABLE_PROPS)), {}, {
          rowKey: rowKey2,
          row: row,
          columns: columns.value,
          rowIndex: row.VIRTUAL_SCROLL_INDEX || rowIndex,
          dataLength: dataLength,
          skipSpansMap: skipSpansMap.value,
          virtualConfig: props.virtualConfig,
          active: (_props$activeRow = props.activeRow) === null || _props$activeRow === void 0 ? void 0 : _props$activeRow.includes(rowValue),
          isHover: props.hoverRow === rowValue
        }, pick.pick(props, properties));
        if (props.onCellClick) {
          trProps.onCellClick = props.onCellClick;
        }
        var trNode = Vue.createVNode(table_components_tr["default"], Vue.mergeProps({
          "key": get.get(row, rowKey2 || "id") || rowIndex
        }, trProps, {
          "onRowMounted": props.handleRowMounted
        }), slots);
        trNodeList.push(trNode);
        if (props.renderExpandedRow) {
          var p = {
            row: row,
            index: rowIndex,
            columns: columns.value,
            tableWidth: props.tableWidth,
            isWidthOverflow: props.isWidthOverflow
          };
          var expandedContent = props.renderExpandedRow(p);
          expandedContent && trNodeList.push(expandedContent);
        }
      });
      var list = [getFullRow(columnLength, "first-full-row")].concat(trNodeList, [getFullRow(columnLength, "last-full-row")]);
      var isEmpty = !((_data$value2 = data.value) !== null && _data$value2 !== void 0 && _data$value2.length) && !props.loading && !props.firstFullRow && !props.lastFullRow;
      var translate = "translateY(".concat((_props$virtualConfig = props.virtualConfig) === null || _props$virtualConfig === void 0 ? void 0 : _props$virtualConfig.translateY.value, "px)");
      var posStyle = (_props$virtualConfig2 = props.virtualConfig) !== null && _props$virtualConfig2 !== void 0 && _props$virtualConfig2.isVirtualScroll.value ? {
        transform: translate,
        "-ms-transform": translate,
        "-moz-transform": translate,
        "-webkit-transform": translate
      } : void 0;
      return Vue.createVNode("tbody", {
        "class": tbodyClasses.value,
        "style": _objectSpread({}, posStyle)
      }, [isEmpty ? renderEmpty(columns.value) : list]);
    };
  }
});

exports.ROW_AND_TD_LISTENERS = ROW_AND_TD_LISTENERS;
exports["default"] = TBody;
exports.extendTableProps = extendTableProps;
