// tdesign-vue-next/cjs/table/components/tfoot.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-449273df.js');
var index = require('../../_chunks/dep-c0cb4d13.js');
require('../../config-provider/hooks/useConfig.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-497fc9a5.js');
var table_utils_index = require('../utils/index.js');
var table_hooks_useFixed = require('../hooks/useFixed.js');
var table_hooks_useRowspanAndColspan = require('../hooks/useRowspanAndColspan.js');
var table_hooks_useClassName = require('../hooks/useClassName.js');
var isFunction = require('../../_chunks/dep-80827572.js');
var isString = require('../../_chunks/dep-be760bc2.js');
var get = require('../../_chunks/dep-9b549c85.js');
require('../../_chunks/dep-f9b59444.js');
require('../../_chunks/dep-51846987.js');
require('../../_chunks/dep-9dfd682d.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-2db31625.js');
require('../../_chunks/dep-0a17438d.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-5090b515.js');
require('../../_chunks/dep-6e64ef0e.js');
require('../../_chunks/dep-39313d05.js');
require('../../_chunks/dep-f9bc7c82.js');
require('../../_chunks/dep-1c6213b5.js');
require('../../_chunks/dep-cfdee5e7.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-b4528c21.js');
require('../../_chunks/dep-6530011a.js');
require('../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6130f266.js');
require('../../_chunks/dep-3ffe18b0.js');
require('../../_chunks/dep-e41b3434.js');
require('../../_chunks/dep-bd59ca66.js');
require('../../_chunks/dep-618d56e7.js');
require('../../_chunks/dep-4e2907e0.js');
require('../../_chunks/dep-8f2069e9.js');
require('../../_chunks/dep-f7508ffa.js');
require('../../_chunks/dep-648bfb84.js');
require('../../_chunks/dep-008d4a55.js');
require('../../_chunks/dep-e25ae470.js');
require('../../_chunks/dep-0e1751ee.js');
require('../../_chunks/dep-bed84553.js');
require('../../_chunks/dep-f3377589.js');
require('../../_chunks/dep-95cb1381.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TFoot = Vue.defineComponent({
  name: "TFoot",
  props: {
    rowKey: String,
    isFixedHeader: Boolean,
    rowAndColFixedPosition: Map,
    footData: Array,
    columns: Array,
    rowAttributes: [Array, Object, Function],
    rowClassName: [Array, String, Object, Function],
    thWidthList: [Object],
    footerSummary: [String, Function],
    rowspanAndColspanInFooter: Function,
    virtualScroll: Boolean
  },
  setup: function setup(props, context) {
    var renderTNode = index.useTNodeJSX();
    var classnames = table_hooks_useClassName["default"]();
    var _toRefs = Vue.toRefs(props),
      footData = _toRefs.footData,
      columns = _toRefs.columns,
      rowKey = _toRefs.rowKey,
      rowspanAndColspanInFooter = _toRefs.rowspanAndColspanInFooter;
    var _useRowspanAndColspan = table_hooks_useRowspanAndColspan["default"](footData, columns, rowKey, rowspanAndColspanInFooter),
      skipSpansMap = _useRowspanAndColspan.skipSpansMap;
    var renderTFootCell = function renderTFootCell(p) {
      var col = p.col,
        row = p.row;
      if (isFunction.isFunction(col.foot)) {
        return col.foot(Vue.h, p);
      }
      if (isString.isString(col.foot) && context.slots[col.foot]) {
        return context.slots[col.foot](p) || col.foot;
      }
      return col.foot || get.get(row, col.colKey);
    };
    return function () {
      var _props$footData;
      if (!columns.value) return null;
      var theadClasses = [classnames.tableFooterClasses.footer, _defineProperty__default["default"]({}, classnames.tableFooterClasses.fixed, props.isFixedHeader)];
      var footerDomList = (_props$footData = props.footData) === null || _props$footData === void 0 ? void 0 : _props$footData.map(function (row, rowIndex) {
        var trAttributes = table_utils_index.formatRowAttributes(props.rowAttributes, {
          row: row,
          rowIndex: rowIndex,
          type: "foot"
        });
        var customClasses = table_utils_index.formatRowClassNames(props.rowClassName, {
          row: row,
          rowIndex: rowIndex,
          type: "foot"
        }, rowKey.value || "id");
        return Vue.createVNode("tr", Vue.mergeProps(trAttributes, {
          "key": rowIndex,
          "class": customClasses
        }), [columns.value.map(function (col, colIndex) {
          var _props$thWidthList;
          var cellSpans = {};
          var spanState = null;
          if (skipSpansMap.value.size) {
            var _spanState, _spanState2;
            var cellKey = table_hooks_useRowspanAndColspan.getCellKey(row, rowKey.value, col.colKey, colIndex, rowIndex);
            spanState = skipSpansMap.value.get(cellKey) || {};
            ((_spanState = spanState) === null || _spanState === void 0 ? void 0 : _spanState.rowspan) > 1 && (cellSpans.rowspan = spanState.rowspan);
            ((_spanState2 = spanState) === null || _spanState2 === void 0 ? void 0 : _spanState2.colspan) > 1 && (cellSpans.colspan = spanState.colspan);
            if (spanState.skipped) return null;
          }
          var tdStyles = table_hooks_useFixed.getColumnFixedStyles(col, colIndex, props.rowAndColFixedPosition, classnames.tableColFixedClasses);
          var style = _objectSpread({}, tdStyles.style);
          if ((_props$thWidthList = props.thWidthList) !== null && _props$thWidthList !== void 0 && _props$thWidthList[col.colKey]) {
            style.width = "".concat(props.thWidthList[col.colKey], "px");
          }
          return Vue.createVNode("td", Vue.mergeProps(_objectSpread({
            key: col.colKey
          }, cellSpans), {
            "class": tdStyles.classes,
            "style": style
          }), [renderTFootCell({
            row: row,
            rowIndex: rowIndex,
            col: col,
            colIndex: colIndex
          })]);
        })]);
      });
      var footerSummary = renderTNode("footerSummary");
      if (!footerSummary && (!props.footData || !props.footData.length)) return null;
      return Vue.createVNode("tfoot", {
        "class": theadClasses,
        "style": {
          visibility: props.virtualScroll ? "hidden" : "visible"
        }
      }, [footerSummary && Vue.createVNode("tr", {
        "class": classnames.tableFullRowClasses.base
      }, [Vue.createVNode("td", {
        "colspan": columns.value.length
      }, [Vue.createVNode("div", {
        "class": classnames.tableFullRowClasses.innerFullElement
      }, [footerSummary])])]), footerDomList]);
    };
  }
});

exports["default"] = TFoot;
