// tdesign-vue-next/es/tree/hooks/useDragHandle.mjs
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { DragPosition } from './useDraggable.mjs';
import { emitEvent } from '../utils/index.mjs';
import 'lodash-es';
import '../utils/adapt.mjs';
import 'vue';
import '../../_chunks/dep-9ce910b6.mjs';
import '../../_chunks/dep-557b68dd.mjs';
import '../../_chunks/dep-26d65784.mjs';
import '../../_chunks/dep-f038e2b4.mjs';
import '../../_chunks/dep-05bddb06.mjs';
import '../../_chunks/dep-57027bda.mjs';
import '../../config-provider/hooks/useConfig.mjs';
import '../../config-provider/utils/context.mjs';
import '../../_chunks/dep-57949dcd.mjs';
import '../../_chunks/dep-cafb5eac.mjs';
import '../../_chunks/dep-f1da7269.mjs';
import '../../_chunks/dep-ea5fbe21.mjs';
import '../../_chunks/dep-192d0c88.mjs';
import '../../_chunks/dep-652c99fe.mjs';
import '../../_chunks/dep-5b369203.mjs';
import '../../_chunks/dep-a4f55e3b.mjs';
import '../../_chunks/dep-98dac781.mjs';
import '../../_chunks/dep-920eff5b.mjs';
import 'tdesign-icons-vue-next';
import '../../checkbox/index.mjs';
import '../../checkbox/checkbox.mjs';
import '../../checkbox/props.mjs';
import '../../checkbox/constants/index.mjs';
import '../../checkbox/hooks/useCheckboxLazyLoad.mjs';
import '../../checkbox/hooks/useKeyboardEvent.mjs';
import '../../checkbox/group.mjs';
import '../../checkbox/checkbox-group-props.mjs';
import './style/css.mjs';
import '../../loading/index.mjs';
import '../../loading/directive.mjs';
import '../../loading/plugin.mjs';
import '../../loading/loading.mjs';
import '../../loading/icon/gradient.mjs';
import '../../_chunks/dep-9b3b77bb.mjs';
import '../../_chunks/dep-4de66dd0.mjs';
import '../../loading/props.mjs';
import '../../_chunks/dep-aa9cf5ff.mjs';
import '../../_chunks/dep-9cac91c1.mjs';
import '../../_chunks/dep-73946420.mjs';

function useDragHandle(state) {
  var props = state.props,
    context = state.context,
    scope = state.scope,
    store = state.store;
  var dragNode = null;
  var handleDragStart = function handleDragStart(state2) {
    var dragEvent = state2.dragEvent,
      node = state2.node;
    dragNode = node;
    var ctx = {
      node: node.getModel(),
      e: dragEvent
    };
    emitEvent(props, context, "drag-start", ctx);
  };
  var handleDragEnd = function handleDragEnd(state2) {
    var dragEvent = state2.dragEvent,
      node = state2.node;
    dragNode = node;
    var ctx = {
      node: node.getModel(),
      e: dragEvent
    };
    emitEvent(props, context, "drag-end", ctx);
  };
  var handleDragOver = function handleDragOver(state2) {
    var dragEvent = state2.dragEvent,
      node = state2.node;
    var ctx = {
      node: node.getModel(),
      e: dragEvent
    };
    emitEvent(props, context, "drag-over", ctx);
  };
  var handleDragLeave = function handleDragLeave(state2) {
    var dragEvent = state2.dragEvent,
      node = state2.node;
    var ctx = {
      node: node.getModel(),
      e: dragEvent
    };
    emitEvent(props, context, "drag-leave", ctx);
  };
  var handleDrop = function handleDrop(state2) {
    var _props$allowDrop;
    var dragEvent = state2.dragEvent,
      node = state2.node,
      dropPosition = state2.dropPosition;
    if (node.value === dragNode.value || node.getParents().some(function (_node) {
      return _node.value === dragNode.value;
    })) return;
    var ctx = {
      dropNode: node.getModel(),
      dragNode: dragNode.getModel(),
      dropPosition: dropPosition,
      e: dragEvent
    };
    if (((_props$allowDrop = props.allowDrop) === null || _props$allowDrop === void 0 ? void 0 : _props$allowDrop.call(props, ctx)) === false) return;
    var nodes = store.getNodes();
    nodes.some(function (_node) {
      if (_node.value === node.value) {
        if (dropPosition === DragPosition.Inside) {
          dragNode.appendTo(store, _node);
        } else if (dropPosition === DragPosition.Before) {
          node.insertBefore(dragNode);
        } else {
          node.insertAfter(dragNode);
        }
        return true;
      }
      return false;
    });
    emitEvent(props, context, "drop", ctx);
  };
  var drag = {
    handleDragStart: handleDragStart,
    handleDragEnd: handleDragEnd,
    handleDragOver: handleDragOver,
    handleDragLeave: handleDragLeave,
    handleDrop: handleDrop
  };
  scope.drag = drag;
  return {
    drag: drag
  };
}

export { useDragHandle as default };
