// tdesign-vue-next/es/tree/hooks/useItemState.mjs
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import '../utils/adapt.mjs';
import { toRefs, ref } from 'vue';
import '../../_chunks/dep-9ce910b6.mjs';
import 'lodash-es';
import '../../_chunks/dep-557b68dd.mjs';
import '../../_chunks/dep-26d65784.mjs';
import '../../_chunks/dep-f038e2b4.mjs';
import '../../_chunks/dep-05bddb06.mjs';
import '../../_chunks/dep-57027bda.mjs';
import '../../config-provider/hooks/useConfig.mjs';
import '../../config-provider/utils/context.mjs';
import '../../_chunks/dep-57949dcd.mjs';
import '../../_chunks/dep-cafb5eac.mjs';
import '../../_chunks/dep-f1da7269.mjs';
import '../../_chunks/dep-ea5fbe21.mjs';
import '../../_chunks/dep-192d0c88.mjs';
import '../../_chunks/dep-652c99fe.mjs';
import '../../_chunks/dep-5b369203.mjs';
import '../../_chunks/dep-a4f55e3b.mjs';
import '../../_chunks/dep-98dac781.mjs';
import '../../_chunks/dep-920eff5b.mjs';
import 'tdesign-icons-vue-next';
import '../../checkbox/index.mjs';
import '../../checkbox/checkbox.mjs';
import '../../checkbox/props.mjs';
import '../../checkbox/constants/index.mjs';
import '../../checkbox/hooks/useCheckboxLazyLoad.mjs';
import '../../checkbox/hooks/useKeyboardEvent.mjs';
import '../../checkbox/group.mjs';
import '../../checkbox/checkbox-group-props.mjs';
import './style/css.mjs';
import '../../loading/index.mjs';
import '../../loading/directive.mjs';
import '../../loading/plugin.mjs';
import '../../loading/loading.mjs';
import '../../loading/icon/gradient.mjs';
import '../../_chunks/dep-9b3b77bb.mjs';
import '../../_chunks/dep-4de66dd0.mjs';
import '../../loading/props.mjs';
import '../../_chunks/dep-aa9cf5ff.mjs';
import '../../_chunks/dep-9cac91c1.mjs';
import '../../_chunks/dep-73946420.mjs';

function useItemState(props, context) {
  var refProps = toRefs(props);
  var treeItemRef = ref(null);
  var treeScope = props.treeScope;
  var store = treeScope.store;
  var node = store.privateMap.get(props.itemKey);
  var state = {
    stateId: props.stateId,
    props: props,
    context: context,
    treeScope: treeScope,
    refProps: refProps,
    node: node,
    treeItemRef: treeItemRef
  };
  return {
    state: state
  };
}

export { useItemState as default };
