// tdesign-vue-next/esm/color-picker/components/format/index.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, ref, watch, createVNode, isVNode } from 'vue';
import _defineProperty from '@babel/runtime/helpers/defineProperty';
import '../../../_chunks/dep-63d8d332.js';
import '@babel/runtime/helpers/classCallCheck';
import '@babel/runtime/helpers/createClass';
import { g as getColorFormatOptions } from '../../../_chunks/dep-7a0dc9bd.js';
import props from '../../props.js';
import { Select, Option } from '../../../select/index.js';
import FormatInputs from './inputs.js';
import { useBaseClassName } from '../../hooks/index.js';
import '@babel/runtime/helpers/slicedToArray';
import 'tinycolor2';
import '../../../_chunks/dep-92a02ffb.js';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '../../../select/select.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/regenerator';
import '../../../common-components/fake-arrow.js';
import '@babel/runtime/helpers/typeof';
import '../../../_chunks/dep-d9440b5f.js';
import '../../../_chunks/dep-a6042a25.js';
import '../../../_chunks/dep-4d7f902f.js';
import '../../../config-provider/hooks/useConfig.js';
import '../../../config-provider/utils/context.js';
import '../../../_chunks/dep-854d2777.js';
import 'dayjs';
import '../../../_chunks/dep-dfeea6f5.js';
import '../../../_chunks/dep-9e860115.js';
import '../../../select-input/index.js';
import '../../../select-input/select-input.js';
import '../../../popup/index.js';
import '../../../popup/popup.js';
import '@popperjs/core';
import '../../../_chunks/dep-41cafcc6.js';
import '../../../_chunks/dep-b3575781.js';
import '../../../_chunks/dep-43a3baa0.js';
import '../../../_chunks/dep-9e2a2dff.js';
import '../../../_chunks/dep-cdc7f78f.js';
import '../../../_chunks/dep-cc2cf248.js';
import '../../../popup/container.js';
import '../../../popup/props.js';
import '../../../_chunks/dep-b239e51b.js';
import '../../../_chunks/dep-c53cb022.js';
import 'tdesign-vue-next/esm/common/style/web/components/popup/_index.less';
import '../../../select-input/props.js';
import '../../../select-input/hooks/useMultiple.js';
import '../../../tag-input/index.js';
import '../../../tag-input/tag-input.js';
import 'tdesign-icons-vue-next';
import '../../../input/index.js';
import '../../../input/input.js';
import '../../../input/props.js';
import '../../../_chunks/dep-a7c4f6fa.js';
import '../../../_chunks/dep-b4a20622.js';
import '../../../_chunks/dep-8079b4fa.js';
import '../../../input/hooks/useInput.js';
import '../../../form/constants/index.js';
import '../../../input/hooks/useLengthLimit.js';
import '../../../_chunks/dep-2721b80a.js';
import '../../../input/hooks/useInputEventHandler.js';
import '../../../input/hooks/useInputWidth.js';
import '../../../input/input-group.js';
import '../../../input/input-group-props.js';
import 'tdesign-vue-next/esm/common/style/web/components/input/_index.less';
import '../../../tag-input/props.js';
import '../../../_chunks/dep-b6de9ec6.js';
import '../../../_chunks/dep-03d84be1.js';
import '../../../tag-input/hooks/useDragSorter.js';
import '../../../tag-input/hooks/useHover.js';
import '../../../tag-input/hooks/useTagScroll.js';
import '../../../tag-input/hooks/useTagList.js';
import '../../../tag/index.js';
import '../../../tag/tag.js';
import '../../../tag/props.js';
import '../../../tag/check-tag.js';
import '../../../tag/check-tag-props.js';
import '../../../_chunks/dep-d19ff2c3.js';
import '../../../tag/check-tag-group.js';
import '../../../tag/check-tag-group-props.js';
import 'tdesign-vue-next/esm/common/style/web/components/tag/_index.less';
import 'tdesign-vue-next/esm/common/style/web/components/tag-input/_index.less';
import '../../../loading/index.js';
import '../../../loading/directive.js';
import '../../../loading/plugin.js';
import '../../../_chunks/dep-c9b67f18.js';
import '../../../loading/icon/gradient.js';
import '../../../loading/props.js';
import 'tdesign-vue-next/esm/common/style/web/components/loading/_index.less';
import '../../../select-input/hooks/useOverlayInnerStyle.js';
import '../../../select-input/hooks/useSingle.js';
import 'tdesign-vue-next/esm/common/style/web/components/select-input/_index.less';
import '../../../select/components/select-panel.js';
import '../../../select/option.js';
import '../../../select/option-props.js';
import '../../../checkbox/index.js';
import '../../../checkbox/checkbox.js';
import '../../../checkbox/props.js';
import '../../../_chunks/dep-735205fa.js';
import '../../../checkbox/constants/index.js';
import '../../../checkbox/hooks/useCheckboxLazyLoad.js';
import '../../../_chunks/dep-1b49e47a.js';
import '../../../checkbox/hooks/useKeyboardEvent.js';
import '../../../checkbox/group.js';
import '../../../checkbox/checkbox-group-props.js';
import '../../../_chunks/dep-02e12eec.js';
import 'tdesign-vue-next/esm/common/style/web/components/checkbox/_index.less';
import '../../../select/utils/index.js';
import '../../../select/constants/index.js';
import '../../../select/option-group.js';
import '../../../select/option-group-props.js';
import '../../../select/props.js';
import '../../../select/hooks/usePanelVirtualScroll.js';
import '../../../select/hooks/useKeyboardControl.js';
import '../../../select/hooks/useSelectOptions.js';
import 'tdesign-vue-next/esm/common/style/web/components/select/_index.less';
import '../../../input-number/index.js';
import '../../../input-number/input-number.js';
import '../../../button/index.js';
import '../../../button/button.js';
import '../../../button/props.js';
import 'tdesign-vue-next/esm/common/style/web/components/button/_index.less';
import '../../../input-number/props.js';
import '../../../input-number/hooks/useInputNumber.js';
import '../../../_chunks/dep-a505ff81.js';
import '../../../_chunks/dep-4780ce51.js';
import 'tdesign-vue-next/esm/common/style/web/components/input-number/_index.less';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !isVNode(s);
}
var FormatPanel = defineComponent({
  name: "FormatPanel",
  inheritAttrs: false,
  props: _objectSpread(_objectSpread({}, props), {}, {
    color: {
      type: Object
    },
    onInputChange: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    }
  }),
  setup: function setup(props2) {
    var baseClassName = useBaseClassName();
    var formatModel = ref(props2.format);
    watch(function () {
      return [props2.format];
    }, function () {
      return formatModel.value = props2.format;
    });
    var handleModeChange = function handleModeChange(v) {
      formatModel.value = v;
    };
    return function () {
      var _slot;
      var newProps = _objectSpread(_objectSpread({}, props2), {}, {
        format: formatModel.value
      });
      var selectInputProps = _objectSpread({}, props2.selectInputProps || {});
      return createVNode("div", {
        "class": "".concat(baseClassName.value, "__format")
      }, [createVNode("div", {
        "class": "".concat(baseClassName.value, "__format--item")
      }, [createVNode(Select, {
        "size": "small",
        "class": "".concat(baseClassName.value, "__format-mode-select"),
        "selectInputProps": _objectSpread({}, selectInputProps),
        "popupProps": _objectSpread({
          overlayClassName: "".concat(baseClassName.value, "__select-options")
        }, selectInputProps.popupProps),
        "modelValue": formatModel.value,
        "onUpdate:modelValue": function onUpdateModelValue($event) {
          return formatModel.value = $event;
        },
        "onChange": handleModeChange
      }, _isSlot(_slot = getColorFormatOptions(props2.enableAlpha).map(function (item) {
        return createVNode(Option, {
          "key": item,
          "value": item,
          "label": item,
          "style": {
            fontSize: "12px"
          }
        }, null);
      })) ? _slot : {
        "default": function _default() {
          return [_slot];
        }
      })]), createVNode("div", {
        "class": "".concat(baseClassName.value, "__format--item")
      }, [createVNode(FormatInputs, newProps, null)])]);
    };
  }
});

export { FormatPanel as default };
