// tdesign-vue-next/cjs/card/card.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-449273df.js');
var index = require('../_chunks/dep-c0cb4d13.js');
var index$2 = require('../_chunks/dep-9ada3ec1.js');
var index$1 = require('../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-497fc9a5.js');
require('@babel/runtime/helpers/defineProperty');
var loading_index = require('../loading/index.js');
var card_props = require('./props.js');
var isString = require('../_chunks/dep-be760bc2.js');
require('../_chunks/dep-f9b59444.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-80827572.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-cfdee5e7.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../_chunks/dep-203a5570.js');
require('../loading/plugin.js');
require('../_chunks/dep-3231adce.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-e3eb5659.js');
require('../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-6130f266.js');
require('../_chunks/dep-3ffe18b0.js');
require('../_chunks/dep-e41b3434.js');
require('../loading/props.js');
require('../_chunks/dep-e46285c0.js');
require('../_chunks/dep-54801354.js');
require('../_chunks/dep-bd59ca66.js');
require('../_chunks/dep-f7508ffa.js');
require('../_chunks/dep-470f211b.js');
require('../_chunks/dep-9b549c85.js');
require('../_chunks/dep-e25ae470.js');
require('../_chunks/dep-bed84553.js');
require('../_chunks/dep-a4b963ab.js');
require('../_chunks/dep-abc005aa.js');

var _Card = Vue.defineComponent({
  name: "TCard",
  props: card_props["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var renderTNodeJSX = index.useTNodeJSX();
    var COMPONENT_NAME = index$1.usePrefixClass("card");
    var _useCommonClassName = index$2.useCommonClassName(),
      SIZE = _useCommonClassName.SIZE;
    var baseCls = Vue.computed(function () {
      var defaultClass = [COMPONENT_NAME.value];
      if (props2.size === "small") defaultClass.push("".concat(SIZE.value[props2.size]));
      if (props2.bordered) defaultClass.push("".concat(COMPONENT_NAME.value, "--bordered"));
      if (props2.shadow) defaultClass.push("".concat(COMPONENT_NAME.value, "--shadow"));
      if (props2.hoverShadow) defaultClass.push("".concat(COMPONENT_NAME.value, "--shadow-hover"));
      return defaultClass;
    });
    var headerCls = Vue.computed(function () {
      return ["".concat(COMPONENT_NAME.value, "__header"), props2.headerBordered && "".concat(COMPONENT_NAME.value, "__title--bordered")];
    });
    var headerWrapperCls = index$1.usePrefixClass("card__header-wrapper");
    var headerAvatarCls = index$1.usePrefixClass("card__avatar");
    var headerTitleCls = index$1.usePrefixClass("card__title");
    var headerSubTitleCls = index$1.usePrefixClass("card__subtitle");
    var headerDescriptionCls = index$1.usePrefixClass("card__description");
    var actionsCls = index$1.usePrefixClass("card__actions");
    var bodyCls = index$1.usePrefixClass("card__body");
    var coverCls = index$1.usePrefixClass("card__cover");
    var footerCls = index$1.usePrefixClass("card__footer");
    var footerWrapperCls = index$1.usePrefixClass("card__footer-wrapper");
    var isPoster2 = Vue.computed(function () {
      return props2.theme === "poster2";
    });
    var showTitle = Vue.computed(function () {
      return props2.title || slots.title;
    });
    var showHeader = Vue.computed(function () {
      return props2.header || slots.header;
    });
    var showSubtitle = Vue.computed(function () {
      return props2.subtitle || slots.subtitle;
    });
    var showAvatar = Vue.computed(function () {
      return props2.avatar || slots.avatar;
    });
    var showDescription = Vue.computed(function () {
      return props2.description || slots.description;
    });
    var showStatus = Vue.computed(function () {
      return props2.status || slots.status;
    });
    var showActions = Vue.computed(function () {
      return props2.actions || slots.actions;
    });
    var showFooter = Vue.computed(function () {
      return props2.footer || slots.footer;
    });
    var showCover = Vue.computed(function () {
      return props2.cover || slots.cover;
    });
    var showLoading = Vue.computed(function () {
      return props2.loading || slots.loading;
    });
    var showContent = Vue.computed(function () {
      return props2.content || slots.content || props2["default"] || slots["default"];
    });
    var isHeaderRender = Vue.computed(function () {
      return showHeader.value || showTitle.value || showSubtitle.value || showDescription.value || showAvatar.value || showStatus.value && isPoster2.value || showActions.value && !isPoster2.value;
    });
    var isFooterRender = Vue.computed(function () {
      return showFooter.value || showActions.value && isPoster2.value;
    });
    var renderContent = function renderContent() {
      var isLoading = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
      return Vue.createVNode("div", {
        "class": baseCls.value
      }, [isHeaderRender.value ? renderHeader() : null, showCover.value ? renderCover() : null, (showContent.value || isLoading) && Vue.createVNode(loading_index.Loading, Vue.mergeProps({
        "loading": isLoading
      }, props2.loadingProps), {
        "default": function _default() {
          return [Vue.createVNode("div", {
            "class": [bodyCls.value, props2.bodyClassName],
            "style": props2.bodyStyle
          }, [renderTNodeJSX("default") || renderTNodeJSX("content")])];
        }
      }), isFooterRender.value && Vue.createVNode("div", {
        "class": [footerCls.value, props2.footerClassName],
        "style": props2.footerStyle
      }, [Vue.createVNode("div", {
        "class": footerWrapperCls.value
      }, [renderTNodeJSX("footer")]), showActions.value && isPoster2.value && Vue.createVNode("div", {
        "class": actionsCls.value
      }, [renderTNodeJSX("actions")])])]);
    };
    var renderHeader = function renderHeader() {
      if (showHeader.value) return Vue.createVNode("div", {
        "class": [headerCls.value, props2.headerClassName],
        "style": props2.headerStyle
      }, [renderTNodeJSX("header")]);
      return Vue.createVNode("div", {
        "class": [headerCls.value, props2.headerClassName],
        "style": props2.headerStyle
      }, [Vue.createVNode("div", {
        "class": headerWrapperCls.value
      }, [showAvatar.value && Vue.createVNode("div", {
        "class": headerAvatarCls.value
      }, [renderTNodeJSX("avatar")]), Vue.createVNode("div", null, [showTitle.value && Vue.createVNode("div", {
        "class": headerTitleCls.value
      }, [renderTNodeJSX("title")]), showSubtitle.value && Vue.createVNode("div", {
        "class": headerSubTitleCls.value
      }, [renderTNodeJSX("subtitle")]), showDescription.value && Vue.createVNode("p", {
        "class": headerDescriptionCls.value
      }, [renderTNodeJSX("description")])])]), showActions.value && !isPoster2.value && Vue.createVNode("div", {
        "class": actionsCls.value
      }, [renderTNodeJSX("actions")]), showStatus.value && Vue.createVNode("div", {
        "class": actionsCls.value
      }, [renderTNodeJSX("status")])]);
    };
    var renderCover = function renderCover() {
      var textCover = isString.isString(props2.cover);
      return Vue.createVNode("div", {
        "class": coverCls.value
      }, [textCover ? Vue.createVNode("img", {
        "src": props2.cover
      }, null) : renderTNodeJSX("cover")]);
    };
    return function () {
      var loadingContent = renderTNodeJSX("loading");
      if (loadingContent) {
        return showLoading.value ? loadingContent : renderContent();
      }
      var isLoading = !!showLoading.value;
      return renderContent(isLoading);
    };
  }
});

exports["default"] = _Card;
