// tdesign-vue-next/lib/tree/utils/adapt.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import * as Vue from 'vue';
export { TransitionGroup, computed, defineComponent, onMounted, reactive, ref, toRefs, watch } from 'vue';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../../_chunks/dep-d9440b5f.js';
import { w as withInstall$1 } from '../../_chunks/dep-c53cb022.js';
export { u as usePrefixClass } from '../../_chunks/dep-4d7f902f.js';
import { u as useDefaultValue } from '../../_chunks/dep-b6de9ec6.js';
export { u as useGlobalIcon } from '../../_chunks/dep-b4a20622.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../_chunks/dep-dfeea6f5.js';
export { u as useLazyLoad } from '../../_chunks/dep-6a1a3748.js';
import { u as useVModel$1 } from '../../_chunks/dep-cc2cf248.js';
import '@babel/runtime/helpers/defineProperty';
export { u as useVirtualScrollNew } from '../../_chunks/dep-8e2df0a3.js';
export { CaretRightSmallIcon as TdCaretRightSmallIcon } from 'tdesign-icons-vue-next';
export { Checkbox as TCheckBox } from '../../checkbox/index.js';
export { Loading as TLoading } from '../../loading/index.js';
export { T as TreeNode, p as privateKey } from '../../_chunks/dep-0a9c5f13.js';
export { useConfig } from '../../config-provider/hooks/useConfig.js';
import '../../_chunks/dep-a6042a25.js';
import '../../config-provider/utils/context.js';
import '../../_chunks/dep-854d2777.js';
import 'dayjs';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../../_chunks/dep-1b49e47a.js';
import '../../_chunks/dep-41cafcc6.js';
import '../../_chunks/dep-b239e51b.js';
import '../../checkbox/checkbox.js';
import '../../checkbox/props.js';
import '../../_chunks/dep-b3575781.js';
import '../../_chunks/dep-43a3baa0.js';
import '../../_chunks/dep-9e2a2dff.js';
import '../../_chunks/dep-a7c4f6fa.js';
import '../../_chunks/dep-8079b4fa.js';
import '../../_chunks/dep-735205fa.js';
import '../../_chunks/dep-cdc7f78f.js';
import '../../checkbox/constants/index.js';
import '../../checkbox/hooks/useCheckboxLazyLoad.js';
import '../../checkbox/hooks/useKeyboardEvent.js';
import '../../_chunks/dep-d19ff2c3.js';
import '../../checkbox/group.js';
import '../../checkbox/checkbox-group-props.js';
import '../../_chunks/dep-02e12eec.js';
import '../../loading/directive.js';
import '../../loading/plugin.js';
import '../../_chunks/dep-c9b67f18.js';
import '../../loading/icon/gradient.js';
import '../../_chunks/dep-9e860115.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../loading/props.js';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/regenerator';
import '../../_chunks/dep-2721b80a.js';

var onDrag = {
  "default": void 0
};
var isVueNext = true;
var TreeItemDefinition = {
  name: "TTreeNode",
  inject: {
    onDrag: onDrag
  }
};
function useRipple(el) {}
function withInstall(construct) {
  return withInstall$1(construct);
}
function getCreateElement(h) {
  return Vue.h;
}
function getScopedSlots(instance) {
  return instance.$slots;
}
function useVModel(props, refsProps) {
  var propName = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : "value";
  var defaultPropName = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : "defaultValue";
  var eventPropName = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : "onChange";
  var modelValue = refsProps.modelValue;
  var vm;
  if (propName === "value") {
    vm = useVModel$1(refsProps[propName], modelValue, props[defaultPropName], props[eventPropName], propName);
  } else {
    vm = useDefaultValue(refsProps[propName], props[defaultPropName], props[eventPropName], propName);
  }
  return vm;
}

export { TreeItemDefinition, getCreateElement, getScopedSlots, isVueNext, useRipple, useVModel, withInstall };
