// tdesign-vue-next/esm/tree/hooks/useItemState.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { toRefs, ref } from 'vue';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../../_chunks/dep-d9440b5f.js';
import '../../config-provider/hooks/useConfig.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../_chunks/dep-dfeea6f5.js';
import '@babel/runtime/helpers/defineProperty';
import 'tdesign-icons-vue-next';
import '../../checkbox/index.js';
import '../../loading/index.js';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/helpers/classCallCheck';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/regenerator';
import '../../_chunks/dep-a6042a25.js';
import '../../config-provider/utils/context.js';
import '../../_chunks/dep-854d2777.js';
import 'dayjs';
import '../../checkbox/checkbox.js';
import '../../checkbox/props.js';
import '../../_chunks/dep-b3575781.js';
import '../../_chunks/dep-41cafcc6.js';
import '../../_chunks/dep-43a3baa0.js';
import '../../_chunks/dep-9e2a2dff.js';
import '../../_chunks/dep-4d7f902f.js';
import '../../_chunks/dep-a7c4f6fa.js';
import '../../_chunks/dep-8079b4fa.js';
import '../../_chunks/dep-735205fa.js';
import '../../_chunks/dep-cdc7f78f.js';
import '../../_chunks/dep-cc2cf248.js';
import '../../checkbox/constants/index.js';
import '../../checkbox/hooks/useCheckboxLazyLoad.js';
import '../../_chunks/dep-1b49e47a.js';
import '../../checkbox/hooks/useKeyboardEvent.js';
import '../../_chunks/dep-d19ff2c3.js';
import '../../checkbox/group.js';
import '../../checkbox/checkbox-group-props.js';
import '../../_chunks/dep-02e12eec.js';
import '../../_chunks/dep-c53cb022.js';
import 'tdesign-vue-next/esm/common/style/web/components/checkbox/_index.less';
import '../../loading/directive.js';
import '../../loading/plugin.js';
import '../../_chunks/dep-c9b67f18.js';
import '../../loading/icon/gradient.js';
import '../../_chunks/dep-9e860115.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../loading/props.js';
import 'tdesign-vue-next/esm/common/style/web/components/loading/_index.less';

function useItemState(props, context) {
  var refProps = toRefs(props);
  var treeItemRef = ref(null);
  var treeScope = props.treeScope;
  var store = treeScope.store;
  var node = store.privateMap.get(props.itemKey);
  var state = {
    stateId: props.stateId,
    props: props,
    context: context,
    treeScope: treeScope,
    refProps: refProps,
    node: node,
    treeItemRef: treeItemRef
  };
  return {
    state: state
  };
}

export { useItemState as default };
