// tdesign-vue-next/cjs/transfer/transfer.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var transfer_components_transferList = require('./components/transfer-list.js');
var transfer_components_transferOperations = require('./components/transfer-operations.js');
var transfer_utils_index = require('./utils/index.js');
var transfer_props = require('./props.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-449273df.js');
var index$1 = require('../_chunks/dep-95cb1381.js');
var index$3 = require('../_chunks/dep-4b52068a.js');
var index = require('../_chunks/dep-a2e46896.js');
require('../_chunks/dep-497fc9a5.js');
var index$2 = require('../_chunks/dep-d0227fbe.js');
var isFunction = require('../_chunks/dep-80827572.js');
var pick = require('../_chunks/dep-008d4a55.js');
var isEqual = require('../_chunks/dep-e46285c0.js');
require('../pagination/index.js');
require('../pagination/pagination.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('tdesign-icons-vue-next');
require('../_chunks/dep-c0cb4d13.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-be760bc2.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-cfdee5e7.js');
require('../_chunks/dep-88ae5d8a.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('../_chunks/dep-f9b59444.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('../input-number/index.js');
require('../input-number/input-number.js');
require('../button/index.js');
require('../button/button.js');
require('../loading/index.js');
require('../_chunks/dep-203a5570.js');
require('../loading/plugin.js');
require('../_chunks/dep-3231adce.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-e3eb5659.js');
require('../_chunks/dep-cce7af6a.js');
require('../_chunks/dep-6130f266.js');
require('../_chunks/dep-3ffe18b0.js');
require('../_chunks/dep-e41b3434.js');
require('../_chunks/dep-9ada3ec1.js');
require('../loading/props.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../_chunks/dep-470f211b.js');
require('../_chunks/dep-54801354.js');
require('../_chunks/dep-bd59ca66.js');
require('../_chunks/dep-f7508ffa.js');
require('../_chunks/dep-9b549c85.js');
require('../_chunks/dep-e25ae470.js');
require('../_chunks/dep-bed84553.js');
require('../_chunks/dep-a4b963ab.js');
require('../_chunks/dep-abc005aa.js');
require('../button/props.js');
require('../_chunks/dep-3cd26986.js');
require('../_chunks/dep-6512c9d2.js');
require('../input/index.js');
require('../input/input.js');
require('../input/props.js');
require('../_chunks/dep-f02d9258.js');
require('../input/hooks/useInput.js');
require('../form/constants/index.js');
require('../input/hooks/useLengthLimit.js');
require('../_chunks/dep-b4528c21.js');
require('../_chunks/dep-d20c4e5e.js');
require('../input/hooks/useInputEventHandler.js');
require('../input/hooks/useInputWidth.js');
require('../_chunks/dep-ba2511b0.js');
require('../input/input-group.js');
require('../input/input-group-props.js');
require('../input-number/props.js');
require('../_chunks/dep-5ea73cce.js');
require('../input-number/hooks/useInputNumber.js');
require('../_chunks/dep-74a0ebc2.js');
require('../_chunks/dep-94323f13.js');
require('../_chunks/dep-75d3928f.js');
require('../select/index.js');
require('../select/select.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../common-components/fake-arrow.js');
require('../select-input/index.js');
require('../select-input/select-input.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../popup/container.js');
require('../popup/props.js');
require('../_chunks/dep-4e2a4791.js');
require('../_chunks/dep-f3377589.js');
require('../_chunks/dep-71d9daf0.js');
require('../select-input/props.js');
require('../select-input/hooks/useMultiple.js');
require('../tag-input/index.js');
require('../tag-input/tag-input.js');
require('../tag-input/props.js');
require('../tag-input/hooks/useDragSorter.js');
require('../tag-input/hooks/useHover.js');
require('../tag-input/hooks/useTagScroll.js');
require('../tag-input/hooks/useTagList.js');
require('../tag/index.js');
require('../tag/tag.js');
require('tinycolor2');
require('../tag/props.js');
require('../tag/check-tag.js');
require('../tag/check-tag-props.js');
require('../_chunks/dep-2e8277b1.js');
require('../tag/check-tag-group.js');
require('../tag/check-tag-group-props.js');
require('../select-input/hooks/useOverlayInnerStyle.js');
require('../select-input/hooks/useSingle.js');
require('../_chunks/dep-0e1751ee.js');
require('../_chunks/dep-4e2907e0.js');
require('../select/components/select-panel.js');
require('../select/option.js');
require('../select/option-props.js');
require('../checkbox/index.js');
require('../checkbox/checkbox.js');
require('../checkbox/props.js');
require('../checkbox/constants/index.js');
require('../checkbox/hooks/useCheckboxLazyLoad.js');
require('../_chunks/dep-12e6ec24.js');
require('../checkbox/hooks/useKeyboardEvent.js');
require('../checkbox/group.js');
require('../checkbox/checkbox-group-props.js');
require('../_chunks/dep-cec24bbf.js');
require('../_chunks/dep-f786e1df.js');
require('../_chunks/dep-618d56e7.js');
require('../_chunks/dep-2cf25d25.js');
require('../select/utils/index.js');
require('../select/constants/index.js');
require('../select/option-group.js');
require('../select/option-group-props.js');
require('../select/props.js');
require('../select/hooks/usePanelVirtualScroll.js');
require('../_chunks/dep-3a800699.js');
require('../_chunks/dep-648bfb84.js');
require('../select/hooks/useKeyboardControl.js');
require('../select/hooks/useSelectOptions.js');
require('../_chunks/dep-8f2069e9.js');
require('../input-adornment/index.js');
require('../input-adornment/input-adornment.js');
require('../input-adornment/props.js');
require('../pagination/props.js');
require('../pagination/hooks/useMoreAction.js');
require('../pagination/hooks/usePaginationClasses.js');
require('../pagination/pagination-mini.js');
require('../pagination/pagination-mini-props.js');
require('../tooltip/index.js');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../tooltip/utils/index.js');
require('./components/transfer-search.js');
require('./hooks/useDragSort.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var _Transfer = Vue.defineComponent({
  name: transfer_utils_index.TRANSFER_NAME,
  props: transfer_props["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var disabled = index.useDisabled();
    var classPrefix = index$1.usePrefixClass();
    var _toRefs = Vue.toRefs(props2),
      value = _toRefs.value,
      modelValue = _toRefs.modelValue,
      checked = _toRefs.checked;
    var _useVModel = index$2.useVModel(value, modelValue, props2.defaultValue, props2.onChange),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      innerValue = _useVModel2[0],
      setInnerValue = _useVModel2[1];
    var _useDefaultValue = index$3.useDefaultValue(checked, props2.defaultChecked, props2.onCheckedChange, "checked"),
      _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 1),
      innerChecked = _useDefaultValue2[0];
    var valueList = Vue.computed(function () {
      return innerValue.value;
    });
    var isTreeMode = Vue.computed(function () {
      var treeSlot = slots.tree;
      return isFunction.isFunction(treeSlot);
    });
    var leftButtonDisabled = Vue.computed(function () {
      return props2.direction === "right";
    });
    var rightButtonDisabled = Vue.computed(function () {
      return props2.direction === "left";
    });
    var transferData = Vue.computed(function () {
      return transfer_utils_index.getTransferData(props2.data, props2.keys, isTreeMode.value);
    });
    var sourceList = Vue.computed(function () {
      return transfer_utils_index.filterTransferData(transferData.value, valueList.value, false, isTreeMode.value);
    });
    var targetList = Vue.computed(function () {
      return transfer_utils_index.filterTransferData(transferData.value, valueList.value, true, isTreeMode.value);
    });
    var checkedValue = Vue.computed(function () {
      return _defineProperty__default["default"](_defineProperty__default["default"]({}, transfer_utils_index.SOURCE, transfer_utils_index.getDataValues(sourceList.value, innerChecked.value, {
        isTreeMode: isTreeMode.value
      })), transfer_utils_index.TARGET, transfer_utils_index.getDataValues(targetList.value, innerChecked.value, {
        isTreeMode: isTreeMode.value
      }));
    });
    var hasFooter = Vue.computed(function () {
      return !!slots.footer || !!props2.footer;
    });
    var showPagination = Vue.computed(function () {
      return !!props2.pagination && !slots.content;
    });
    var showSearch = Vue.computed(function () {
      return !!props2.search;
    });
    var footerOption = Vue.computed(function () {
      var footer = props2.footer || "";
      return transfer_utils_index.getTransferListOption(footer);
    });
    var emptyOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.empty);
    });
    var searchOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.search);
    });
    var checkAllOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.showCheckAll);
    });
    var disabledOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(disabled.value);
    });
    var titleOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.title);
    });
    var paginationOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.pagination);
    });
    var handleCheckedChange = function handleCheckedChange(val, listType) {
      var _props2$onCheckedChan;
      var sourceChecked = listType === transfer_utils_index.SOURCE ? val : checkedValue.value[transfer_utils_index.SOURCE];
      var targetChecked = listType === transfer_utils_index.TARGET ? val : checkedValue.value[transfer_utils_index.TARGET];
      var checked2 = [].concat(_toConsumableArray__default["default"](sourceChecked), _toConsumableArray__default["default"](targetChecked));
      var event = {
        checked: checked2,
        sourceChecked: sourceChecked,
        targetChecked: targetChecked,
        type: listType
      };
      innerChecked.value = checked2;
      (_props2$onCheckedChan = props2.onCheckedChange) === null || _props2$onCheckedChan === void 0 || _props2$onCheckedChan.call(props2, event);
    };
    var transferTo = function transferTo(toDirection) {
      var oldTargetValue = JSON.parse(JSON.stringify(valueList.value));
      var newTargetValue;
      var selfCheckedValue = toDirection === transfer_utils_index.TARGET ? checkedValue.value[transfer_utils_index.SOURCE] : checkedValue.value[transfer_utils_index.TARGET];
      if (toDirection === transfer_utils_index.SOURCE) {
        newTargetValue = oldTargetValue.filter(function (v) {
          return !selfCheckedValue.some(function (cv) {
            return isEqual.isEqual(cv, v);
          });
        });
      } else if (props2.targetSort === "original") {
        var remainValue = transferData.value.reduce(function (acc, data) {
          if (oldTargetValue.some(function (v) {
            return isEqual.isEqual(v, data.value);
          }) && data.disabled) {
            return acc.concat(data.value);
          }
          return acc;
        }, []);
        newTargetValue = transfer_utils_index.getDataValues(transferData.value, oldTargetValue.concat(selfCheckedValue), {
          isTreeMode: isTreeMode.value,
          remainValue: remainValue
        });
      } else if (props2.targetSort === "unshift") {
        newTargetValue = selfCheckedValue.concat(oldTargetValue);
      } else {
        newTargetValue = oldTargetValue.concat(selfCheckedValue);
      }
      handleCheckedChange([], toDirection === transfer_utils_index.SOURCE ? transfer_utils_index.TARGET : transfer_utils_index.SOURCE);
      var params = {
        type: toDirection,
        movedValue: selfCheckedValue
      };
      setInnerValue(newTargetValue, params);
    };
    var transferToRight = function transferToRight() {
      transferTo(transfer_utils_index.TARGET);
    };
    var transferToLeft = function transferToLeft() {
      transferTo(transfer_utils_index.SOURCE);
    };
    var handleScroll = function handleScroll(e, listType) {
      var _props2$onScroll;
      var target = e.target;
      var bottomDistance = target.scrollHeight - target.scrollTop - target.clientHeight;
      var event = {
        e: e,
        bottomDistance: bottomDistance,
        type: listType
      };
      (_props2$onScroll = props2.onScroll) === null || _props2$onScroll === void 0 || _props2$onScroll.call(props2, event);
    };
    var handleSearch = function handleSearch(e) {
      var _props2$onSearch;
      (_props2$onSearch = props2.onSearch) === null || _props2$onSearch === void 0 || _props2$onSearch.call(props2, e);
    };
    var handlePageChange = function handlePageChange(pageInfo, listType) {
      var _props2$onPageChange;
      (_props2$onPageChange = props2.onPageChange) === null || _props2$onPageChange === void 0 || _props2$onPageChange.call(props2, pageInfo, {
        type: listType
      });
    };
    var handleDataChange = function handleDataChange(data, movedValue) {
      setInnerValue(data, {
        type: transfer_utils_index.TARGET,
        movedValue: movedValue
      });
    };
    var renderTransferList = function renderTransferList(listType) {
      var scopedSlots = pick.pick(slots, ["title", "empty", "footer", "operation", "transferItem", "default", "tree"]);
      return Vue.createVNode(transfer_components_transferList["default"], {
        "checkboxProps": props2.checkboxProps,
        "transferItem": props2.transferItem,
        "list-type": listType,
        "title": titleOption.value[listType],
        "data-source": listType === transfer_utils_index.SOURCE ? sourceList.value : targetList.value,
        "checked-value": checkedValue.value[listType],
        "disabled": disabledOption.value[listType],
        "search": searchOption.value[listType],
        "pagination": paginationOption.value[listType],
        "check-all": checkAllOption.value[listType],
        "footer": footerOption.value[listType],
        "empty": emptyOption.value[listType],
        "onCheckedChange": function onCheckedChange($event) {
          return handleCheckedChange($event, listType);
        },
        "onScroll": function onScroll($event) {
          return handleScroll($event, listType);
        },
        "onSearch": handleSearch,
        "onPageChange": function onPageChange($event) {
          return handlePageChange($event, listType);
        },
        "isTreeMode": isTreeMode.value,
        "onDataChange": handleDataChange,
        "currentValue": valueList.value,
        "draggable": props2.targetDraggable && listType === transfer_utils_index.TARGET
      }, _isSlot(scopedSlots) ? scopedSlots : {
        "default": function _default() {
          return [scopedSlots];
        }
      });
    };
    return function () {
      return Vue.createVNode("div", {
        "class": ["".concat(classPrefix.value, "-transfer"), showSearch.value ? "".concat(classPrefix.value, "-transfer__search") : "", hasFooter.value ? "".concat(classPrefix.value, "-transfer__footer") : "", showPagination.value ? "".concat(classPrefix.value, "-transfer__pagination") : "", isTreeMode.value ? "".concat(classPrefix.value, "-transfer--with-tree") : ""]
      }, [renderTransferList(transfer_utils_index.SOURCE), Vue.createVNode(transfer_components_transferOperations["default"], {
        "leftDisabled": disabledOption.value[transfer_utils_index.TARGET] || leftButtonDisabled.value || checkedValue.value[transfer_utils_index.TARGET].length === 0,
        "rightDisabled": disabledOption.value[transfer_utils_index.SOURCE] || rightButtonDisabled.value || checkedValue.value[transfer_utils_index.SOURCE].length === 0,
        "operation": props2.operation,
        "onMoveToRight": transferToRight,
        "onMoveToLeft": transferToLeft
      }, {
        operation: slots.operation
      }), renderTransferList(transfer_utils_index.TARGET)]);
    };
  }
});

exports["default"] = _Transfer;
