// tdesign-vue-next/cjs/image/image.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var observe = require('../_chunks/dep-12e6ec24.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-449273df.js');
var index = require('../_chunks/dep-c0cb4d13.js');
var index$1 = require('../_chunks/dep-e04a8c18.js');
require('../_chunks/dep-497fc9a5.js');
var image_props = require('./props.js');
var space_index = require('../space/index.js');
var isFunction = require('../_chunks/dep-80827572.js');
var omit = require('../_chunks/dep-3a800699.js');
var isString = require('../_chunks/dep-be760bc2.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('../_chunks/dep-f9b59444.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-cfdee5e7.js');
require('../_chunks/dep-2c1cb23b.js');
require('../_chunks/dep-b4528c21.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../space/space.js');
require('../space/props.js');
require('../_chunks/dep-cec24bbf.js');
require('../_chunks/dep-95cb1381.js');
require('../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-6130f266.js');
require('../_chunks/dep-3ffe18b0.js');
require('../_chunks/dep-e41b3434.js');
require('../_chunks/dep-d20c4e5e.js');
require('../_chunks/dep-abc005aa.js');
require('../_chunks/dep-e25ae470.js');
require('../_chunks/dep-648bfb84.js');
require('../_chunks/dep-4e2907e0.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var _Image = Vue.defineComponent({
  name: "TImage",
  props: _objectSpread(_objectSpread({}, image_props["default"]), {}, {
    onClick: Function
  }),
  setup: function setup(props2) {
    var divRef = Vue.ref();
    var imgRef = Vue.ref();
    var io = null;
    var _toRefs = Vue.toRefs(props2),
      src = _toRefs.src;
    var renderTNodeJSX = index.useTNodeJSX();
    Vue.onMounted(function () {
      var _imgRef$value;
      if ((_imgRef$value = imgRef.value) !== null && _imgRef$value !== void 0 && _imgRef$value.complete && !props2.lazy) {
        triggerHandleLoad();
      }
      if (!props2.lazy || !divRef.value) return;
      var ioObserver = observe.observe(divRef.value, null, handleLoadImage, 0);
      io = ioObserver;
    });
    Vue.onUnmounted(function () {
      divRef.value && io && io.unobserve(divRef.value);
    });
    var _useConfig = configProvider_hooks_useConfig.useConfig("image"),
      classPrefix = _useConfig.classPrefix,
      globalConfig = _useConfig.globalConfig;
    var imageStrSrc = Vue.ref(src.value);
    Vue.watch([src, globalConfig], function (_ref) {
      var _ref2 = _slicedToArray__default["default"](_ref, 2),
        src2 = _ref2[0],
        globalConfig2 = _ref2[1];
      var _ref3 = globalConfig2 || {},
        replaceImageSrc = _ref3.replaceImageSrc;
      var tmpUrl = isFunction.isFunction(replaceImageSrc) ? replaceImageSrc(props2) : src2;
      if (tmpUrl === imageStrSrc.value && imageStrSrc.value) return;
      imageStrSrc.value = tmpUrl;
    }, {
      immediate: true
    });
    var _useImagePreviewUrl = index$1.useImagePreviewUrl(imageStrSrc),
      previewUrl = _useImagePreviewUrl.previewUrl;
    Vue.watch([previewUrl], function () {
      hasError.value = false;
      isLoaded.value = false;
    });
    var shouldLoad = Vue.ref(!props2.lazy);
    var handleLoadImage = function handleLoadImage() {
      shouldLoad.value = true;
    };
    var isLoaded = Vue.ref(false);
    var handleLoad = function handleLoad(e) {
      var _props2$onLoad;
      isLoaded.value = true;
      (_props2$onLoad = props2.onLoad) === null || _props2$onLoad === void 0 || _props2$onLoad.call(props2, {
        e: e
      });
    };
    var triggerHandleLoad = function triggerHandleLoad() {
      var loadEvent = new Event("load");
      Object.defineProperty(loadEvent, "target", {
        value: imgRef.value,
        enumerable: true
      });
      handleLoad(loadEvent);
    };
    var hasError = Vue.ref(false);
    var handleError = function handleError(e) {
      var _props2$onError;
      hasError.value = true;
      (_props2$onError = props2.onError) === null || _props2$onError === void 0 || _props2$onError.call(props2, {
        e: e
      });
      if (props2.fallback) {
        imageStrSrc.value = props2.fallback;
        hasError.value = false;
      }
    };
    var hasMouseEvent = Vue.computed(function () {
      return props2.overlayTrigger === "hover";
    });
    var imageClasses = Vue.computed(function () {
      return ["".concat(classPrefix.value, "-image"), "".concat(classPrefix.value, "-image--fit-").concat(props2.fit), "".concat(classPrefix.value, "-image--position-").concat(props2.position)];
    });
    var shouldShowOverlay = Vue.ref(!hasMouseEvent.value);
    var handleToggleOverlay = function handleToggleOverlay() {
      if (hasMouseEvent.value) {
        shouldShowOverlay.value = !shouldShowOverlay.value;
      }
    };
    var renderPlaceholder = function renderPlaceholder() {
      var placeholder = renderTNodeJSX("placeholder");
      if (!placeholder) return null;
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-image__placeholder")
      }, [placeholder]);
    };
    var renderGalleryShadow = function renderGalleryShadow() {
      if (!props2.gallery) return null;
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-image__gallery-shadow")
      }, null);
    };
    var renderOverlay = function renderOverlay() {
      var overlayContent = renderTNodDefault("overlayContent");
      if (!overlayContent) return null;
      return Vue.createVNode("div", {
        "class": ["".concat(classPrefix.value, "-image__overlay-content"), !shouldShowOverlay.value && "".concat(classPrefix.value, "-image__overlay-content--hidden")]
      }, [overlayContent]);
    };
    function renderImageSrcset() {
      return Vue.createVNode("picture", null, [Object.entries(props2.srcset).map(function (_ref4) {
        var _ref5 = _slicedToArray__default["default"](_ref4, 2),
          type = _ref5[0],
          url = _ref5[1];
        return Vue.createVNode("source", {
          "type": type,
          "srcset": url
        }, null);
      }), renderImage()]);
    }
    function renderImage() {
      var url = isString.isString(imageStrSrc.value) ? imageStrSrc.value : previewUrl.value;
      return Vue.createVNode("img", {
        "ref": imgRef,
        "src": url,
        "onError": handleError,
        "onLoad": handleLoad,
        "class": imageClasses.value,
        "alt": props2.alt,
        "referrerpolicy": props2.referrerpolicy
      }, null);
    }
    var renderTNodDefault = index.useTNodeDefault();
    return function () {
      return Vue.createVNode("div", Vue.mergeProps({
        "ref": divRef,
        "class": ["".concat(classPrefix.value, "-image__wrapper"), "".concat(classPrefix.value, "-image__wrapper--shape-").concat(props2.shape), props2.gallery && "".concat(classPrefix.value, "-image__wrapper--gallery"), hasMouseEvent.value && "".concat(classPrefix.value, "-image__wrapper--need-hover")],
        "onMouseenter": handleToggleOverlay,
        "onMouseleave": handleToggleOverlay,
        "onClick": function onClick(e) {
          var _props2$onClick;
          return (_props2$onClick = props2.onClick) === null || _props2$onClick === void 0 ? void 0 : _props2$onClick.call(props2, {
            e: e
          });
        }
      }, omit.omit(props2, ["src", "alt", "fit", "position", "shape", "placeholder", "loading", "error", "overlayTrigger", "overlayContent", "lazy", "gallery", "onLoad", "onError", "onClick"])), [renderPlaceholder(), renderGalleryShadow(), (hasError.value || !shouldLoad.value) && Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-image")
      }, null), !(hasError.value || !shouldLoad.value) && (props2.srcset && Object.keys(props2.srcset).length ? renderImageSrcset() : renderImage()), !(hasError.value || !shouldLoad.value) && !isLoaded.value && Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-image__loading")
      }, [renderTNodeJSX("loading") || Vue.createVNode(space_index.Space, {
        "direction": "vertical",
        "size": 8,
        "align": "center"
      }, {
        "default": function _default() {
          return [Vue.createVNode(tdesignIconsVueNext.ImageIcon, null, null), isString.isString(props2.loading) ? props2.loading : globalConfig.value.loadingText];
        }
      })]), hasError.value && Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-image__error")
      }, [renderTNodDefault("error", {
        defaultNode: Vue.createVNode(space_index.Space, {
          "direction": "vertical",
          "size": 8,
          "align": "center"
        }, {
          "default": function _default() {
            return [Vue.createVNode(tdesignIconsVueNext.ImageErrorIcon, null, null), isString.isString(props2.error) ? props2.error : globalConfig.value.errorText];
          }
        })
      })]), renderOverlay()]);
    };
  }
});

exports["default"] = _Image;
