// tdesign-vue-next/cjs/table/hooks/useRowSelect.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var Vue = require('vue');
var _typeof = require('@babel/runtime/helpers/typeof');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-449273df.js');
require('../../config-provider/hooks/useConfig.js');
var index = require('../../_chunks/dep-4b52068a.js');
var log = require('../../_chunks/dep-b4528c21.js');
require('../../_chunks/dep-497fc9a5.js');
var utils = require('../../_chunks/dep-07dc7736.js');
var table_utils_index = require('../utils/index.js');
var checkbox_index = require('../../checkbox/index.js');
var radio_index = require('../../radio/index.js');
var intersection = require('../../_chunks/dep-f786e1df.js');
var get = require('../../_chunks/dep-9b549c85.js');
var isFunction = require('../../_chunks/dep-80827572.js');
require('../../_chunks/dep-f9b59444.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-6e64ef0e.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-6e7a2100.js');
require('../../_chunks/dep-cfdee5e7.js');
require('../../_chunks/dep-1c6213b5.js');
require('../../_chunks/dep-0a17438d.js');
require('../../_chunks/dep-5090b515.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-ed620b03.js');
require('../../_chunks/dep-75d3928f.js');
require('../../_chunks/dep-e129d089.js');
require('../../_chunks/dep-ceabf07b.js');
require('../../_chunks/dep-e25ae470.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../_chunks/dep-c0cb4d13.js');
require('../../_chunks/dep-51846987.js');
require('../../_chunks/dep-be760bc2.js');
require('../../_chunks/dep-9dfd682d.js');
require('../../_chunks/dep-2db31625.js');
require('../../_chunks/dep-39313d05.js');
require('../../_chunks/dep-f9bc7c82.js');
require('../../_chunks/dep-9ada3ec1.js');
require('../../_chunks/dep-95cb1381.js');
require('../../_chunks/dep-a2e46896.js');
require('../../_chunks/dep-6512c9d2.js');
require('../../_chunks/dep-f02d9258.js');
require('../../_chunks/dep-3cd26986.js');
require('../../_chunks/dep-e3eb5659.js');
require('../../_chunks/dep-d0227fbe.js');
require('../../checkbox/constants/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../_chunks/dep-12e6ec24.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../_chunks/dep-2e8277b1.js');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../_chunks/dep-cec24bbf.js');
require('../../_chunks/dep-2cf25d25.js');
require('../../_chunks/dep-bd59ca66.js');
require('../../_chunks/dep-618d56e7.js');
require('../../_chunks/dep-abc005aa.js');
require('../../radio/radio.js');
require('../../radio/props.js');
require('../../radio/constants/index.js');
require('../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6130f266.js');
require('../../_chunks/dep-3ffe18b0.js');
require('../../_chunks/dep-e41b3434.js');
require('../../_chunks/dep-3a800699.js');
require('../../_chunks/dep-648bfb84.js');
require('../../_chunks/dep-4e2907e0.js');
require('../../radio/group.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../radio/radio-group-props.js');
require('../../radio/radio-button.js');
require('../../_chunks/dep-ba2511b0.js');
require('../../radio/hooks/useKeyboard.js');
require('../../watermark/hooks/index.js');
require('../../_chunks/dep-801a06fe.js');
require('../../_chunks/dep-f3377589.js');
require('../../_chunks/dep-e46285c0.js');
require('../../_chunks/dep-54801354.js');
require('../../_chunks/dep-f7508ffa.js');
require('../../_chunks/dep-59ce9497.js');
require('../../_chunks/dep-d20c4e5e.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _typeof__default = /*#__PURE__*/_interopDefaultLegacy(_typeof);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function useRowSelect(props, tableSelectedClasses) {
  var _toRefs = Vue.toRefs(props),
    selectedRowKeys = _toRefs.selectedRowKeys,
    columns = _toRefs.columns,
    rowKey = _toRefs.rowKey,
    data = _toRefs.data,
    reserveSelectedRowOnPaginate = _toRefs.reserveSelectedRowOnPaginate,
    pagination = _toRefs.pagination,
    disableDataPage = _toRefs.disableDataPage;
  var currentPaginateData = Vue.ref(table_utils_index.getLocalPaginationPageData(data.value, pagination.value, disableDataPage.value));
  var selectedRowClassNames = Vue.ref();
  var _useDefaultValue = index.useDefaultValue(selectedRowKeys, props.defaultSelectedRowKeys || [], props.onSelectChange, "selectedRowKeys"),
    _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
    tSelectedRowKeys = _useDefaultValue2[0],
    setTSelectedRowKeys = _useDefaultValue2[1];
  var selectedRowDataMap = Vue.ref(/* @__PURE__ */new Map());
  var selectColumn = Vue.computed(function () {
    return props.columns.find(function (_ref) {
      var type = _ref.type;
      return ["multiple", "single"].includes(type);
    });
  });
  var selectionType = Vue.computed(function () {
    var _selectColumn$value;
    return props.rowSelectionType || ((_selectColumn$value = selectColumn.value) === null || _selectColumn$value === void 0 ? void 0 : _selectColumn$value.type) || "single";
  });
  var canSelectedRows = Vue.computed(function () {
    var currentData = reserveSelectedRowOnPaginate.value ? data.value : currentPaginateData.value;
    return currentData.filter(function (row, rowIndex) {
      return !isDisabled(row, rowIndex);
    });
  });
  var intersectionKeys = Vue.computed(function () {
    return intersection.intersection(tSelectedRowKeys.value, canSelectedRows.value.map(function (t) {
      return get.get(t, props.rowKey || "id");
    }));
  });
  var allowUncheck = Vue.computed(function () {
    var _selectColumn$value2, _selectColumn$value3;
    if (props.rowSelectionAllowUncheck) return true;
    var singleSelectCol = selectionType.value === "single";
    if (!singleSelectCol || !((_selectColumn$value2 = selectColumn.value) !== null && _selectColumn$value2 !== void 0 && _selectColumn$value2.checkProps) || !("allowUncheck" in ((_selectColumn$value3 = selectColumn.value) === null || _selectColumn$value3 === void 0 ? void 0 : _selectColumn$value3.checkProps))) return false;
    return selectColumn.value.checkProps.allowUncheck;
  });
  Vue.watch([data, columns, tSelectedRowKeys, selectColumn, rowKey], function () {
    var _selectColumn$value4;
    var disabledRowFunc = function disabledRowFunc(p) {
      return selectColumn.value.disabled(p) ? tableSelectedClasses.disabled : "";
    };
    var disabledRowClass = (_selectColumn$value4 = selectColumn.value) !== null && _selectColumn$value4 !== void 0 && _selectColumn$value4.disabled ? disabledRowFunc : void 0;
    var selected = new Set(tSelectedRowKeys.value);
    var selectedRowClassFunc = function selectedRowClassFunc(_ref2) {
      var row = _ref2.row;
      var rowId = get.get(row, props.rowKey || "id");
      return selected.has(rowId) ? tableSelectedClasses.selected : "";
    };
    var selectedRowClass = selected.size ? selectedRowClassFunc : void 0;
    selectedRowClassNames.value = [disabledRowClass, selectedRowClass];
  }, {
    immediate: true
  });
  var updateCurrentPaginateData = function updateCurrentPaginateData() {
    currentPaginateData.value = table_utils_index.getLocalPaginationPageData(data.value, pagination.value, disableDataPage.value);
  };
  Vue.watch([data, function () {
    var _pagination$value;
    return (_pagination$value = pagination.value) === null || _pagination$value === void 0 ? void 0 : _pagination$value.current;
  }, function () {
    var _pagination$value2;
    return (_pagination$value2 = pagination.value) === null || _pagination$value2 === void 0 ? void 0 : _pagination$value2.pageSize;
  }, disableDataPage], function () {
    updateCurrentPaginateData();
  });
  function isDisabled(row, rowIndex) {
    return utils.isRowSelectedDisabled(selectColumn.value, row, rowIndex);
  }
  function getSelectedHeader() {
    return function () {
      var isIndeterminate = intersectionKeys.value.length > 0 && intersectionKeys.value.length < canSelectedRows.value.length;
      var isChecked = canSelectedRows.value.length !== 0 && intersectionKeys.value.length === canSelectedRows.value.length;
      return Vue.createVNode(checkbox_index.Checkbox, {
        "checked": isChecked,
        "indeterminate": isIndeterminate,
        "disabled": !canSelectedRows.value.length,
        "onChange": handleSelectAll
      }, null);
    };
  }
  function getRowSelectDisabledData(p) {
    var col = p.col,
      row = p.row,
      rowIndex = p.rowIndex;
    var disabled = isFunction.isFunction(col.disabled) ? col.disabled({
      row: row,
      rowIndex: rowIndex
    }) : col.disabled;
    var checkProps = isFunction.isFunction(col.checkProps) ? col.checkProps({
      row: row,
      rowIndex: rowIndex
    }) : col.checkProps;
    return {
      disabled: disabled || (checkProps === null || checkProps === void 0 ? void 0 : checkProps.disabled),
      checkProps: checkProps
    };
  }
  function renderSelectCell(p) {
    var column = p.col,
      _p$row = p.row,
      row = _p$row === void 0 ? {} : _p$row;
    var checked = tSelectedRowKeys.value.includes(get.get(row, props.rowKey || "id"));
    var _getRowSelectDisabled = getRowSelectDisabledData(p),
      disabled = _getRowSelectDisabled.disabled,
      checkProps = _getRowSelectDisabled.checkProps;
    var selectBoxProps = _objectSpread(_objectSpread({
      checked: checked,
      disabled: disabled
    }, checkProps), {}, {
      onClick: function onClick(e) {
        if (_typeof__default["default"](e) === "object" && "e" in e) {
          var _e$e;
          (_e$e = e.e) === null || _e$e === void 0 || _e$e.stopPropagation();
        } else {
          e === null || e === void 0 || e.stopPropagation();
        }
      },
      onChange: function onChange() {
        return handleSelectChange(row);
      }
    });
    if (column.type === "single") return Vue.createVNode(radio_index.Radio, selectBoxProps, null);
    if (column.type === "multiple") {
      var _props$indeterminateS;
      var isIndeterminate = (_props$indeterminateS = props.indeterminateSelectedRowKeys) !== null && _props$indeterminateS !== void 0 && _props$indeterminateS.length ? props.indeterminateSelectedRowKeys.includes(get.get(row, props.rowKey)) : false;
      return Vue.createVNode(checkbox_index.Checkbox, Vue.mergeProps({
        "indeterminate": isIndeterminate
      }, selectBoxProps), null);
    }
    return null;
  }
  function handleSelectChange() {
    var row = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
    var selectedRowKeys2 = _toConsumableArray__default["default"](tSelectedRowKeys.value);
    var reRowKey = props.rowKey || "id";
    var id = get.get(row, reRowKey);
    var selectedRowIndex = selectedRowKeys2.indexOf(id);
    var isExisted = selectedRowIndex !== -1;
    if (selectionType.value === "multiple") {
      isExisted ? selectedRowKeys2.splice(selectedRowIndex, 1) : selectedRowKeys2.push(id);
    } else if (selectionType.value === "single") {
      selectedRowKeys2 = isExisted && allowUncheck.value ? [] : [id];
    } else {
      log.log.warn("Table", "`column.type` must be one of `multiple` and `single`");
      return;
    }
    setTSelectedRowKeys(selectedRowKeys2, {
      selectedRowData: selectedRowKeys2.map(function (t) {
        return selectedRowDataMap.value.get(t);
      }),
      currentRowKey: id,
      currentRowData: row,
      type: isExisted ? "uncheck" : "check"
    });
  }
  function handleSelectAll(checked) {
    var _selectedRowKeys$valu;
    var reRowKey = props.rowKey || "id";
    var canSelectedRowKeys = canSelectedRows.value.map(function (record) {
      return get.get(record, reRowKey);
    });
    var disabledSelectedRowKeys = ((_selectedRowKeys$valu = selectedRowKeys.value) === null || _selectedRowKeys$valu === void 0 ? void 0 : _selectedRowKeys$valu.filter(function (id) {
      return !canSelectedRowKeys.includes(id);
    })) || [];
    var allIds = checked ? [].concat(_toConsumableArray__default["default"](disabledSelectedRowKeys), _toConsumableArray__default["default"](canSelectedRowKeys)) : _toConsumableArray__default["default"](disabledSelectedRowKeys);
    setTSelectedRowKeys(allIds, {
      selectedRowData: checked ? allIds.map(function (t) {
        return selectedRowDataMap.value.get(t);
      }) : [],
      type: checked ? "check" : "uncheck",
      currentRowKey: "CHECK_ALL_BOX"
    });
  }
  function formatToRowSelectColumn(col) {
    var isSelection = ["multiple", "single"].includes(col.type);
    if (!isSelection) return col;
    return _objectSpread(_objectSpread({}, col), {}, {
      width: col.width || 64,
      className: [tableSelectedClasses.checkCell, col.className],
      cell: function cell(_, p) {
        return renderSelectCell(p);
      },
      title: col.type === "multiple" ? getSelectedHeader() : col.title
    });
  }
  var onInnerSelectRowClick = function onInnerSelectRowClick(_ref3) {
    var row = _ref3.row,
      index = _ref3.index;
    var selectedColIndex = props.columns.findIndex(function (item) {
      return item.colKey === "row-select";
    });
    var disabled = false;
    if (selectedColIndex !== -1) {
      var _getRowSelectDisabled2;
      disabled = (_getRowSelectDisabled2 = getRowSelectDisabledData({
        row: row,
        rowIndex: index,
        col: props.columns[selectedColIndex],
        colIndex: selectedColIndex
      })) === null || _getRowSelectDisabled2 === void 0 ? void 0 : _getRowSelectDisabled2.disabled;
    }
    if (disabled) return;
    handleSelectChange(row);
  };
  Vue.watch([data, rowKey], function () {
    for (var i = 0, len = data.value.length; i < len; i++) {
      selectedRowDataMap.value.set(get.get(data.value[i], rowKey.value || "id"), data.value[i]);
    }
  }, {
    immediate: true
  });
  var showRowSelect = Vue.computed(function () {
    return Boolean(selectColumn.value || props.selectOnRowClick || props.selectedRowKeys);
  });
  var clearAllSelectedRowKeys = function clearAllSelectedRowKeys() {
    setTSelectedRowKeys([], {
      selectedRowData: [],
      currentRowKey: void 0,
      currentRowData: void 0,
      type: "uncheck"
    });
  };
  var handleRowSelectWithAreaSelection = function handleRowSelectWithAreaSelection(_ref4) {
    var activeRowList = _ref4.activeRowList,
      action = _ref4.action;
    if (!showRowSelect.value) return;
    if (action === "clear") {
      clearAllSelectedRowKeys();
      return;
    }
    if (action === "select-all") {
      handleSelectAll(true);
      return;
    }
    if (selectionType.value === "single") {
      if (action === "space-one-selection") {
        handleSelectChange(activeRowList[0].row);
      }
      return;
    }
    var validAreaSelection = activeRowList.filter(function (_ref5) {
      var row = _ref5.row,
        rowIndex = _ref5.rowIndex;
      return !getRowSelectDisabledData({
        row: row,
        rowIndex: rowIndex,
        col: selectColumn.value,
        colIndex: void 0
      }).disabled;
    });
    if (!validAreaSelection.length) return;
    var areaSelectionKeys = validAreaSelection.map(function (_ref6) {
      var row = _ref6.row;
      return get.get(row, props.rowKey);
    });
    var intersectionKeys2 = intersection.intersection(tSelectedRowKeys.value, areaSelectionKeys);
    var toCheck = intersectionKeys2.length !== areaSelectionKeys.length;
    var clearedKeys = tSelectedRowKeys.value.filter(function (key) {
      return !areaSelectionKeys.includes(key);
    });
    var newSelectedRowKeys = toCheck ? _toConsumableArray__default["default"](new Set(tSelectedRowKeys.value.concat(areaSelectionKeys))) : clearedKeys;
    var currentRowData = action === "space-one-selection" ? activeRowList[0].row : void 0;
    setTSelectedRowKeys(newSelectedRowKeys, {
      selectedRowData: activeRowList,
      currentRowKey: get.get(currentRowData, props.rowKey),
      currentRowData: currentRowData,
      type: toCheck ? "check" : "uncheck"
    });
  };
  return {
    selectColumn: selectColumn,
    showRowSelect: showRowSelect,
    selectedRowClassNames: selectedRowClassNames,
    currentPaginateData: currentPaginateData,
    setTSelectedRowKeys: setTSelectedRowKeys,
    formatToRowSelectColumn: formatToRowSelectColumn,
    onInnerSelectRowClick: onInnerSelectRowClick,
    handleRowSelectWithAreaSelection: handleRowSelectWithAreaSelection
  };
}

exports["default"] = useRowSelect;
