// tdesign-vue-next/cjs/table/hooks/useTreeData.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var table_hooks_useTreeDataExpand = require('../../_chunks/dep-3952fe3d.js');
var table_hooks_useClassName = require('./useClassName.js');
var table_components_tr = require('../components/tr.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-449273df.js');
var index$1 = require('../../_chunks/dep-c0cb4d13.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
var index = require('../../_chunks/dep-88ae5d8a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-497fc9a5.js');
var get = require('../../_chunks/dep-9b549c85.js');
require('../../_chunks/dep-4b52068a.js');
require('../../_chunks/dep-cfdee5e7.js');
require('../../_chunks/dep-1c6213b5.js');
require('../../_chunks/dep-0a17438d.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-5090b515.js');
require('../../_chunks/dep-6e64ef0e.js');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
require('../../_chunks/dep-07dc7736.js');
require('../../_chunks/dep-ed620b03.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-75d3928f.js');
require('../../_chunks/dep-e129d089.js');
require('../../_chunks/dep-ceabf07b.js');
require('../../_chunks/dep-b4528c21.js');
require('../../_chunks/dep-6130f266.js');
require('../../_chunks/dep-37eb0be5.js');
require('../../_chunks/dep-e600142f.js');
require('../../_chunks/dep-95cb1381.js');
require('../../_chunks/dep-f9b59444.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-80827572.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-6e7a2100.js');
require('../utils/index.js');
require('../../_chunks/dep-e25ae470.js');
require('./useFixed.js');
require('../../_chunks/dep-6530011a.js');
require('../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-3ffe18b0.js');
require('../../_chunks/dep-e41b3434.js');
require('../../_chunks/dep-51846987.js');
require('../../_chunks/dep-be760bc2.js');
require('../../_chunks/dep-bd59ca66.js');
require('../../_chunks/dep-618d56e7.js');
require('../../_chunks/dep-4e2907e0.js');
require('../../_chunks/dep-8f2069e9.js');
require('../../_chunks/dep-f7508ffa.js');
require('../../_chunks/dep-648bfb84.js');
require('../../_chunks/dep-008d4a55.js');
require('../../_chunks/dep-0e1751ee.js');
require('../../_chunks/dep-bed84553.js');
require('../../_chunks/dep-f3377589.js');
require('../components/ellipsis.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../popup/props.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../_chunks/dep-9ada3ec1.js');
require('../../_chunks/dep-e3eb5659.js');
require('../../_chunks/dep-d0227fbe.js');
require('../../popup/container.js');
require('../../_chunks/dep-ba2511b0.js');
require('../../_chunks/dep-4e2a4791.js');
require('../../_chunks/dep-71d9daf0.js');
require('../../_chunks/dep-9dfd682d.js');
require('../../_chunks/dep-2db31625.js');
require('../../_chunks/dep-39313d05.js');
require('../../_chunks/dep-f9bc7c82.js');
require('../../_chunks/dep-abc005aa.js');
require('../../tooltip/utils/index.js');
require('../../_chunks/dep-3a800699.js');
require('../base-table-props.js');
require('./useLazyLoad.js');
require('../../_chunks/dep-12e6ec24.js');
require('./useRowspanAndColspan.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function useTreeData(props, context) {
  var _toRefs = Vue.toRefs(props),
    data = _toRefs.data,
    columns = _toRefs.columns;
  var _useConfig = configProvider_hooks_useConfig.useConfig("table", props.locale),
    t = _useConfig.t,
    globalConfig = _useConfig.globalConfig;
  var _useGlobalIcon = index.useGlobalIcon({
      AddRectangleIcon: tdesignIconsVueNext.AddRectangleIcon,
      MinusRectangleIcon: tdesignIconsVueNext.MinusRectangleIcon
    }),
    AddRectangleIcon = _useGlobalIcon.AddRectangleIcon,
    MinusRectangleIcon = _useGlobalIcon.MinusRectangleIcon;
  var store = Vue.ref(new table_hooks_useTreeDataExpand.TableTreeStore());
  var treeNodeCol = Vue.shallowRef();
  var dataSource = Vue.ref([]);
  var _useClassName = table_hooks_useClassName["default"](),
    tableTreeClasses = _useClassName.tableTreeClasses;
  var renderTNode = index$1.useTNodeDefault();
  var rowDataKeys = Vue.computed(function () {
    var _props$tree;
    return {
      rowKey: props.rowKey || "id",
      childrenKey: ((_props$tree = props.tree) === null || _props$tree === void 0 ? void 0 : _props$tree.childrenKey) || "children"
    };
  });
  var _useTreeDataExpand = table_hooks_useTreeDataExpand.useTreeDataExpand(props, {
      store: store,
      dataSource: dataSource,
      rowDataKeys: rowDataKeys
    }),
    tExpandedTreeNode = _useTreeDataExpand.tExpandedTreeNode,
    isDefaultExpandAllExecute = _useTreeDataExpand.isDefaultExpandAllExecute,
    isDefaultExpandedTreeNodesExecute = _useTreeDataExpand.isDefaultExpandedTreeNodesExecute,
    expandAll = _useTreeDataExpand.expandAll,
    foldAll = _useTreeDataExpand.foldAll,
    updateExpandOnDataChange = _useTreeDataExpand.updateExpandOnDataChange,
    onExpandFoldIconClick = _useTreeDataExpand.onExpandFoldIconClick;
  var checkedColumn = Vue.computed(function () {
    return columns.value.find(function (col) {
      return col.colKey === "row-select";
    });
  });
  Vue.watch(checkedColumn, function (column) {
    if (!store.value) return;
    store.value.updateDisabledState(dataSource.value, column, rowDataKeys.value);
  });
  var foldIcon = function foldIcon(context2) {
    var params = _objectSpread(_objectSpread({}, context2), {}, {
      type: "fold"
    });
    var defaultFoldIcon = t(globalConfig.value.treeExpandAndFoldIcon, Vue.h, params) || Vue.createVNode(MinusRectangleIcon, null, null);
    return renderTNode("treeExpandAndFoldIcon", {
      defaultNode: defaultFoldIcon,
      params: params
    });
  };
  var expandIcon = function expandIcon(context2) {
    var params = _objectSpread(_objectSpread({}, context2), {}, {
      type: "expand"
    });
    var defaultExpandIcon = t(globalConfig.value.treeExpandAndFoldIcon, Vue.h, params) || Vue.createVNode(AddRectangleIcon, null, null);
    return renderTNode("treeExpandAndFoldIcon", {
      defaultNode: defaultExpandIcon,
      params: params
    });
  };
  Vue.watch([data], function () {
    if (props.tree) {
      resetData(data.value);
    } else {
      dataSource.value = data.value;
    }
  }, {
    immediate: true
  });
  Vue.onUnmounted(function () {
    var _store$value$treeData;
    if (!props.tree) return;
    (_store$value$treeData = store.value.treeDataMap) === null || _store$value$treeData === void 0 || _store$value$treeData.clear();
    store.value = null;
  });
  Vue.watch(function () {
    var _props$tree2;
    return [columns.value, (_props$tree2 = props.tree) === null || _props$tree2 === void 0 ? void 0 : _props$tree2.treeNodeColumnIndex];
  }, function () {
    treeNodeCol.value = getTreeNodeColumnCol();
  }, {
    immediate: true
  });
  function resetData(data2) {
    var _tExpandedTreeNode$va;
    var columns2 = props.columns,
      expandedTreeNodes = props.expandedTreeNodes,
      defaultExpandedTreeNodes = props.defaultExpandedTreeNodes,
      tree = props.tree;
    store.value.initialTreeStore(data2, columns2, rowDataKeys.value);
    var defaultNeedExpand = Boolean(!isDefaultExpandedTreeNodesExecute.value && (defaultExpandedTreeNodes === null || defaultExpandedTreeNodes === void 0 ? void 0 : defaultExpandedTreeNodes.length));
    var needExpandAll = Boolean((tree === null || tree === void 0 ? void 0 : tree.defaultExpandAll) && !isDefaultExpandAllExecute.value);
    if ((_tExpandedTreeNode$va = tExpandedTreeNode.value) !== null && _tExpandedTreeNode$va !== void 0 && _tExpandedTreeNode$va.length && !!(expandedTreeNodes || defaultNeedExpand) || needExpandAll) {
      updateExpandOnDataChange(data2);
      isDefaultExpandedTreeNodesExecute.value = true;
    } else {
      dataSource.value = _toConsumableArray__default["default"](data2);
    }
  }
  function getTreeNodeStyle(level) {
    var _props$tree3, _props$tree4;
    if (level === void 0) return;
    var indent = ((_props$tree3 = props.tree) === null || _props$tree3 === void 0 ? void 0 : _props$tree3.indent) === void 0 ? 24 : (_props$tree4 = props.tree) === null || _props$tree4 === void 0 ? void 0 : _props$tree4.indent;
    return indent ? {
      paddingLeft: "".concat(level * indent || 1, "px")
    } : {};
  }
  function toggleExpandData(p, trigger) {
    var _store$value, _props$onTreeExpandCh;
    var currentData = _objectSpread({}, p);
    if (p.row.VIRTUAL_SCROLL_INDEX !== void 0) {
      currentData.rowIndex = p.row.VIRTUAL_SCROLL_INDEX;
    }
    dataSource.value = _toConsumableArray__default["default"](store.value.toggleExpandData(currentData, dataSource.value, rowDataKeys.value));
    var rowValue = get.get(p.row, rowDataKeys.value.rowKey);
    var rowState = (_store$value = store.value) === null || _store$value === void 0 || (_store$value = _store$value.treeDataMap) === null || _store$value === void 0 ? void 0 : _store$value.get(rowValue);
    (_props$onTreeExpandCh = props.onTreeExpandChange) === null || _props$onTreeExpandCh === void 0 || _props$onTreeExpandCh.call(props, {
      row: p.row,
      rowIndex: p.rowIndex,
      rowState: rowState,
      trigger: trigger
    });
  }
  function getTreeNodeColumnCol() {
    var _props$tree5;
    var columns2 = props.columns;
    var treeNodeColumnIndex = ((_props$tree5 = props.tree) === null || _props$tree5 === void 0 ? void 0 : _props$tree5.treeNodeColumnIndex) || 0;
    while ((_columns2$treeNodeCol = columns2[treeNodeColumnIndex]) !== null && _columns2$treeNodeCol !== void 0 && _columns2$treeNodeCol.type || ((_columns2$treeNodeCol2 = columns2[treeNodeColumnIndex]) === null || _columns2$treeNodeCol2 === void 0 ? void 0 : _columns2$treeNodeCol2.colKey) === "__EXPAND_ROW_ICON_COLUMN__") {
      var _columns2$treeNodeCol, _columns2$treeNodeCol2;
      treeNodeColumnIndex += 1;
    }
    return columns2[treeNodeColumnIndex];
  }
  function formatTreeColumn(col) {
    if (!props.tree || col.colKey !== treeNodeCol.value.colKey) return col;
    var newCol = _objectSpread({}, treeNodeCol.value);
    newCol.cell = function (h2, p) {
      var cellInfo = table_components_tr.renderCell(_objectSpread(_objectSpread({}, p), {}, {
        col: _objectSpread({}, treeNodeCol.value)
      }), context.slots, {
        cellEmptyContent: props.cellEmptyContent
      });
      var currentState = store.value.treeDataMap.get(get.get(p.row, rowDataKeys.value.rowKey));
      var colStyle = getTreeNodeStyle(currentState === null || currentState === void 0 ? void 0 : currentState.level);
      var classes = _defineProperty__default["default"]({}, tableTreeClasses.inlineCol, !!col.ellipsis);
      var childrenNodes = get.get(p.row, rowDataKeys.value.childrenKey);
      if (childrenNodes && childrenNodes instanceof Array || childrenNodes === true) {
        var _store$value$treeData2;
        var iconNode = (_store$value$treeData2 = store.value.treeDataMap.get(get.get(p.row, rowDataKeys.value.rowKey))) !== null && _store$value$treeData2 !== void 0 && _store$value$treeData2.expanded ? foldIcon(p) : expandIcon(p);
        return Vue.createVNode("div", {
          "class": [tableTreeClasses.col, classes],
          "style": colStyle
        }, [!!(childrenNodes.length || childrenNodes === true) && Vue.createVNode("span", {
          "class": tableTreeClasses.icon,
          "onClick": function onClick(e) {
            onExpandFoldIconClick(p, "expand-fold-icon");
            e.stopPropagation();
          }
        }, [iconNode]), cellInfo]);
      }
      return Vue.createVNode("div", {
        "style": colStyle,
        "class": [classes, tableTreeClasses.leafNode]
      }, [Vue.createVNode("span", {
        "class": tableTreeClasses.icon
      }, null), cellInfo]);
    };
    if (newCol.ellipsis === true) {
      newCol.ellipsis = function (h2, p) {
        return table_components_tr.renderCell(_objectSpread(_objectSpread({}, p), {}, {
          col: _objectSpread({}, treeNodeCol.value)
        }), context.slots);
      };
    }
    return newCol;
  }
  function setData(key, newRowData) {
    var rowIndex = store.value.updateData(key, newRowData, dataSource.value, rowDataKeys.value);
    var newData = _toConsumableArray__default["default"](dataSource.value);
    newData[rowIndex] = newRowData;
    dataSource.value = newData;
  }
  function getData(key) {
    return store.value.getData(key);
  }
  function remove(key) {
    dataSource.value = _toConsumableArray__default["default"](store.value.remove(key, dataSource.value, rowDataKeys.value));
  }
  function removeChildren(key) {
    dataSource.value = _toConsumableArray__default["default"](store.value.removeChildren(key, dataSource.value, rowDataKeys.value));
  }
  function appendTo() {
    var key = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "";
    var newData = arguments.length > 1 ? arguments[1] : undefined;
    if (!key) {
      dataSource.value = store.value.appendToRoot(newData, dataSource.value, rowDataKeys.value);
      return;
    }
    dataSource.value = _toConsumableArray__default["default"](store.value.appendTo(key, newData, dataSource.value, rowDataKeys.value));
  }
  function insertAfter(rowValue, newData) {
    dataSource.value = _toConsumableArray__default["default"](store.value.insertAfter(rowValue, newData, dataSource.value, rowDataKeys.value));
  }
  function insertBefore(rowValue, newData) {
    dataSource.value = _toConsumableArray__default["default"](store.value.insertBefore(rowValue, newData, dataSource.value, rowDataKeys.value));
  }
  function swapData(params) {
    var r = store.value.swapData(dataSource.value, params, rowDataKeys.value);
    if (r.result) {
      dataSource.value = _toConsumableArray__default["default"](r.dataSource);
    } else {
      var _props$onAbnormalDrag;
      var params2 = {
        code: r.code,
        reason: r.reason
      };
      (_props$onAbnormalDrag = props.onAbnormalDragSort) === null || _props$onAbnormalDrag === void 0 || _props$onAbnormalDrag.call(props, params2);
    }
  }
  function getTreeNode() {
    return store.value.getTreeNode(dataSource.value, rowDataKeys.value);
  }
  function getTreeExpandedRow() {
    var type = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "data";
    return store.value.getTreeExpandedRow(dataSource.value, rowDataKeys.value, type);
  }
  return {
    store: store,
    rowDataKeys: rowDataKeys,
    dataSource: dataSource,
    swapData: swapData,
    setData: setData,
    getData: getData,
    remove: remove,
    removeChildren: removeChildren,
    appendTo: appendTo,
    insertAfter: insertAfter,
    insertBefore: insertBefore,
    formatTreeColumn: formatTreeColumn,
    toggleExpandData: toggleExpandData,
    expandAll: expandAll,
    foldAll: foldAll,
    getTreeNode: getTreeNode,
    resetData: resetData,
    getTreeExpandedRow: getTreeExpandedRow,
    onExpandFoldIconClick: onExpandFoldIconClick
  };
}

exports["default"] = useTreeData;
