// tdesign-vue-next/cjs/tree/hooks/useTreeItem.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-449273df.js');
var index = require('../../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-497fc9a5.js');
var index$1 = require('../../_chunks/dep-387cd58a.js');
require('tdesign-icons-vue-next');
require('../../checkbox/index.js');
require('../../loading/index.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/regenerator');
var tree_hooks_useItemEvents = require('./useItemEvents.js');
var tree_hooks_useRenderIcon = require('./useRenderIcon.js');
var tree_hooks_useRenderLabel = require('./useRenderLabel.js');
var tree_hooks_useRenderLine = require('./useRenderLine.js');
var tree_hooks_useRenderOperations = require('./useRenderOperations.js');
var tree_hooks_useDraggable = require('./useDraggable.js');
require('../../_chunks/dep-f9b59444.js');
require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-80827572.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-6e64ef0e.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-6e7a2100.js');
require('../../_chunks/dep-12e6ec24.js');
require('../../_chunks/dep-51846987.js');
require('../../_chunks/dep-be760bc2.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../_chunks/dep-c0cb4d13.js');
require('../../_chunks/dep-9dfd682d.js');
require('../../_chunks/dep-2db31625.js');
require('../../_chunks/dep-0a17438d.js');
require('../../_chunks/dep-5090b515.js');
require('../../_chunks/dep-39313d05.js');
require('../../_chunks/dep-f9bc7c82.js');
require('../../_chunks/dep-1c6213b5.js');
require('../../_chunks/dep-cfdee5e7.js');
require('../../_chunks/dep-9ada3ec1.js');
require('../../_chunks/dep-a2e46896.js');
require('../../_chunks/dep-6512c9d2.js');
require('../../_chunks/dep-f02d9258.js');
require('../../_chunks/dep-3cd26986.js');
require('../../_chunks/dep-e3eb5659.js');
require('../../_chunks/dep-d0227fbe.js');
require('../../checkbox/constants/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../_chunks/dep-2e8277b1.js');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../_chunks/dep-cec24bbf.js');
require('../../_chunks/dep-f786e1df.js');
require('../../_chunks/dep-bd59ca66.js');
require('../../_chunks/dep-618d56e7.js');
require('../../_chunks/dep-2cf25d25.js');
require('../../_chunks/dep-abc005aa.js');
require('../../_chunks/dep-203a5570.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-3231adce.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6130f266.js');
require('../../_chunks/dep-3ffe18b0.js');
require('../../_chunks/dep-e41b3434.js');
require('../../loading/props.js');
require('../../_chunks/dep-e46285c0.js');
require('../../_chunks/dep-54801354.js');
require('../../_chunks/dep-f7508ffa.js');
require('../../_chunks/dep-470f211b.js');
require('../../_chunks/dep-9b549c85.js');
require('../../_chunks/dep-e25ae470.js');
require('../../_chunks/dep-bed84553.js');
require('../../_chunks/dep-a4b963ab.js');
require('../../_chunks/dep-88ae5d8a.js');
require('../utils/index.js');
require('../../_chunks/dep-801a06fe.js');
require('../../_chunks/dep-f3377589.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function useTreeItem(state) {
  var treeScope = state.treeScope,
    treeItemRef = state.treeItemRef;
  var virtualConfig = treeScope.virtualConfig,
    treeContentRef = treeScope.treeContentRef,
    scrollProps = treeScope.scrollProps;
  var classPrefix = index.usePrefixClass().value;
  var componentName = index.usePrefixClass("tree").value;
  var _useItemEvents = tree_hooks_useItemEvents["default"](state),
    handleClick = _useItemEvents.handleClick;
  var _useRenderIcon = tree_hooks_useRenderIcon["default"](state),
    renderIcon = _useRenderIcon.renderIcon;
  var _useRenderLabel = tree_hooks_useRenderLabel["default"](state),
    renderLabel = _useRenderLabel.renderLabel;
  var _useRenderLine = tree_hooks_useRenderLine["default"](state),
    renderLine = _useRenderLine.renderLine;
  var _useRenderOperations = tree_hooks_useRenderOperations["default"](state),
    renderOperations = _useRenderOperations.renderOperations;
  var _useDraggable = tree_hooks_useDraggable["default"](state),
    dragStates = _useDraggable.dragStates,
    handleDragStart = _useDraggable.handleDragStart,
    handleDragEnd = _useDraggable.handleDragEnd,
    handleDragOver = _useDraggable.handleDragOver,
    handleDragLeave = _useDraggable.handleDragLeave,
    handleDrop = _useDraggable.handleDrop;
  var _useLazyLoad = index$1.useLazyLoad(treeContentRef, treeItemRef, Vue.reactive(_objectSpread({}, scrollProps === null || scrollProps === void 0 ? void 0 : scrollProps.value))),
    hasLazyLoadHolder = _useLazyLoad.hasLazyLoadHolder,
    tRowHeight = _useLazyLoad.tRowHeight;
  function tryNotifyVirtualScrollRowUpdate() {
    var node = state.node;
    var isVirtual = virtualConfig === null || virtualConfig === void 0 ? void 0 : virtualConfig.isVirtualScroll.value;
    if (isVirtual) {
      var timer = setTimeout(function () {
        virtualConfig.handleRowMounted({
          ref: treeItemRef,
          data: node
        });
        clearTimeout(timer);
      }, 100);
    }
  }
  Vue.onMounted(function () {
    tryNotifyVirtualScrollRowUpdate();
  });
  Vue.onUpdated(function () {
    tryNotifyVirtualScrollRowUpdate();
  });
  var getItemStyles = function getItemStyles() {
    var node = state.node;
    var level = node.level;
    var levelStyle = "--level: ".concat(level, ";");
    var strStyle = "".concat(levelStyle);
    return strStyle;
  };
  var getItemClassList = function getItemClassList() {
    var node = state.node;
    var isDragOver = dragStates.isDragOver,
      isDragging = dragStates.isDragging,
      dropPosition = dragStates.dropPosition;
    var list = [];
    list.push("".concat(componentName, "__item"));
    list.push(_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(componentName, "__item--open"), node.expanded), "".concat(classPrefix, "-is-active"), node.isActivable() ? node.actived : false), "".concat(classPrefix, "-is-disabled"), node.isDisabled()));
    list.push(_defineProperty__default["default"]({}, "".concat(componentName, "__item--draggable"), !node.isDisabled() && node.isDraggable()));
    if (node.visible) {
      list.push("".concat(componentName, "__item--visible"));
    } else {
      list.push("".concat(componentName, "__item--hidden"));
    }
    if (node.vmIsLocked) {
      list.push("".concat(componentName, "__item--locked"));
    }
    if (node.vmIsRest) {
      list.push("".concat(componentName, "__item--matched"));
    }
    list.push(_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(componentName, "__item--dragging"), isDragging), "".concat(componentName, "__item--tip-top"), isDragOver && dropPosition < 0), "".concat(componentName, "__item--tip-bottom"), isDragOver && dropPosition > 0), "".concat(componentName, "__item--tip-highlight"), !isDragging && isDragOver && dropPosition === 0));
    return list;
  };
  var renderItem = function renderItem(h) {
    var itemNodes = [];
    var iconNode = renderIcon(h);
    var lineNode = renderLine(h);
    if (lineNode) {
      itemNodes.push(lineNode);
    }
    if (iconNode) {
      itemNodes.push(iconNode);
    }
    var labelNode = renderLabel(h);
    if (labelNode) {
      itemNodes.push(labelNode);
    }
    var opNode = renderOperations(h);
    if (opNode) {
      itemNodes.push(opNode);
    }
    return itemNodes;
  };
  var renderItemNode = function renderItemNode(h) {
    var node = state.node,
      props = state.props;
    if (!node) return null;
    var level = node.level,
      value = node.value;
    var styles = getItemStyles();
    var classList = getItemClassList();
    var treeState = state;
    treeState.stateId = props.stateId;
    var itemNode = Vue.createVNode("div", {
      "ref": "treeItemRef",
      "class": classList,
      "data-value": value,
      "data-level": level,
      "style": styles,
      "onClick": function onClick(evt) {
        return handleClick(evt);
      },
      "draggable": !node.isDisabled() && node.isDraggable(),
      "onDragstart": function onDragstart(evt) {
        return handleDragStart(evt);
      },
      "onDragend": function onDragend(evt) {
        return handleDragEnd(evt);
      },
      "onDragover": function onDragover(evt) {
        return handleDragOver(evt);
      },
      "onDragleave": function onDragleave(evt) {
        return handleDragLeave(evt);
      },
      "onDrop": function onDrop(evt) {
        return handleDrop(evt);
      }
    }, [hasLazyLoadHolder.value ? [Vue.createVNode("div", null, null)] : renderItem(h)]);
    return itemNode;
  };
  return {
    hasLazyLoadHolder: hasLazyLoadHolder,
    tRowHeight: tRowHeight,
    renderItemNode: renderItemNode
  };
}

exports["default"] = useTreeItem;
