// tdesign-vue-next/cjs/date-picker/components/panel/RangePanel.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-449273df.js');
var index = require('../../../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
var log = require('../../../_chunks/dep-b4528c21.js');
require('../../../_chunks/dep-497fc9a5.js');
var datePicker_components_panel_PanelContent = require('./PanelContent.js');
var datePicker_components_panel_ExtraContent = require('./ExtraContent.js');
var format = require('../../../_chunks/dep-d303c019.js');
var datePicker_hooks_useDisableDate = require('../../hooks/useDisableDate.js');
require('tdesign-icons-vue-next');
require('dayjs');
var datePicker_hooks_useTableData = require('../../hooks/useTableData.js');
require('../../../_chunks/dep-abacafd8.js');
var isArray = require('../../../_chunks/dep-959f4847.js');
var isFunction = require('../../../_chunks/dep-80827572.js');
var configProvider_hooks_useConfig = require('../../../config-provider/hooks/useConfig.js');
require('../../../_chunks/dep-f9b59444.js');
require('../../../_chunks/dep-31abb282.js');
require('../../../_chunks/dep-c71a9cd7.js');
require('../../../_chunks/dep-39529ef1.js');
require('../../../_chunks/dep-f4c1cb5b.js');
require('../../../_chunks/dep-4896d30f.js');
require('../../../_chunks/dep-2610df9f.js');
require('../../../_chunks/dep-2c7a67b8.js');
require('../../../_chunks/dep-db6459dd.js');
require('../../../_chunks/dep-4d7a3a91.js');
require('../../../_chunks/dep-6e64ef0e.js');
require('../../../_chunks/dep-ae380218.js');
require('../../../_chunks/dep-fdac7521.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../base/Header.js');
require('../../../pagination/index.js');
require('../../../pagination/pagination.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../../_chunks/dep-c0cb4d13.js');
require('../../../_chunks/dep-51846987.js');
require('../../../_chunks/dep-be760bc2.js');
require('../../../_chunks/dep-9dfd682d.js');
require('../../../_chunks/dep-2db31625.js');
require('../../../_chunks/dep-0a17438d.js');
require('../../../_chunks/dep-5090b515.js');
require('../../../_chunks/dep-39313d05.js');
require('../../../_chunks/dep-f9bc7c82.js');
require('../../../_chunks/dep-1c6213b5.js');
require('../../../_chunks/dep-cfdee5e7.js');
require('../../../_chunks/dep-4b52068a.js');
require('../../../_chunks/dep-88ae5d8a.js');
require('../../../_chunks/dep-d0227fbe.js');
require('../../../input-number/index.js');
require('../../../input-number/input-number.js');
require('../../../button/index.js');
require('../../../button/button.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-203a5570.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-3231adce.js');
require('../../../loading/icon/gradient.js');
require('../../../_chunks/dep-e3eb5659.js');
require('../../../_chunks/dep-cce7af6a.js');
require('../../../_chunks/dep-6130f266.js');
require('../../../_chunks/dep-3ffe18b0.js');
require('../../../_chunks/dep-df5442c5.js');
require('../../../_chunks/dep-e41b3434.js');
require('../../../_chunks/dep-9ada3ec1.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-141cbae1.js');
require('../../../_chunks/dep-c277c7cb.js');
require('../../../_chunks/dep-7653f80f.js');
require('../../../_chunks/dep-6e7a2100.js');
require('../../../_chunks/dep-e46285c0.js');
require('../../../_chunks/dep-54801354.js');
require('../../../_chunks/dep-bd59ca66.js');
require('../../../_chunks/dep-f7508ffa.js');
require('../../../_chunks/dep-470f211b.js');
require('../../../_chunks/dep-9b549c85.js');
require('../../../_chunks/dep-e25ae470.js');
require('../../../_chunks/dep-bed84553.js');
require('../../../_chunks/dep-a4b963ab.js');
require('../../../_chunks/dep-abc005aa.js');
require('../../../button/props.js');
require('../../../_chunks/dep-a2e46896.js');
require('../../../_chunks/dep-6512c9d2.js');
require('../../../_chunks/dep-3cd26986.js');
require('../../../input/index.js');
require('../../../input/input.js');
require('../../../input/props.js');
require('../../../_chunks/dep-f02d9258.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../_chunks/dep-d20c4e5e.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../_chunks/dep-ba2511b0.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../input-number/props.js');
require('../../../_chunks/dep-5ea73cce.js');
require('../../../input-number/hooks/useInputNumber.js');
require('../../../_chunks/dep-74a0ebc2.js');
require('../../../_chunks/dep-94323f13.js');
require('../../../_chunks/dep-75d3928f.js');
require('../../../select/index.js');
require('../../../select/select.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../../common-components/fake-arrow.js');
require('../../../select-input/index.js');
require('../../../select-input/select-input.js');
require('../../../popup/index.js');
require('../../../popup/popup.js');
require('@popperjs/core');
require('../../../popup/container.js');
require('../../../popup/props.js');
require('../../../_chunks/dep-4e2a4791.js');
require('../../../_chunks/dep-f3377589.js');
require('../../../_chunks/dep-71d9daf0.js');
require('../../../select-input/props.js');
require('../../../select-input/hooks/useMultiple.js');
require('../../../tag-input/index.js');
require('../../../tag-input/tag-input.js');
require('../../../tag-input/props.js');
require('../../../tag-input/hooks/useDragSorter.js');
require('../../../tag-input/hooks/useHover.js');
require('../../../tag-input/hooks/useTagScroll.js');
require('../../../tag-input/hooks/useTagList.js');
require('../../../tag/index.js');
require('../../../tag/tag.js');
require('tinycolor2');
require('../../../tag/props.js');
require('../../../tag/check-tag.js');
require('../../../tag/check-tag-props.js');
require('../../../_chunks/dep-2e8277b1.js');
require('../../../tag/check-tag-group.js');
require('../../../tag/check-tag-group-props.js');
require('../../../select-input/hooks/useOverlayInnerStyle.js');
require('../../../select-input/hooks/useSingle.js');
require('../../../_chunks/dep-008d4a55.js');
require('../../../_chunks/dep-0e1751ee.js');
require('../../../_chunks/dep-4e2907e0.js');
require('../../../select/components/select-panel.js');
require('../../../select/option.js');
require('../../../select/option-props.js');
require('../../../checkbox/index.js');
require('../../../checkbox/checkbox.js');
require('../../../checkbox/props.js');
require('../../../checkbox/constants/index.js');
require('../../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../../_chunks/dep-12e6ec24.js');
require('../../../checkbox/hooks/useKeyboardEvent.js');
require('../../../checkbox/group.js');
require('../../../checkbox/checkbox-group-props.js');
require('../../../_chunks/dep-cec24bbf.js');
require('../../../_chunks/dep-f786e1df.js');
require('../../../_chunks/dep-618d56e7.js');
require('../../../_chunks/dep-2cf25d25.js');
require('../../../select/utils/index.js');
require('../../../select/constants/index.js');
require('../../../select/option-group.js');
require('../../../select/option-group-props.js');
require('../../../select/props.js');
require('../../../select/hooks/usePanelVirtualScroll.js');
require('../../../_chunks/dep-3a800699.js');
require('../../../_chunks/dep-648bfb84.js');
require('../../../select/hooks/useKeyboardControl.js');
require('../../../select/hooks/useSelectOptions.js');
require('../../../_chunks/dep-8f2069e9.js');
require('../../../input-adornment/index.js');
require('../../../input-adornment/input-adornment.js');
require('../../../input-adornment/props.js');
require('../../../pagination/props.js');
require('../../../pagination/hooks/useMoreAction.js');
require('../../../pagination/hooks/usePaginationClasses.js');
require('../../../pagination/pagination-mini.js');
require('../../../pagination/pagination-mini-props.js');
require('../../../tooltip/index.js');
require('../../../tooltip/tooltip.js');
require('../../../tooltip/props.js');
require('../../../tooltip/utils/index.js');
require('../../hooks/useSelectRange.js');
require('../base/Table.js');
require('../base/Cell.js');
require('../../../time-picker/index.js');
require('../../../time-picker/time-picker.js');
require('../../../_chunks/dep-f12e600d.js');
require('../../../time-picker/panel/time-picker-panel.js');
require('../../../_chunks/dep-d4a90e1c.js');
require('../../../time-picker/props.js');
require('../../../_chunks/dep-7183635a.js');
require('../../../_chunks/dep-2a84fc5a.js');
require('../../../time-picker/time-range-picker.js');
require('../../../range-input/index.js');
require('../../../range-input/range-input.js');
require('../../../range-input/props.js');
require('../../../range-input/range-input-popup.js');
require('../../../range-input/range-input-popup-props.js');
require('../../../time-picker/time-range-picker-props.js');
require('../../utils/index.js');
require('../../../_chunks/dep-59ce9497.js');
require('../base/Footer.js');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');
require('../../../_chunks/dep-ed620b03.js');
require('../../../_chunks/dep-64b2629b.js');
require('../../../_chunks/dep-ceabf07b.js');
require('../../../_chunks/dep-e600142f.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TRangePanel = Vue.defineComponent({
  name: "TRangePanel",
  props: {
    hoverValue: Array,
    range: [Array, Function],
    activeIndex: Number,
    isFirstValueSelected: Boolean,
    disabled: {
      type: [Boolean, Array]
    },
    disableDate: [Object, Array, Function],
    disableTime: Function,
    mode: {
      type: String,
      "default": "date"
    },
    format: String,
    presetsPlacement: {
      type: String,
      "default": "bottom"
    },
    value: Array,
    timePickerProps: Object,
    presets: Object,
    popupVisible: Boolean,
    enableTimePicker: Boolean,
    panelPreselection: Boolean,
    firstDayOfWeek: Number,
    year: Array,
    month: Array,
    time: Array,
    cancelRangeSelectLimit: Boolean,
    defaultTime: Array,
    onClick: Function,
    onCellClick: Function,
    onCellMouseEnter: Function,
    onCellMouseLeave: Function,
    onJumperClick: Function,
    onConfirmClick: Function,
    onPresetClick: Function,
    onYearChange: Function,
    onMonthChange: Function,
    onTimePickerChange: Function,
    needConfirm: Boolean,
    cell: {
      type: Function
    }
  },
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var COMPONENT_NAME = index.usePrefixClass("date-range-picker__panel");
    var _useConfig = configProvider_hooks_useConfig.useConfig("datePicker"),
      globalConfig = _useConfig.globalConfig;
    var format$1 = Vue.computed(function () {
      var _getDefaultFormat;
      return (_getDefaultFormat = format.getDefaultFormat({
        mode: props.mode,
        format: props.format,
        enableTimePicker: props.enableTimePicker
      })) === null || _getDefaultFormat === void 0 ? void 0 : _getDefaultFormat.format;
    });
    var hidePreselection = !props.panelPreselection && props.value.length === 2;
    var disableDateOptions = Vue.computed(function () {
      var startDateValue = new Date(format.parseToDayjs(props.value[0], format$1.value, "start").toDate().setHours(0, 0, 0));
      var endDateValue = new Date(format.parseToDayjs(props.value[1], format$1.value, "end").toDate().setHours(23, 59, 59));
      var start = props.isFirstValueSelected && props.activeIndex === 1 ? startDateValue : void 0;
      var end = props.isFirstValueSelected && props.activeIndex === 0 ? endDateValue : void 0;
      if (props.disabled && isArray.isArray(props.disabled)) {
        if (props.disabled[0]) start = startDateValue;else if (props.disabled[1]) end = endDateValue;
      }
      return datePicker_hooks_useDisableDate.useDisableDate({
        format: format$1.value,
        mode: props.mode,
        disableDate: props.disableDate,
        start: start,
        end: end
      });
    });
    var rangeValue = Vue.computed(function () {
      var startRange = props.range;
      var endRange = props.range;
      if (isArray.isArray(props.range)) {
        if (props.range.length !== 2) {
          log.log.warn("DateRangePicker", "`range` length must be 2 when `range` is an array.");
        }
        var first = props.range[0];
        var second = props.range[1];
        if ((isArray.isArray(first) || isFunction.isFunction(first)) && (isArray.isArray(second) || isFunction.isFunction(second))) {
          startRange = first;
          endRange = second;
        }
      }
      return {
        start: startRange,
        end: endRange
      };
    });
    var startTableData = Vue.computed(function () {
      var disableDate = isFunction.isFunction(props.disableDate) ? props.disableDate({
        partial: "start",
        value: props.value[0]
      }) : disableDateOptions.value.disableDate;
      return datePicker_hooks_useTableData.useTableData(_objectSpread(_objectSpread({
        isRange: true,
        start: props.value[0] ? format.parseToDayjs(props.value[0], format$1.value).toDate() : void 0,
        end: props.value[1] ? format.parseToDayjs(props.value[1], format$1.value).toDate() : void 0,
        hoverStart: !hidePreselection && props.hoverValue[0] ? format.parseToDayjs(props.hoverValue[0], format$1.value).toDate() : void 0,
        hoverEnd: !hidePreselection && props.hoverValue[1] ? format.parseToDayjs(props.hoverValue[1], format$1.value).toDate() : void 0,
        year: props.year[0],
        month: props.month[0],
        mode: props.mode,
        firstDayOfWeek: props.firstDayOfWeek || globalConfig.value.firstDayOfWeek,
        range: rangeValue.value.start
      }, disableDateOptions.value), {}, {
        disableDate: disableDate,
        cancelRangeSelectLimit: props.cancelRangeSelectLimit
      }));
    });
    var endTableData = Vue.computed(function () {
      var disableDate = isFunction.isFunction(props.disableDate) ? props.disableDate({
        partial: "end",
        value: props.value
      }) : disableDateOptions.value.disableDate;
      return datePicker_hooks_useTableData.useTableData(_objectSpread(_objectSpread({
        isRange: true,
        start: props.value[0] ? format.parseToDayjs(props.value[0], format$1.value).toDate() : void 0,
        end: props.value[1] ? format.parseToDayjs(props.value[1], format$1.value).toDate() : void 0,
        hoverStart: !hidePreselection && props.hoverValue[0] ? format.parseToDayjs(props.hoverValue[0], format$1.value).toDate() : void 0,
        hoverEnd: !hidePreselection && props.hoverValue[1] ? format.parseToDayjs(props.hoverValue[1], format$1.value).toDate() : void 0,
        year: props.mode === "year" && props.year[1] - props.year[0] <= 9 ? props.year[1] + 9 : props.year[1],
        month: props.month[1],
        mode: props.mode,
        firstDayOfWeek: props.firstDayOfWeek || globalConfig.value.firstDayOfWeek,
        range: rangeValue.value.end
      }, disableDateOptions.value), {}, {
        disableDate: disableDate,
        cancelRangeSelectLimit: props.cancelRangeSelectLimit
      }));
    });
    var panelContentProps = Vue.computed(function () {
      return {
        format: format$1.value,
        mode: props.mode,
        firstDayOfWeek: props.firstDayOfWeek || globalConfig.value.firstDayOfWeek,
        internalYear: props.year,
        popupVisible: props.popupVisible,
        enableTimePicker: props.enableTimePicker,
        timePickerProps: props.timePickerProps,
        onMonthChange: props.onMonthChange,
        onYearChange: props.onYearChange,
        onJumperClick: props.onJumperClick,
        onCellClick: props.onCellClick,
        onCellMouseEnter: props.onCellMouseEnter,
        onCellMouseLeave: props.onCellMouseLeave,
        onTimePickerChange: props.onTimePickerChange,
        disableTime: props.disableTime,
        defaultTime: props.defaultTime,
        cell: props.cell
      };
    });
    return function () {
      return Vue.createVNode("div", {
        "class": [COMPONENT_NAME.value, _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--direction-row"), ["left", "right"].includes(props.presetsPlacement))],
        "onClick": function onClick(e) {
          var _props$onClick;
          return (_props$onClick = props.onClick) === null || _props$onClick === void 0 ? void 0 : _props$onClick.call(props, {
            e: e
          });
        }
      }, [["top", "left"].includes(props.presetsPlacement) ? Vue.createVNode(datePicker_components_panel_ExtraContent["default"], {
        "presets": props.presets,
        "selectedValue": props.value[props.activeIndex],
        "enableTimePicker": props.enableTimePicker,
        "onPresetClick": props.onPresetClick,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "needConfirm": props.needConfirm
      }, {
        presets: slots.presets
      }) : null, Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "-content-wrapper")
      }, [!props.enableTimePicker ? [Vue.createVNode(datePicker_components_panel_PanelContent["default"], Vue.mergeProps({
        "key": "startPanel",
        "partial": "start",
        "year": props.year[0],
        "month": props.month[0],
        "time": props.time[props.activeIndex],
        "value": props.value,
        "tableData": startTableData.value,
        "range": rangeValue.value.start
      }, panelContentProps.value), slots), Vue.createVNode(datePicker_components_panel_PanelContent["default"], Vue.mergeProps({
        "key": "endPanel",
        "partial": "end",
        "year": props.year[1],
        "month": props.month[1],
        "time": props.time[props.activeIndex],
        "value": props.value,
        "tableData": endTableData.value,
        "range": rangeValue.value.end
      }, panelContentProps.value), slots)] : Vue.createVNode(datePicker_components_panel_PanelContent["default"], Vue.mergeProps({
        "key": "start",
        "partial": props.activeIndex ? "end" : "start",
        "year": props.activeIndex ? props.year[1] : props.year[0],
        "month": props.activeIndex ? props.month[1] : props.month[0],
        "time": props.activeIndex ? props.time[1] : props.time[0],
        "value": props.value,
        "tableData": props.activeIndex ? endTableData.value : startTableData.value,
        "range": props.activeIndex ? rangeValue.value.end : rangeValue.value.start
      }, panelContentProps.value), null)]), ["bottom", "right"].includes(props.presetsPlacement) ? Vue.createVNode(datePicker_components_panel_ExtraContent["default"], {
        "presets": props.presets,
        "selectedValue": props.value[props.activeIndex],
        "enableTimePicker": props.enableTimePicker,
        "onPresetClick": props.onPresetClick,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "needConfirm": props.needConfirm
      }, {
        presets: slots.presets
      }) : null]);
    };
  }
});

exports["default"] = TRangePanel;
