// tdesign-vue-next/cjs/cascader/cascader.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var cascader_components_Panel = require('./components/Panel.js');
var selectInput_index = require('../select-input/index.js');
var commonComponents_fakeArrow = require('../common-components/fake-arrow.js');
var cascader_props = require('./props.js');
var cascader_utils_className = require('./utils/className.js');
var cascader_utils_effect = require('./utils/effect.js');
var cascader_utils_helper = require('./utils/helper.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-449273df.js');
var index$4 = require('../_chunks/dep-c0cb4d13.js');
var index$1 = require('../_chunks/dep-9ada3ec1.js');
var index = require('../_chunks/dep-95cb1381.js');
var index$2 = require('../_chunks/dep-a2e46896.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-497fc9a5.js');
var index$3 = require('../_chunks/dep-f02d9258.js');
var index$5 = require('../_chunks/dep-5ea73cce.js');
var cascader_hooks_index = require('./hooks/index.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('./components/Item.js');
require('tdesign-icons-vue-next');
require('../checkbox/index.js');
require('../checkbox/checkbox.js');
require('../checkbox/props.js');
require('../_chunks/dep-3cd26986.js');
require('../_chunks/dep-e3eb5659.js');
require('../_chunks/dep-d0227fbe.js');
require('../_chunks/dep-cfdee5e7.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-6e64ef0e.js');
require('../checkbox/constants/index.js');
require('../checkbox/hooks/useCheckboxLazyLoad.js');
require('../_chunks/dep-12e6ec24.js');
require('../checkbox/hooks/useKeyboardEvent.js');
require('../_chunks/dep-2e8277b1.js');
require('../_chunks/dep-be760bc2.js');
require('../checkbox/group.js');
require('../checkbox/checkbox-group-props.js');
require('../_chunks/dep-cec24bbf.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-f786e1df.js');
require('../_chunks/dep-bd59ca66.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-80827572.js');
require('../_chunks/dep-618d56e7.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-2cf25d25.js');
require('../_chunks/dep-abc005aa.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('../_chunks/dep-f9b59444.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('../_chunks/dep-6512c9d2.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../loading/index.js');
require('../_chunks/dep-203a5570.js');
require('../loading/plugin.js');
require('../_chunks/dep-3231adce.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-6130f266.js');
require('../_chunks/dep-3ffe18b0.js');
require('../_chunks/dep-e41b3434.js');
require('../loading/props.js');
require('../_chunks/dep-e46285c0.js');
require('../_chunks/dep-54801354.js');
require('../_chunks/dep-f7508ffa.js');
require('../_chunks/dep-470f211b.js');
require('../_chunks/dep-9b549c85.js');
require('../_chunks/dep-e25ae470.js');
require('../_chunks/dep-bed84553.js');
require('../_chunks/dep-a4b963ab.js');
require('../_chunks/dep-88ae5d8a.js');
require('../select-input/select-input.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../popup/container.js');
require('../popup/props.js');
require('../_chunks/dep-ba2511b0.js');
require('../_chunks/dep-4e2a4791.js');
require('../_chunks/dep-f3377589.js');
require('../_chunks/dep-71d9daf0.js');
require('../select-input/props.js');
require('../select-input/hooks/useMultiple.js');
require('../tag-input/index.js');
require('../tag-input/tag-input.js');
require('../input/index.js');
require('../input/input.js');
require('../input/props.js');
require('../input/hooks/useInput.js');
require('../form/constants/index.js');
require('../input/hooks/useLengthLimit.js');
require('../_chunks/dep-b4528c21.js');
require('../_chunks/dep-d20c4e5e.js');
require('../input/hooks/useInputEventHandler.js');
require('../input/hooks/useInputWidth.js');
require('../input/input-group.js');
require('../input/input-group-props.js');
require('../tag-input/props.js');
require('../_chunks/dep-4b52068a.js');
require('../tag-input/hooks/useDragSorter.js');
require('../tag-input/hooks/useHover.js');
require('../tag-input/hooks/useTagScroll.js');
require('../tag-input/hooks/useTagList.js');
require('../tag/index.js');
require('../tag/tag.js');
require('tinycolor2');
require('../tag/props.js');
require('../tag/check-tag.js');
require('../tag/check-tag-props.js');
require('../tag/check-tag-group.js');
require('../tag/check-tag-group-props.js');
require('../select-input/hooks/useOverlayInnerStyle.js');
require('../select-input/hooks/useSingle.js');
require('../_chunks/dep-008d4a55.js');
require('../_chunks/dep-0e1751ee.js');
require('../_chunks/dep-4e2907e0.js');
require('../_chunks/dep-a73f6bfc.js');
require('../_chunks/dep-2bee593d.js');
require('mitt');
require('../_chunks/dep-6f04645f.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../_chunks/dep-e129d089.js');
require('../_chunks/dep-ceabf07b.js');
require('../_chunks/dep-ed620b03.js');
require('../_chunks/dep-75d3928f.js');
require('../_chunks/dep-37eb0be5.js');
require('../_chunks/dep-e600142f.js');
require('../_chunks/dep-64b2629b.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var _Cascader = Vue.defineComponent({
  name: "TCascader",
  props: cascader_props["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var COMPONENT_NAME = index.usePrefixClass("cascader");
    var classPrefix = index.usePrefixClass();
    var _useCommonClassName = index$1.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS;
    var overlayClassName = index.usePrefixClass("cascader__popup");
    var _useConfig = configProvider_hooks_useConfig.useConfig("cascader"),
      globalConfig = _useConfig.globalConfig;
    var isDisabled = index$2.useDisabled();
    var isReadonly = index$3.useReadonly();
    var renderTNodeJSX = index$4.useTNodeJSX();
    var _useCascaderContext = cascader_hooks_index.useCascaderContext(props2),
      cascaderContext = _useCascaderContext.cascaderContext,
      innerValue = _useCascaderContext.innerValue,
      isFilterable = _useCascaderContext.isFilterable,
      getCascaderItems = _useCascaderContext.getCascaderItems;
    var displayValue = Vue.computed(function () {
      return props2.multiple ? cascader_utils_helper.getMultipleContent(cascaderContext.value) : cascader_utils_helper.getSingleContent(cascaderContext.value);
    });
    var panels = Vue.computed(function () {
      return cascader_utils_helper.getPanels(cascaderContext.value.treeNodes);
    });
    var inputPlaceholder = Vue.computed(function () {
      var _props2$placeholder;
      return cascaderContext.value.visible && !props2.multiple && cascader_utils_helper.getSingleContent(cascaderContext.value) || ((_props2$placeholder = props2.placeholder) !== null && _props2$placeholder !== void 0 ? _props2$placeholder : globalConfig.value.placeholder);
    });
    var renderSuffixIcon = function renderSuffixIcon() {
      if (props2.suffixIcon || slots.suffixIcon) {
        return renderTNodeJSX("suffixIcon");
      }
      var visible = cascaderContext.value.visible;
      return Vue.createVNode(commonComponents_fakeArrow["default"], {
        "overlayClassName": cascader_utils_className.getFakeArrowIconClass(classPrefix.value, STATUS.value, cascaderContext.value),
        "isActive": visible
      }, null);
    };
    var valueDisplayParams = Vue.computed(function () {
      var arrayValue = innerValue.value instanceof Array ? innerValue.value : [innerValue.value];
      var displayValue2 = props2.multiple && props2.minCollapsedNum ? arrayValue.slice(0, props2.minCollapsedNum) : innerValue.value;
      var options = getCascaderItems(arrayValue);
      return {
        value: innerValue.value,
        selectedOptions: options,
        onClose: function onClose(index) {
          cascader_utils_effect.handleRemoveTagEffect(cascaderContext.value, index, props2.onRemove);
        },
        displayValue: displayValue2
      };
    });
    var renderValueDisplay = function renderValueDisplay() {
      return renderTNodeJSX("valueDisplay", {
        params: valueDisplayParams.value
      });
    };
    var renderLabel = function renderLabel() {
      var label = renderTNodeJSX("label");
      if (props2.multiple) return label;
      if (!label) return null;
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-tag-input__prefix")
      }, [label]);
    };
    var cascaderClassNames = Vue.computed(function () {
      return [COMPONENT_NAME.value, props2.multiple ? "".concat(COMPONENT_NAME.value, "--multiple") : "".concat(COMPONENT_NAME.value, "--single")];
    });
    return function () {
      var _props2$popupProps;
      var _cascaderContext$valu = cascaderContext.value,
        setVisible = _cascaderContext$valu.setVisible,
        visible = _cascaderContext$valu.visible,
        inputVal = _cascaderContext$valu.inputVal,
        setInputVal = _cascaderContext$valu.setInputVal;
      var selectInputEvents = index$5.useEventForward(props2.selectInputProps, {
        onTagChange: function onTagChange(val, ctx) {
          if (ctx.trigger === "enter") return;
          cascader_utils_effect.handleRemoveTagEffect(cascaderContext.value, ctx.index, props2.onRemove);
        },
        onInputChange: function onInputChange(value) {
          if (!isFilterable.value) return;
          setInputVal("".concat(value));
        },
        onPopupVisibleChange: function onPopupVisibleChange(val, context) {
          if (isDisabled.value) return;
          setVisible(val, context);
        },
        onBlur: function onBlur(val, context) {
          var _props2$onBlur;
          (_props2$onBlur = props2.onBlur) === null || _props2$onBlur === void 0 || _props2$onBlur.call(props2, {
            value: cascaderContext.value.value,
            inputValue: context.inputValue || "",
            e: context.e
          });
        },
        onFocus: function onFocus(val, context) {
          var _props2$onFocus;
          (_props2$onFocus = props2.onFocus) === null || _props2$onFocus === void 0 || _props2$onFocus.call(props2, {
            value: cascaderContext.value.value,
            e: context.e
          });
        },
        onClear: function onClear() {
          cascader_utils_effect.closeIconClickEffect(cascaderContext.value);
        }
      });
      return Vue.createVNode(selectInput_index.SelectInput, Vue.mergeProps({
        "class": cascaderClassNames.value,
        "value": displayValue.value,
        "inputValue": visible ? inputVal : "",
        "popupVisible": visible,
        "keys": props2.keys,
        "allowInput": isFilterable.value,
        "min-collapsed-num": props2.minCollapsedNum,
        "collapsed-items": props2.collapsedItems,
        "readonly": isReadonly.value,
        "disabled": isDisabled.value,
        "clearable": props2.clearable,
        "placeholder": inputPlaceholder.value,
        "multiple": props2.multiple,
        "loading": props2.loading,
        "status": props2.status,
        "tips": props2.tips,
        "borderless": props2.borderless,
        "label": renderLabel,
        "valueDisplay": renderValueDisplay,
        "prefixIcon": props2.prefixIcon,
        "suffix": props2.suffix,
        "suffixIcon": function suffixIcon() {
          return renderSuffixIcon();
        },
        "popupProps": _objectSpread(_objectSpread({}, props2.popupProps), {}, {
          overlayInnerStyle: panels.value.length && !props2.loading ? {
            width: "auto"
          } : "",
          overlayClassName: [overlayClassName.value, (_props2$popupProps = props2.popupProps) === null || _props2$popupProps === void 0 ? void 0 : _props2$popupProps.overlayClassName]
        }),
        "inputProps": _objectSpread({
          size: props2.size
        }, props2.inputProps),
        "tagInputProps": _objectSpread({
          size: props2.size
        }, props2.tagInputProps),
        "tagProps": _objectSpread({}, props2.tagProps)
      }, selectInputEvents.value), {
        label: slots.label,
        suffix: slots.suffix,
        prefixIcon: slots.prefixIcon,
        panel: function panel() {
          return Vue.createVNode(Vue.Fragment, null, [renderTNodeJSX("panelTopContent"), Vue.createVNode(cascader_components_Panel["default"], {
            "option": props2.option,
            "options": props2.options,
            "empty": props2.empty,
            "trigger": props2.trigger,
            "loading": props2.loading,
            "loadingText": props2.loadingText,
            "cascaderContext": cascaderContext.value
          }, {
            option: slots.option,
            empty: slots.empty,
            loadingText: slots.loadingText,
            columnHeader: slots.columnHeader,
            columnFooter: slots.columnFooter
          }), renderTNodeJSX("panelBottomContent")]);
        },
        collapsedItems: slots.collapsedItems
      });
    };
  }
});

exports["default"] = _Cascader;
