// tdesign-vue-next/lib/tree/hooks/useTreeNodes.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { ref, watch, createVNode } from 'vue';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../../_chunks/dep-d9440b5f.js';
import '../../config-provider/hooks/useConfig.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../_chunks/dep-dfeea6f5.js';
import '@babel/runtime/helpers/defineProperty';
import 'tdesign-icons-vue-next';
import '../../checkbox/index.js';
import '../../loading/index.js';
import { p as privateKey } from '../../_chunks/dep-0a9c5f13.js';
import TreeItem from '../tree-item.js';
import useTreeEvents from './useTreeEvents.js';
import '../../_chunks/dep-a6042a25.js';
import '../../config-provider/utils/context.js';
import '../../_chunks/dep-854d2777.js';
import 'dayjs';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../../checkbox/checkbox.js';
import '../../checkbox/props.js';
import '../../_chunks/dep-b3575781.js';
import '../../_chunks/dep-41cafcc6.js';
import '../../_chunks/dep-43a3baa0.js';
import '../../_chunks/dep-9e2a2dff.js';
import '../../_chunks/dep-4d7f902f.js';
import '../../_chunks/dep-a7c4f6fa.js';
import '../../_chunks/dep-8079b4fa.js';
import '../../_chunks/dep-735205fa.js';
import '../../_chunks/dep-cdc7f78f.js';
import '../../_chunks/dep-cc2cf248.js';
import '../../checkbox/constants/index.js';
import '../../checkbox/hooks/useCheckboxLazyLoad.js';
import '../../_chunks/dep-1b49e47a.js';
import '../../checkbox/hooks/useKeyboardEvent.js';
import '../../_chunks/dep-d19ff2c3.js';
import '../../checkbox/group.js';
import '../../checkbox/checkbox-group-props.js';
import '../../_chunks/dep-02e12eec.js';
import '../../_chunks/dep-c53cb022.js';
import '../../loading/directive.js';
import '../../loading/plugin.js';
import '../../_chunks/dep-c9b67f18.js';
import '../../loading/icon/gradient.js';
import '../../_chunks/dep-9e860115.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../loading/props.js';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/regenerator';
import '../../_chunks/dep-2721b80a.js';
import '../utils/adapt.js';
import '../../_chunks/dep-b6de9ec6.js';
import '../../_chunks/dep-b4a20622.js';
import '../../_chunks/dep-6a1a3748.js';
import '../../_chunks/dep-8e2df0a3.js';
import '../../_chunks/dep-b239e51b.js';
import './useItemState.js';
import './useTreeItem.js';
import './useItemEvents.js';
import './useRenderIcon.js';
import '../utils/index.js';
import './useRenderLabel.js';
import './useRenderLine.js';
import './useRenderOperations.js';
import './useDraggable.js';
import './useTreeAction.js';

function useTreeNodes(state) {
  var store = state.store,
    scope = state.scope,
    allNodes = state.allNodes,
    nodes = state.nodes,
    virtualConfig = state.virtualConfig;
  var _useTreeEvents = useTreeEvents(state),
    handleClick = _useTreeEvents.handleClick,
    handleChange = _useTreeEvents.handleChange;
  var nodesEmpty = ref(false);
  var cacheMap = /* @__PURE__ */new Map();
  var refresh = function refresh() {
    allNodes.value = store.getNodes();
  };
  var refreshVisibleNodes = function refreshVisibleNodes() {
    var isVirtual = virtualConfig === null || virtualConfig === void 0 ? void 0 : virtualConfig.isVirtualScroll.value;
    if (isVirtual) return;
    var list = [];
    var hasVisibleNode = false;
    allNodes.value.forEach(function (node) {
      if (node.visible) {
        hasVisibleNode = true;
        cacheMap.set(node.value, node.value);
      }
      if (cacheMap.has(node.value)) {
        list.push(node);
      }
    });
    cacheMap.forEach(function (value) {
      if (!store.getNode(value)) {
        cacheMap["delete"](value);
      }
    });
    nodes.value = list;
    nodesEmpty.value = !hasVisibleNode;
  };
  var refreshVirtualNodes = function refreshVirtualNodes() {
    var isVirtual = virtualConfig === null || virtualConfig === void 0 ? void 0 : virtualConfig.isVirtualScroll.value;
    if (!isVirtual) return;
    var list = virtualConfig.visibleData.value;
    nodes.value = list;
    nodesEmpty.value = list.length <= 0;
  };
  var renderItem = function renderItem(h, node, index, stateId) {
    var rowIndex = node.VIRTUAL_SCROLL_INDEX || index;
    var nodeUniqueId = node[privateKey];
    var treeItem = createVNode(TreeItem, {
      "key": nodeUniqueId,
      "rowIndex": rowIndex,
      "stateId": stateId,
      "itemKey": nodeUniqueId,
      "treeScope": scope,
      "onClick": handleClick,
      "onChange": handleChange
    }, null);
    return treeItem;
  };
  var renderTreeNodes = function renderTreeNodes(h) {
    var stateId = "render-".concat(new Date().getTime());
    var treeNodeViews = nodes.value.map(function (node, index) {
      return renderItem(h, node, index, stateId);
    });
    return treeNodeViews;
  };
  watch(allNodes, refreshVisibleNodes);
  watch(virtualConfig.visibleData, refreshVirtualNodes);
  refresh();
  refreshVisibleNodes();
  refreshVirtualNodes();
  store.emitter.on("update", refresh);
  return {
    nodesEmpty: nodesEmpty,
    renderTreeNodes: renderTreeNodes
  };
}

export { useTreeNodes as default };
