// tdesign-vue-next/lib/tree/hooks/useTreeState.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { useVModel } from '../utils/adapt.js';
import { ref, toRefs } from 'vue';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../../_chunks/dep-d9440b5f.js';
import '../../_chunks/dep-a6042a25.js';
import '../../_chunks/dep-c53cb022.js';
import '../../_chunks/dep-4d7f902f.js';
import '../../config-provider/hooks/useConfig.js';
import '../../config-provider/utils/context.js';
import '../../_chunks/dep-854d2777.js';
import 'dayjs';
import '@babel/runtime/helpers/slicedToArray';
import '../../_chunks/dep-b6de9ec6.js';
import '../../_chunks/dep-b4a20622.js';
import '../../_chunks/dep-dfeea6f5.js';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '@babel/runtime/helpers/defineProperty';
import '../../_chunks/dep-6a1a3748.js';
import '../../_chunks/dep-1b49e47a.js';
import '../../_chunks/dep-41cafcc6.js';
import '../../_chunks/dep-cc2cf248.js';
import '../../_chunks/dep-8e2df0a3.js';
import '../../_chunks/dep-b239e51b.js';
import 'tdesign-icons-vue-next';
import '../../checkbox/index.js';
import '../../checkbox/checkbox.js';
import '../../checkbox/props.js';
import '../../_chunks/dep-b3575781.js';
import '../../_chunks/dep-43a3baa0.js';
import '../../_chunks/dep-9e2a2dff.js';
import '../../_chunks/dep-a7c4f6fa.js';
import '../../_chunks/dep-8079b4fa.js';
import '../../_chunks/dep-735205fa.js';
import '../../_chunks/dep-cdc7f78f.js';
import '../../checkbox/constants/index.js';
import '../../checkbox/hooks/useCheckboxLazyLoad.js';
import '../../checkbox/hooks/useKeyboardEvent.js';
import '../../_chunks/dep-d19ff2c3.js';
import '../../checkbox/group.js';
import '../../checkbox/checkbox-group-props.js';
import '../../_chunks/dep-02e12eec.js';
import '../../loading/index.js';
import '../../loading/directive.js';
import '../../loading/plugin.js';
import '../../_chunks/dep-c9b67f18.js';
import '../../loading/icon/gradient.js';
import '../../_chunks/dep-9e860115.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../loading/props.js';
import '../../_chunks/dep-0a9c5f13.js';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/regenerator';
import '../../_chunks/dep-2721b80a.js';

function useTreeState(props, context) {
  var treeContentRef = ref();
  var nodes = ref([]);
  var allNodes = ref([]);
  var isScrolling = ref(false);
  var refProps = toRefs(props);
  var vmValue = useVModel(props, refProps, "value", "defaultValue", "onChange", "change");
  var vmActived = useVModel(props, refProps, "actived", "defaultActived", "onActive", "active");
  var vmExpanded = useVModel(props, refProps, "expanded", "defaultExpanded", "onExpand", "expand");
  function setStore(store) {
    state.store = store;
    state.scope.store = store;
    allNodes.value = store.getNodes();
  }
  var state = {
    props: props,
    context: context,
    store: null,
    treeContentRef: treeContentRef,
    nodes: nodes,
    allNodes: allNodes,
    isScrolling: isScrolling,
    mouseEvent: null,
    virtualConfig: null,
    scope: {
      store: null,
      treeContentRef: treeContentRef,
      treeProps: props,
      scopedSlots: {},
      virtualConfig: null
    },
    setStore: setStore,
    refProps: refProps,
    vmValue: vmValue,
    vmActived: vmActived,
    vmExpanded: vmExpanded
  };
  return {
    state: state
  };
}

export { useTreeState as default };
