// tdesign-vue-next/cjs/rate/rate.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-449273df.js');
var index = require('../_chunks/dep-c0cb4d13.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-497fc9a5.js');
var index$1 = require('../_chunks/dep-d0227fbe.js');
require('@babel/runtime/helpers/defineProperty');
var rate_props = require('./props.js');
var tooltip_index = require('../tooltip/index.js');
var isArray = require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-f9b59444.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-80827572.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-be760bc2.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-cfdee5e7.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../popup/props.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../_chunks/dep-9ada3ec1.js');
require('../_chunks/dep-95cb1381.js');
require('../_chunks/dep-e3eb5659.js');
require('../popup/container.js');
require('../_chunks/dep-ba2511b0.js');
require('../_chunks/dep-4e2a4791.js');
require('../_chunks/dep-f3377589.js');
require('../_chunks/dep-71d9daf0.js');
require('../_chunks/dep-abc005aa.js');
require('../tooltip/utils/index.js');
require('../_chunks/dep-3a800699.js');
require('../_chunks/dep-e25ae470.js');
require('../_chunks/dep-648bfb84.js');
require('../_chunks/dep-4e2907e0.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

var _Rate = Vue.defineComponent({
  name: "TRate",
  props: rate_props["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var renderTNodeJSX = index.useTNodeJSX();
    var _useConfig = configProvider_hooks_useConfig.useConfig("rate"),
      globalConfig = _useConfig.globalConfig;
    var activeColor = Vue.computed(function () {
      return isArray.isArray(props2.color) ? props2.color[0] : props2.color;
    });
    var defaultColor = Vue.computed(function () {
      return isArray.isArray(props2.color) ? props2.color[1] : "var(--td-bg-color-component)";
    });
    var _toRefs = Vue.toRefs(props2),
      inputValue = _toRefs.value,
      modelValue = _toRefs.modelValue;
    var _useVModel = index$1.useVModel(inputValue, modelValue, props2.defaultValue, props2.onChange),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      starValue = _useVModel2[0],
      setStarValue = _useVModel2[1];
    var hoverValue = Vue.ref(void 0);
    var root = Vue.ref();
    var displayValue = Vue.computed(function () {
      return Number(hoverValue.value || starValue.value);
    });
    var displayTexts = Vue.computed(function () {
      return props2.texts.length === 0 ? globalConfig.value.rateText : props2.texts;
    });
    var displayText = Vue.computed(function () {
      return displayTexts.value[Math.ceil(displayValue.value - 1)];
    });
    var RateIcon = function RateIcon(iconProps) {
      if (slots.icon !== void 0) {
        return renderTNodeJSX("icon", {
          params: iconProps
        });
      }
      return Vue.createVNode(tdesignIconsVueNext.StarFilledIcon, iconProps, null);
    };
    var getStarValue = function getStarValue(event, index) {
      if (props2.allowHalf) {
        var _root$value$getBoundi = root.value.getBoundingClientRect(),
          left = _root$value$getBoundi.left;
        var firstStar = root.value.firstChild.nextSibling;
        var _firstStar$getBoundin = firstStar.getBoundingClientRect(),
          width = _firstStar$getBoundin.width;
        var clientX = event.clientX;
        var starMiddle = width * (index - 0.5) + props2.gap * (index - 1);
        if (clientX - left >= starMiddle) return index;
        if (clientX - left < starMiddle) return index - 0.5;
      }
      return index;
    };
    var mouseEnterHandler = function mouseEnterHandler(event, index) {
      if (props2.disabled) return;
      hoverValue.value = getStarValue(event, index);
    };
    var mouseLeaveHandler = function mouseLeaveHandler() {
      if (props2.disabled) return;
      hoverValue.value = void 0;
    };
    var clickHandler = function clickHandler(event, index) {
      if (props2.disabled) return;
      var value = getStarValue(event, index);
      if (props2.clearable && value === starValue.value) {
        hoverValue.value = void 0;
        setStarValue(0);
      } else {
        setStarValue(value);
      }
    };
    var getStarCls = function getStarCls(index) {
      if (props2.allowHalf && index + 0.5 === displayValue.value) return "".concat(classPrefix.value, "-rate__item--half");
      if (index >= displayValue.value) return "";
      if (index < displayValue.value) return "".concat(classPrefix.value, "-rate__item--full");
    };
    var _useConfig2 = configProvider_hooks_useConfig.useConfig("classPrefix"),
      classPrefix = _useConfig2.classPrefix;
    return function () {
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-rate"),
        "onMouseleave": mouseLeaveHandler
      }, [Vue.createVNode("ul", {
        "class": "".concat(classPrefix.value, "-rate__list"),
        "style": {
          gap: "".concat(props2.gap, "px")
        },
        "ref": root
      }, [_toConsumableArray__default["default"](Array(Number(props2.count))).map(function (_, index) {
        return Vue.createVNode("li", {
          "key": index,
          "class": ["".concat(classPrefix.value, "-rate__item"), getStarCls(index)],
          "onClick": function onClick(event) {
            return clickHandler(event, index + 1);
          },
          "onMousemove": function onMousemove(event) {
            return mouseEnterHandler(event, index + 1);
          }
        }, [props2.showText ? Vue.createVNode(tooltip_index.Tooltip, {
          "key": index,
          "content": displayText.value
        }, {
          "default": function _default() {
            return [Vue.createVNode("div", {
              "class": "".concat(classPrefix.value, "-rate__star-top")
            }, [Vue.createVNode(RateIcon, {
              "size": props2.size,
              "color": activeColor.value
            }, null)]), Vue.createVNode("div", {
              "class": "".concat(classPrefix.value, "-rate__star-bottom")
            }, [Vue.createVNode(RateIcon, {
              "size": props2.size,
              "color": defaultColor.value
            }, null)])];
          }
        }) : Vue.createVNode(Vue.Fragment, null, [Vue.createVNode("div", {
          "class": "".concat(classPrefix.value, "-rate__star-top")
        }, [Vue.createVNode(RateIcon, {
          "size": props2.size,
          "color": activeColor.value
        }, null)]), Vue.createVNode("div", {
          "class": "".concat(classPrefix.value, "-rate__star-bottom")
        }, [Vue.createVNode(RateIcon, {
          "size": props2.size,
          "color": defaultColor.value
        }, null)])])]);
      })]), props2.showText && Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-rate__text")
      }, [displayText.value])]);
    };
  }
});

exports["default"] = _Rate;
