// tdesign-vue-next/cjs/badge/badge.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var badge_props = require('./props.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-449273df.js');
var index = require('../_chunks/dep-c0cb4d13.js');
var index$1 = require('../_chunks/dep-95cb1381.js');
require('../_chunks/dep-497fc9a5.js');
require('../_chunks/dep-f9b59444.js');
require('../_chunks/dep-51846987.js');
require('../_chunks/dep-80827572.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-be760bc2.js');
require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-cfdee5e7.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

var _Badge = Vue.defineComponent({
  name: "TBadge",
  inheritAttrs: false,
  props: badge_props["default"],
  setup: function setup(props2, _ref) {
    var attrs = _ref.attrs;
    var renderContent = index.useContent();
    var renderTNodeJSX = index.useTNodeJSX();
    var displayCount = Vue.computed(function () {
      var count = renderTNodeJSX("count");
      if (Number.isNaN(Number(count))) {
        return count;
      }
      count = Number(props2.count);
      return count > props2.maxCount ? "".concat(props2.maxCount, "+") : count;
    });
    var getOffset = function getOffset() {
      if (!props2.offset) return {};
      var _props2$offset = _slicedToArray__default["default"](props2.offset, 2),
        xOffset = _props2$offset[0],
        yOffset = _props2$offset[1];
      xOffset = Number.isNaN(Number(xOffset)) ? xOffset : "".concat(xOffset, "px");
      yOffset = Number.isNaN(Number(yOffset)) ? yOffset : "".concat(yOffset, "px");
      return {
        xOffset: xOffset,
        yOffset: yOffset
      };
    };
    var classPrefix = index$1.usePrefixClass();
    var COMPONENT_NAME = index$1.usePrefixClass("badge");
    var isHidden = Vue.computed(function () {
      return !props2.showZero && (displayCount.value === 0 || displayCount.value === "0");
    });
    var badgeClassNames = Vue.computed(function () {
      return [_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--dot"), !!props2.dot), "".concat(COMPONENT_NAME.value, "--circle"), !props2.dot && props2.shape === "circle"), "".concat(COMPONENT_NAME.value, "--round"), !props2.dot && props2.shape === "round"), "".concat(classPrefix.value, "-size-s"), props2.size === "small")];
    });
    var inlineStyle = Vue.computed(function () {
      var _getOffset = getOffset(),
        xOffset = _getOffset.xOffset,
        yOffset = _getOffset.yOffset;
      return {
        background: props2.color,
        right: xOffset,
        top: yOffset
      };
    });
    return function () {
      return Vue.createVNode("div", Vue.mergeProps({
        "class": COMPONENT_NAME.value
      }, attrs), [renderContent("default", "content"), isHidden.value ? null : Vue.createVNode("sup", {
        "class": badgeClassNames.value,
        "style": inlineStyle.value
      }, [props2.dot ? null : displayCount.value])]);
    };
  }
});

exports["default"] = _Badge;
