// tdesign-vue-next/cjs/breadcrumb/breadcrumb-item.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var breadcrumb_breadcrumbItemProps = require('./breadcrumb-item-props.js');
var tooltip_index = require('../tooltip/index.js');
var dom = require('../_chunks/dep-51846987.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
var index = require('../_chunks/dep-c0cb4d13.js');
var index$1 = require('../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-497fc9a5.js');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../popup/props.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../_chunks/dep-9ada3ec1.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-31abb282.js');
require('../_chunks/dep-c71a9cd7.js');
require('../_chunks/dep-f9b59444.js');
require('dayjs');
require('../_chunks/dep-39529ef1.js');
require('../_chunks/dep-f4c1cb5b.js');
require('../_chunks/dep-4896d30f.js');
require('../_chunks/dep-80827572.js');
require('../_chunks/dep-2610df9f.js');
require('../_chunks/dep-2c7a67b8.js');
require('../_chunks/dep-db6459dd.js');
require('../_chunks/dep-4d7a3a91.js');
require('../_chunks/dep-6e64ef0e.js');
require('../_chunks/dep-959f4847.js');
require('../_chunks/dep-ae380218.js');
require('../_chunks/dep-fdac7521.js');
require('../_chunks/dep-df5442c5.js');
require('../_chunks/dep-141cbae1.js');
require('../_chunks/dep-c277c7cb.js');
require('../_chunks/dep-7653f80f.js');
require('../_chunks/dep-6e7a2100.js');
require('../_chunks/dep-e3eb5659.js');
require('../_chunks/dep-d0227fbe.js');
require('../_chunks/dep-cfdee5e7.js');
require('../_chunks/dep-1c6213b5.js');
require('../_chunks/dep-0a17438d.js');
require('../_chunks/dep-5090b515.js');
require('../popup/container.js');
require('../_chunks/dep-ba2511b0.js');
require('../_chunks/dep-be760bc2.js');
require('../_chunks/dep-4e2a4791.js');
require('../_chunks/dep-f3377589.js');
require('../_chunks/dep-71d9daf0.js');
require('../_chunks/dep-449273df.js');
require('../_chunks/dep-abc005aa.js');
require('../_chunks/dep-9dfd682d.js');
require('../_chunks/dep-2db31625.js');
require('../_chunks/dep-39313d05.js');
require('../_chunks/dep-f9bc7c82.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../tooltip/utils/index.js');
require('../_chunks/dep-3a800699.js');
require('../_chunks/dep-e25ae470.js');
require('../_chunks/dep-648bfb84.js');
require('../_chunks/dep-4e2907e0.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var localTBreadcrumbOrigin = {
  separator: "",
  theme: "light",
  maxItemWidth: void 0
};
var _BreadcrumbItem = Vue.defineComponent({
  name: "TBreadcrumbItem",
  inheritAttrs: false,
  props: _objectSpread(_objectSpread({}, breadcrumb_breadcrumbItemProps["default"]), {}, {
    isEllipsisItem: Boolean
  }),
  setup: function setup(props2, _ref) {
    var attrs = _ref.attrs;
    var renderContent = index.useContent();
    var renderTNodeJSX = index.useTNodeJSX();
    var breadcrumbText = Vue.ref();
    var localTBreadcrumb = Vue.inject("tBreadcrumb", localTBreadcrumbOrigin);
    var themeClassName = Vue.ref(localTBreadcrumb === null || localTBreadcrumb === void 0 ? void 0 : localTBreadcrumb.theme);
    var isCutOff = Vue.ref(false);
    var COMPONENT_NAME = index$1.usePrefixClass("breadcrumb__item");
    var separatorClass = index$1.usePrefixClass("breadcrumb__separator");
    var disableClass = index$1.usePrefixClass("is-disabled");
    var linkClass = index$1.usePrefixClass("link");
    var maxLengthClass = index$1.usePrefixClass("breadcrumb__inner");
    var textFlowClass = index$1.usePrefixClass("breadcrumb--text-overflow");
    var maxWithStyle = Vue.computed(function () {
      var maxItemWidth = localTBreadcrumb === null || localTBreadcrumb === void 0 ? void 0 : localTBreadcrumb.maxItemWidth;
      var maxWith = props2.maxWidth || maxItemWidth || "120";
      return {
        maxWidth: "".concat(maxWith, "px")
      };
    });
    var _getCurrentInstance = Vue.getCurrentInstance(),
      proxy = _getCurrentInstance.proxy;
    Vue.onMounted(function () {
      if (breadcrumbText.value) {
        isCutOff.value = dom.isTextEllipsis(breadcrumbText.value);
      }
    });
    Vue.onBeforeUpdate(function () {
      if (breadcrumbText.value) {
        isCutOff.value = dom.isTextEllipsis(breadcrumbText.value);
      }
    });
    var handleClick = function handleClick() {
      var router = props2.router || proxy.$router;
      if (props2.to && router) {
        props2.replace ? router.replace(props2.to) : router.push(props2.to);
      } else {
        window.location.href = props2.href;
      }
    };
    var bindEvent = function bindEvent(e) {
      if (!props2.disabled) {
        e.preventDefault();
        if (props2.target === "_blank") {
          props2.href ? window.open(props2.href) : window.open(props2.to);
        } else {
          handleClick();
        }
      }
    };
    return function () {
      var itemClass = [COMPONENT_NAME.value, themeClassName.value];
      var textClass = [textFlowClass.value];
      if (props2.disabled) {
        textClass.push(disableClass.value);
      }
      var listeners = {
        onClick: function onClick(e) {
          if (props2.disabled) {
            e.stopPropagation();
            return;
          }
        }
      };
      var _content = renderContent("default", "content");
      var textContent = Vue.createVNode("span", {
        "class": maxLengthClass.value,
        style: maxWithStyle.value
      }, [renderTNodeJSX("icon"), Vue.createVNode("span", {
        "ref": breadcrumbText,
        "class": "".concat(maxLengthClass.value, "-text")
      }, [_content])]);
      var itemContent = Vue.createVNode("span", _objectSpread({
        "class": textClass
      }, listeners), [textContent]);
      if ((props2.href || props2.to) && !props2.disabled) {
        textClass.push(linkClass.value);
        itemContent = Vue.createVNode("a", Vue.mergeProps({
          "class": textClass,
          "href": props2.href,
          "target": props2.target
        }, listeners, {
          "onClick": bindEvent
        }), [textContent]);
      }
      if (props2 !== null && props2 !== void 0 && props2.isEllipsisItem) {
        itemContent = Vue.createVNode("div", {
          "style": "display: flex"
        }, [_content]);
      }
      return Vue.createVNode("div", Vue.mergeProps({
        "class": itemClass
      }, attrs, {
        "onClick": props2.disabled ? void 0 : props2.onClick
      }), [isCutOff.value ? Vue.createVNode(tooltip_index.Tooltip, {
        "content": function content() {
          return _content;
        }
      }, _isSlot(itemContent) ? itemContent : {
        "default": function _default() {
          return [itemContent];
        }
      }) : itemContent, Vue.createVNode("span", {
        "class": separatorClass.value,
        "style": {
          textOverflow: isCutOff.value ? "ellipsis" : "clip"
        }
      }, [localTBreadcrumb.separator])]);
    };
  }
});

exports["default"] = _BreadcrumbItem;
