// tdesign-vue-next/cjs/color-picker/components/panel/index.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var color = require('../../../_chunks/dep-25a2fcb0.js');
var constants = require('../../../_chunks/dep-5a84661a.js');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
var format = require('../../../_chunks/dep-4d3a6850.js');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-449273df.js');
var index = require('../../../_chunks/dep-9ada3ec1.js');
var configProvider_hooks_useConfig = require('../../../config-provider/hooks/useConfig.js');
var index$2 = require('../../../_chunks/dep-4b52068a.js');
require('../../../_chunks/dep-497fc9a5.js');
var index$1 = require('../../../_chunks/dep-d0227fbe.js');
var colorPicker_colorPickerPanelProps = require('../../color-picker-panel-props.js');
var colorPicker_hooks_index = require('../../hooks/index.js');
var colorPicker_components_format_index = require('../format/index.js');
var colorPicker_components_panel_alpha = require('./alpha.js');
var colorPicker_components_panel_header = require('./header.js');
var colorPicker_components_panel_hue = require('./hue.js');
var colorPicker_components_panel_linearGradient = require('./linear-gradient.js');
var colorPicker_components_panel_saturation = require('./saturation.js');
var colorPicker_components_panel_swatches = require('./swatches.js');
var isUndefined = require('../../../_chunks/dep-2cf25d25.js');
var cloneDeep = require('../../../_chunks/dep-141cbae1.js');
require('tinycolor2');
require('../../../_chunks/dep-df5442c5.js');
require('../../../_chunks/dep-3ffe18b0.js');
require('../../../_chunks/dep-f9b59444.js');
require('../../../_chunks/dep-31abb282.js');
require('../../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../../_chunks/dep-39529ef1.js');
require('../../../_chunks/dep-f4c1cb5b.js');
require('../../../_chunks/dep-4896d30f.js');
require('../../../_chunks/dep-80827572.js');
require('../../../_chunks/dep-2610df9f.js');
require('../../../_chunks/dep-2c7a67b8.js');
require('../../../_chunks/dep-db6459dd.js');
require('../../../_chunks/dep-4d7a3a91.js');
require('../../../_chunks/dep-6e64ef0e.js');
require('../../../_chunks/dep-959f4847.js');
require('../../../_chunks/dep-ae380218.js');
require('../../../_chunks/dep-fdac7521.js');
require('../../../_chunks/dep-c277c7cb.js');
require('../../../_chunks/dep-7653f80f.js');
require('../../../_chunks/dep-6e7a2100.js');
require('../../../_chunks/dep-cfdee5e7.js');
require('../../../_chunks/dep-1c6213b5.js');
require('../../../_chunks/dep-0a17438d.js');
require('../../../_chunks/dep-5090b515.js');
require('../../../_chunks/dep-95cb1381.js');
require('../../props.js');
require('../../../select/index.js');
require('../../../select/select.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../../common-components/fake-arrow.js');
require('../../../_chunks/dep-cce7af6a.js');
require('../../../_chunks/dep-6130f266.js');
require('../../../_chunks/dep-e41b3434.js');
require('../../../select-input/index.js');
require('../../../select-input/select-input.js');
require('../../../popup/index.js');
require('../../../popup/popup.js');
require('@popperjs/core');
require('../../../_chunks/dep-51846987.js');
require('../../../_chunks/dep-be760bc2.js');
require('../../../_chunks/dep-c0cb4d13.js');
require('../../../_chunks/dep-9dfd682d.js');
require('../../../_chunks/dep-2db31625.js');
require('../../../_chunks/dep-39313d05.js');
require('../../../_chunks/dep-f9bc7c82.js');
require('../../../_chunks/dep-e3eb5659.js');
require('../../../popup/container.js');
require('../../../popup/props.js');
require('../../../_chunks/dep-ba2511b0.js');
require('../../../_chunks/dep-4e2a4791.js');
require('../../../_chunks/dep-f3377589.js');
require('../../../_chunks/dep-71d9daf0.js');
require('../../../_chunks/dep-abc005aa.js');
require('../../../select-input/props.js');
require('../../../select-input/hooks/useMultiple.js');
require('../../../tag-input/index.js');
require('../../../tag-input/tag-input.js');
require('tdesign-icons-vue-next');
require('../../../input/index.js');
require('../../../input/input.js');
require('../../../input/props.js');
require('../../../_chunks/dep-a2e46896.js');
require('../../../_chunks/dep-6512c9d2.js');
require('../../../_chunks/dep-88ae5d8a.js');
require('../../../_chunks/dep-f02d9258.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../_chunks/dep-b4528c21.js');
require('../../../_chunks/dep-d20c4e5e.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../tag-input/props.js');
require('../../../_chunks/dep-5ea73cce.js');
require('../../../tag-input/hooks/useDragSorter.js');
require('../../../tag-input/hooks/useHover.js');
require('../../../tag-input/hooks/useTagScroll.js');
require('../../../tag-input/hooks/useTagList.js');
require('../../../tag/index.js');
require('../../../tag/tag.js');
require('../../../tag/props.js');
require('../../../tag/check-tag.js');
require('../../../tag/check-tag-props.js');
require('../../../_chunks/dep-2e8277b1.js');
require('../../../tag/check-tag-group.js');
require('../../../tag/check-tag-group-props.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-203a5570.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-3231adce.js');
require('../../../loading/icon/gradient.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-e46285c0.js');
require('../../../_chunks/dep-54801354.js');
require('../../../_chunks/dep-bd59ca66.js');
require('../../../_chunks/dep-f7508ffa.js');
require('../../../_chunks/dep-470f211b.js');
require('../../../_chunks/dep-9b549c85.js');
require('../../../_chunks/dep-e25ae470.js');
require('../../../_chunks/dep-bed84553.js');
require('../../../_chunks/dep-a4b963ab.js');
require('../../../select-input/hooks/useOverlayInnerStyle.js');
require('../../../select-input/hooks/useSingle.js');
require('../../../_chunks/dep-008d4a55.js');
require('../../../_chunks/dep-0e1751ee.js');
require('../../../_chunks/dep-4e2907e0.js');
require('../../../select/components/select-panel.js');
require('../../../select/option.js');
require('../../../select/option-props.js');
require('../../../checkbox/index.js');
require('../../../checkbox/checkbox.js');
require('../../../checkbox/props.js');
require('../../../_chunks/dep-3cd26986.js');
require('../../../checkbox/constants/index.js');
require('../../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../../_chunks/dep-12e6ec24.js');
require('../../../checkbox/hooks/useKeyboardEvent.js');
require('../../../checkbox/group.js');
require('../../../checkbox/checkbox-group-props.js');
require('../../../_chunks/dep-cec24bbf.js');
require('../../../_chunks/dep-f786e1df.js');
require('../../../_chunks/dep-618d56e7.js');
require('../../../select/utils/index.js');
require('../../../select/constants/index.js');
require('../../../select/option-group.js');
require('../../../select/option-group-props.js');
require('../../../select/props.js');
require('../../../select/hooks/usePanelVirtualScroll.js');
require('../../../_chunks/dep-3a800699.js');
require('../../../_chunks/dep-648bfb84.js');
require('../../../select/hooks/useKeyboardControl.js');
require('../../../select/hooks/useSelectOptions.js');
require('../../../_chunks/dep-8f2069e9.js');
require('../format/inputs.js');
require('../../../input-number/index.js');
require('../../../input-number/input-number.js');
require('../../../button/index.js');
require('../../../button/button.js');
require('../../../button/props.js');
require('../../../input-number/props.js');
require('../../../input-number/hooks/useInputNumber.js');
require('../../../_chunks/dep-74a0ebc2.js');
require('../../../_chunks/dep-94323f13.js');
require('../../../_chunks/dep-75d3928f.js');
require('../../../_chunks/dep-801a06fe.js');
require('./slider.js');
require('../../../_chunks/dep-8fa69637.js');
require('./base-props.js');
require('../../../radio/index.js');
require('../../../radio/radio.js');
require('../../../radio/props.js');
require('../../../radio/constants/index.js');
require('../../../radio/group.js');
require('../../../radio/radio-group-props.js');
require('../../../radio/radio-button.js');
require('../../../radio/hooks/useKeyboard.js');
require('../../../watermark/hooks/index.js');
require('../../../_chunks/dep-59ce9497.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

/**
 * Checks if `value` is `null`.
 *
 * @static
 * @memberOf _
 * @since 0.1.0
 * @category Lang
 * @param {*} value The value to check.
 * @returns {boolean} Returns `true` if `value` is `null`, else `false`.
 * @example
 *
 * _.isNull(null);
 * // => true
 *
 * _.isNull(void 0);
 * // => false
 */
function isNull(value) {
  return value === null;
}

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var ColorPanel = Vue.defineComponent({
  name: "ColorPanel",
  props: _objectSpread({}, colorPicker_colorPickerPanelProps["default"]),
  setup: function setup(props2) {
    var baseClassName = colorPicker_hooks_index.useBaseClassName();
    var _useCommonClassName = index.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS;
    var _useConfig = configProvider_hooks_useConfig.useConfig("colorPicker"),
      t = _useConfig.t,
      globalConfig = _useConfig.globalConfig;
    var statusClassNames = STATUS.value;
    var _toRefs = Vue.toRefs(props2),
      inputValue = _toRefs.value,
      modelValue = _toRefs.modelValue,
      recentColors = _toRefs.recentColors;
    var _useVModel = index$1.useVModel(inputValue, modelValue, props2.defaultValue, props2.onChange),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      innerValue = _useVModel2[0],
      setInnerValue = _useVModel2[1];
    var _useDefaultValue = index$2.useDefaultValue(recentColors, props2.defaultRecentColors, props2.onRecentColorsChange, "recentColors"),
      _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
      innerRecentColors = _useDefaultValue2[0],
      setInnerRecentColors = _useDefaultValue2[1];
    var getModeByColor = function getModeByColor(input) {
      if (props2.colorModes.length === 1) return props2.colorModes[0];
      return props2.colorModes.includes("linear-gradient") && color.Color.isGradientColor(input) ? "linear-gradient" : "monochrome";
    };
    var mode = Vue.ref(getModeByColor(innerValue.value));
    var isGradient = Vue.computed(function () {
      return mode.value === "linear-gradient";
    });
    var defaultEmptyColor = Vue.computed(function () {
      return isGradient.value ? constants.DEFAULT_LINEAR_GRADIENT : constants.DEFAULT_COLOR;
    });
    var color$1 = Vue.ref(new color.Color(innerValue.value || defaultEmptyColor.value));
    var formatModel = Vue.ref(format.initColorFormat(props2.format, props2.enableAlpha));
    var addRecentlyUsedColor = function addRecentlyUsedColor() {
      if (innerRecentColors.value === null || innerRecentColors.value === false) {
        return;
      }
      var colors = cloneDeep.cloneDeep(innerRecentColors.value) || [];
      var currentColor = color$1.value.isGradient ? color$1.value.linearGradient : color$1.value.rgba;
      var index = colors.indexOf(currentColor);
      if (index > -1) {
        colors.splice(index, 1);
      }
      colors.unshift(currentColor);
      if (colors.length > constants.TD_COLOR_USED_COLORS_MAX_SIZE) {
        colors.length = constants.TD_COLOR_USED_COLORS_MAX_SIZE;
      }
      handleRecentlyUsedColorsChange(colors);
    };
    var handleRecentlyUsedColorsChange = function handleRecentlyUsedColorsChange(colors) {
      setInnerRecentColors(colors);
    };
    var emitColorChange = function emitColorChange(trigger) {
      var value = color$1.value.getFormattedColor(props2.format, props2.enableAlpha);
      setInnerValue(value, {
        color: color.getColorObject(color$1.value),
        trigger: trigger || "palette-saturation-brightness"
      });
    };
    Vue.watch(function () {
      return innerValue.value;
    }, function (newColor) {
      var newMode = getModeByColor(newColor);
      mode.value = newMode;
      color$1.value.isGradient = newMode === "linear-gradient";
      var currentColor = color$1.value.getFormattedColor(props2.format, props2.enableAlpha);
      if (currentColor !== newColor) {
        color$1.value.update(newColor);
      }
    });
    var handleModeChange = function handleModeChange(newMode) {
      mode.value = newMode;
      var isGradientMode = newMode === "linear-gradient";
      color$1.value.isGradient = isGradientMode;
      if (isGradientMode) {
        color$1.value.update(color$1.value.gradientColors.length > 0 ? color$1.value.linearGradient : constants.DEFAULT_LINEAR_GRADIENT);
      } else {
        color$1.value.update(color$1.value.rgba);
      }
      emitColorChange();
    };
    var handleSatAndValueChange = function handleSatAndValueChange(_ref) {
      var saturation = _ref.saturation,
        value = _ref.value;
      var _color$value = color$1.value,
        sat = _color$value.saturation,
        val = _color$value.value;
      var changeTrigger = "palette-saturation-brightness";
      if (value !== val && saturation !== sat) {
        color$1.value.saturation = saturation;
        color$1.value.value = value;
        changeTrigger = "palette-saturation-brightness";
      } else if (saturation !== sat) {
        color$1.value.saturation = saturation;
        changeTrigger = "palette-saturation";
      } else if (value !== val) {
        color$1.value.value = value;
        changeTrigger = "palette-brightness";
      } else {
        return;
      }
      emitColorChange(changeTrigger);
    };
    var handleHueChange = function handleHueChange(hue) {
      var _props2$onPaletteBarC;
      color$1.value.hue = hue;
      emitColorChange("palette-hue-bar");
      (_props2$onPaletteBarC = props2.onPaletteBarChange) === null || _props2$onPaletteBarC === void 0 || _props2$onPaletteBarC.call(props2, {
        color: color.getColorObject(color$1.value)
      });
    };
    var handleAlphaChange = function handleAlphaChange(alpha) {
      color$1.value.alpha = alpha;
      emitColorChange("palette-alpha-bar");
    };
    var handleInputChange = function handleInputChange(input, alpha) {
      color$1.value.update(input);
      color$1.value.alpha = alpha;
      emitColorChange("input");
    };
    var handleGradientChange = function handleGradientChange(_ref2) {
      var key = _ref2.key,
        payload = _ref2.payload;
      var trigger = "palette-saturation-brightness";
      switch (key) {
        case "degree":
          color$1.value.gradientDegree = payload;
          trigger = "input";
          break;
        case "selectedId":
          color$1.value.gradientSelectedId = payload;
          break;
        case "colors":
          color$1.value.gradientColors = payload;
          break;
      }
      emitColorChange(trigger);
    };
    var handleSetColor = function handleSetColor(value, trigger) {
      var newMode = getModeByColor(value);
      mode.value = newMode;
      color$1.value.isGradient = newMode === "linear-gradient";
      color$1.value.update(value);
      emitColorChange(trigger);
    };
    return function () {
      var baseProps = {
        color: color$1.value,
        disabled: props2.disabled
      };
      var onlySupportGradient = props2.colorModes.length === 1 && props2.colorModes.includes("linear-gradient");
      var recentColors2 = innerRecentColors.value;
      if (onlySupportGradient && Array.isArray(recentColors2)) {
        recentColors2 = recentColors2.filter(function (color2) {
          return color.Color.isGradientColor(color2);
        });
      }
      var showUsedColors = !!Array.isArray(recentColors2);
      var systemColors = props2.swatchColors;
      if (isUndefined.isUndefined(systemColors)) {
        systemColors = _toConsumableArray__default["default"](constants.DEFAULT_SYSTEM_SWATCH_COLORS);
      }
      if (isNull(systemColors)) {
        systemColors = [];
      }
      if (onlySupportGradient) {
        systemColors = systemColors.filter(function (color2) {
          return color.Color.isGradientColor(color2);
        });
      }
      var showSystemColors = Array.isArray(systemColors) && systemColors.length;
      var renderSwatches = function renderSwatches() {
        if (!showSystemColors && !showUsedColors) return null;
        return Vue.createVNode(Vue.Fragment, null, [Vue.createVNode("div", {
          "class": "".concat(baseClassName.value, "__swatches-wrap")
        }, [showUsedColors ? Vue.createVNode(colorPicker_components_panel_swatches["default"], Vue.mergeProps(baseProps, {
          "title": t(globalConfig.value.recentColorTitle),
          "editable": true,
          "colors": recentColors2,
          "handleAddColor": addRecentlyUsedColor,
          "onSetColor": function onSetColor(color2) {
            return handleSetColor(color2, "recent");
          },
          "onChange": handleRecentlyUsedColorsChange
        }), null) : null, showSystemColors ? Vue.createVNode(colorPicker_components_panel_swatches["default"], Vue.mergeProps(baseProps, {
          "title": t(globalConfig.value.swatchColorTitle),
          "colors": systemColors,
          "onSetColor": function onSetColor(color2) {
            return handleSetColor(color2, "preset");
          }
        }), null) : null])]);
      };
      return Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__panel"), props2.disabled ? statusClassNames.disabled : false]
      }, [Vue.createVNode(colorPicker_components_panel_header["default"], Vue.mergeProps(props2, {
        "mode": mode.value,
        "onModeChange": handleModeChange
      }), null), Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__body")]
      }, [isGradient.value ? Vue.createVNode(colorPicker_components_panel_linearGradient["default"], Vue.mergeProps(baseProps, {
        "onChange": handleGradientChange,
        "enableMultipleGradient": props2.enableMultipleGradient
      }), null) : null, Vue.createVNode(colorPicker_components_panel_saturation["default"], Vue.mergeProps(baseProps, {
        "onChange": handleSatAndValueChange
      }), null), Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__sliders-wrapper")]
      }, [Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__sliders")]
      }, [Vue.createVNode(colorPicker_components_panel_hue["default"], Vue.mergeProps(baseProps, {
        "onChange": handleHueChange
      }), null), props2.enableAlpha ? Vue.createVNode(colorPicker_components_panel_alpha["default"], Vue.mergeProps(baseProps, {
        "onChange": handleAlphaChange
      }), null) : null]), props2.showPrimaryColorPreview ? Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__sliders-preview"), "".concat(baseClassName.value, "--bg-alpha")]
      }, [Vue.createVNode("span", {
        "class": "".concat(baseClassName.value, "__sliders-preview-inner"),
        "style": {
          background: isGradient.value ? color$1.value.linearGradient : color$1.value.rgba
        }
      }, null)]) : null]), Vue.createVNode(colorPicker_components_format_index["default"], Vue.mergeProps(props2, {
        "color": color$1.value,
        "format": formatModel.value,
        "onInputChange": handleInputChange
      }), null), renderSwatches()])]);
    };
  }
});

exports["default"] = ColorPanel;
