// tdesign-vue-next/cjs/color-picker/components/panel/linear-gradient.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var constants = require('../../../_chunks/dep-5a84661a.js');
var color = require('../../../_chunks/dep-25a2fcb0.js');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/toConsumableArray');
var inputNumber_index = require('../../../input-number/index.js');
var colorPicker_hooks_index = require('../../hooks/index.js');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-449273df.js');
var index = require('../../../_chunks/dep-9ada3ec1.js');
require('../../../config-provider/hooks/useConfig.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../../_chunks/dep-497fc9a5.js');
var colorPicker_components_panel_baseProps = require('./base-props.js');
var cloneDeep = require('../../../_chunks/dep-141cbae1.js');
require('tinycolor2');
require('../../../_chunks/dep-df5442c5.js');
require('../../../_chunks/dep-3ffe18b0.js');
require('../../../input-number/input-number.js');
require('tdesign-icons-vue-next');
require('../../../button/index.js');
require('../../../button/button.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-203a5570.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-3231adce.js');
require('../../../loading/icon/gradient.js');
require('../../../_chunks/dep-95cb1381.js');
require('../../../_chunks/dep-e3eb5659.js');
require('../../../_chunks/dep-cce7af6a.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../../_chunks/dep-6130f266.js');
require('../../../_chunks/dep-e41b3434.js');
require('../../../_chunks/dep-f9b59444.js');
require('../../../_chunks/dep-31abb282.js');
require('../../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../../_chunks/dep-39529ef1.js');
require('../../../_chunks/dep-f4c1cb5b.js');
require('../../../_chunks/dep-4896d30f.js');
require('../../../_chunks/dep-80827572.js');
require('../../../_chunks/dep-2610df9f.js');
require('../../../_chunks/dep-2c7a67b8.js');
require('../../../_chunks/dep-db6459dd.js');
require('../../../_chunks/dep-4d7a3a91.js');
require('../../../_chunks/dep-6e64ef0e.js');
require('../../../_chunks/dep-959f4847.js');
require('../../../_chunks/dep-ae380218.js');
require('../../../_chunks/dep-fdac7521.js');
require('../../../_chunks/dep-c277c7cb.js');
require('../../../_chunks/dep-7653f80f.js');
require('../../../_chunks/dep-6e7a2100.js');
require('../../../_chunks/dep-51846987.js');
require('../../../_chunks/dep-be760bc2.js');
require('../../../_chunks/dep-c0cb4d13.js');
require('../../../_chunks/dep-9dfd682d.js');
require('../../../_chunks/dep-2db31625.js');
require('../../../_chunks/dep-0a17438d.js');
require('../../../_chunks/dep-5090b515.js');
require('../../../_chunks/dep-39313d05.js');
require('../../../_chunks/dep-f9bc7c82.js');
require('../../../_chunks/dep-1c6213b5.js');
require('../../../_chunks/dep-cfdee5e7.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-e46285c0.js');
require('../../../_chunks/dep-54801354.js');
require('../../../_chunks/dep-bd59ca66.js');
require('../../../_chunks/dep-f7508ffa.js');
require('../../../_chunks/dep-470f211b.js');
require('../../../_chunks/dep-9b549c85.js');
require('../../../_chunks/dep-e25ae470.js');
require('../../../_chunks/dep-bed84553.js');
require('../../../_chunks/dep-a4b963ab.js');
require('../../../_chunks/dep-abc005aa.js');
require('../../../button/props.js');
require('../../../_chunks/dep-a2e46896.js');
require('../../../_chunks/dep-6512c9d2.js');
require('../../../_chunks/dep-3cd26986.js');
require('../../../input/index.js');
require('../../../input/input.js');
require('../../../input/props.js');
require('../../../_chunks/dep-88ae5d8a.js');
require('../../../_chunks/dep-f02d9258.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../_chunks/dep-d0227fbe.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../_chunks/dep-b4528c21.js');
require('../../../_chunks/dep-d20c4e5e.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../_chunks/dep-ba2511b0.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../input-number/props.js');
require('../../../_chunks/dep-5ea73cce.js');
require('../../../input-number/hooks/useInputNumber.js');
require('../../../_chunks/dep-74a0ebc2.js');
require('../../../_chunks/dep-94323f13.js');
require('../../../_chunks/dep-75d3928f.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var DELETE_KEYS = ["delete", "backspace"];
var LinearGradient = Vue.defineComponent({
  name: "LinearGradient",
  inheritAttrs: false,
  props: _objectSpread(_objectSpread({}, colorPicker_components_panel_baseProps["default"]), {}, {
    enableMultipleGradient: {
      type: Boolean,
      "default": true
    }
  }),
  setup: function setup(props) {
    var baseClassName = colorPicker_hooks_index.useBaseClassName();
    var _useCommonClassName = index.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS;
    var statusClassNames = STATUS.value;
    var refSlider = Vue.ref(null);
    var sliderRect = Vue.reactive({
      left: 0,
      width: constants.GRADIENT_SLIDER_DEFAULT_WIDTH
    });
    var isDragging = Vue.ref(false);
    var isMoved = Vue.ref(false);
    var degree = Vue.ref(props.color.gradientDegree);
    var selectedId = Vue.ref(props.color.gradientSelectedId);
    var colors = Vue.ref(cloneDeep.cloneDeep(props.color.gradientColors));
    Vue.watch(function () {
      return props.color.gradientDegree;
    }, function (value) {
      return degree.value = value;
    });
    Vue.watch(function () {
      return props.color.gradientSelectedId;
    }, function (value) {
      return selectedId.value = value;
    });
    Vue.watch(function () {
      return props.color.gradientColors;
    }, function (value) {
      colors.value = cloneDeep.cloneDeep(value);
    }, {
      deep: true
    });
    var handleChange = function handleChange(key, payload, addUsedColor) {
      if (props.disabled) {
        return;
      }
      props.onChange({
        key: key,
        payload: payload,
        addUsedColor: addUsedColor
      });
    };
    var handleDegreeChange = function handleDegreeChange(value) {
      if (props.disabled || value === props.color.gradientDegree) {
        return;
      }
      degree.value = value;
      handleChange("degree", value, true);
    };
    var handleSelectedIdChange = function handleSelectedIdChange(value) {
      if (props.disabled) {
        return;
      }
      selectedId.value = value;
      handleChange("selectedId", value);
    };
    var handleColorsChange = function handleColorsChange(value, isEnded) {
      if (props.disabled) {
        return;
      }
      colors.value = value;
      handleChange("colors", value, isEnded);
    };
    var updateActiveThumbLeft = function updateActiveThumbLeft(left) {
      var index = colors.value.findIndex(function (c) {
        return c.id === selectedId.value;
      });
      if (index === -1) {
        return;
      }
      var point = colors.value[index];
      left = Math.max(0, Math.min(sliderRect.width, left));
      var percentLeft = left / sliderRect.width * 100;
      colors.value.splice(index, 1, {
        color: point.color,
        left: percentLeft,
        id: point.id
      });
      handleColorsChange(colors.value);
    };
    var handleStart = function handleStart(id) {
      var rect = refSlider.value.getBoundingClientRect();
      sliderRect.left = rect.left;
      sliderRect.width = rect.width || constants.GRADIENT_SLIDER_DEFAULT_WIDTH;
      if (isDragging.value || props.disabled) {
        return;
      }
      isMoved.value = false;
      isDragging.value = true;
      handleSelectedIdChange(id);
      refSlider.value.focus();
      window.addEventListener("mousemove", handleMove, false);
      window.addEventListener("mouseup", _handleEnd, false);
      window.addEventListener("contextmenu", _handleEnd, false);
    };
    var handleMove = function handleMove(e) {
      if (!isDragging.value || props.disabled) {
        return;
      }
      var left = e.clientX - sliderRect.left;
      isMoved.value = true;
      updateActiveThumbLeft(left);
    };
    var _handleEnd = function handleEnd() {
      if (!isDragging.value) {
        return;
      }
      setTimeout(function () {
        isDragging.value = false;
      }, 0);
      if (isMoved.value) {
        handleColorsChange(colors.value, true);
        isMoved.value = false;
      }
      window.removeEventListener("mousemove", handleMove, false);
      window.removeEventListener("mouseup", _handleEnd, false);
      window.removeEventListener("contextmenu", _handleEnd, false);
    };
    var handleKeyup = function handleKeyup(e) {
      if (props.disabled) {
        return;
      }
      var points = colors.value;
      var pos = points.findIndex(function (c) {
        return c.id === selectedId.value;
      });
      var length = points.length;
      if (DELETE_KEYS.includes(e.key.toLocaleLowerCase()) && length > 2 && pos >= 0 && pos <= length - 1) {
        points.splice(pos, 1);
        if (!points[pos]) {
          pos = points[pos + 1] ? pos + 1 : points[pos - 1] ? pos - 1 : 0;
        }
        var current = points[pos];
        handleColorsChange(points, true);
        handleSelectedIdChange(current === null || current === void 0 ? void 0 : current.id);
      }
    };
    var handleThumbBarClick = function handleThumbBarClick(e) {
      if (props.disabled || !props.enableMultipleGradient) {
        return;
      }
      var left = e.offsetX;
      var percentLeft = left / sliderRect.width * 100;
      var newPoint = color.genGradientPoint(percentLeft, props.color.rgba);
      colors.value.push(newPoint);
      handleColorsChange(colors.value, true);
      handleSelectedIdChange(newPoint.id);
    };
    Vue.onMounted(function () {
      var rect = refSlider.value.getBoundingClientRect();
      sliderRect.left = rect.left;
      sliderRect.width = rect.width || constants.GRADIENT_SLIDER_DEFAULT_WIDTH;
    });
    Vue.onBeforeUnmount(function () {
      window.removeEventListener("mousemove", handleMove, false);
      window.removeEventListener("mouseup", _handleEnd, false);
      window.removeEventListener("contextmenu", _handleEnd, false);
    });
    return function () {
      var _props$color = props.color,
        linearGradient = _props$color.linearGradient,
        gradientColors = _props$color.gradientColors;
      var thumbBackground = color.gradientColors2string({
        points: gradientColors,
        degree: 90
      });
      return Vue.createVNode("div", {
        "class": "".concat(baseClassName.value, "__gradient")
      }, [Vue.createVNode("div", {
        "class": "".concat(baseClassName.value, "__gradient-slider")
      }, [Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__slider"), "".concat(baseClassName.value, "--bg-alpha")],
        "onKeyup": handleKeyup,
        "tabindex": 0,
        "ref": refSlider
      }, [Vue.createVNode("ul", {
        "class": "gradient-thumbs",
        "onClick": handleThumbBarClick,
        "style": {
          background: thumbBackground
        }
      }, [colors.value.map(function (t) {
        var left = "".concat(Math.round(t.left * 100) / 100, "%");
        return Vue.createVNode("li", {
          "class": ["".concat(baseClassName.value, "__thumb"), "gradient-thumbs__item", selectedId.value === t.id ? statusClassNames.active : ""],
          "key": t.id,
          "title": "".concat(t.color, " ").concat(left),
          "style": {
            color: t.color,
            left: left
          },
          "onClick": function onClick(e) {
            return e.stopPropagation();
          },
          "onMousedown": function onMousedown() {
            return handleStart(t.id);
          }
        }, [Vue.createVNode("span", {
          "class": ["gradient-thumbs__item-inner", "".concat(baseClassName.value, "--bg-alpha")]
        }, null)]);
      })])])]), Vue.createVNode("div", {
        "class": "".concat(baseClassName.value, "__gradient-degree"),
        "title": "".concat(degree.value, "deg")
      }, [Vue.createVNode(inputNumber_index.InputNumber, {
        "size": "small",
        "theme": "normal",
        "min": 0,
        "max": 360,
        "step": 1,
        "format": function format(value) {
          return "".concat(value, "\xB0");
        },
        "modelValue": degree.value,
        "onUpdate:modelValue": function onUpdateModelValue($event) {
          return degree.value = $event;
        },
        "onBlur": handleDegreeChange,
        "onEnter": handleDegreeChange,
        "disabled": props.disabled
      }, null)]), Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__gradient-preview"), "".concat(baseClassName.value, "--bg-alpha")]
      }, [Vue.createVNode("span", {
        "class": "preview-inner",
        "style": {
          background: linearGradient
        }
      }, null)])]);
    };
  }
});

exports["default"] = LinearGradient;
