// tdesign-vue-next/lib/color-picker/utils/index.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

export { C as Color, c as cmyk2rgb, d as cmykInputToColor, b as genGradientPoint, f as genId, g as getColorObject, e as getColorWithoutAlpha, a as gradientColors2string, i as isGradientColor, p as parseGradientString, r as rgb2cmyk } from '../../_chunks/dep-63d8d332.js';
export { A as ALPHA_FORMAT_MAP, C as COLOR_FORMAT_INPUTS, c as COLOR_MODES, a as DEFAULT_COLOR, D as DEFAULT_LINEAR_GRADIENT, b as DEFAULT_SYSTEM_SWATCH_COLORS, F as FORMATS, G as GRADIENT_SLIDER_DEFAULT_WIDTH, d as SATURATION_PANEL_DEFAULT_HEIGHT, S as SATURATION_PANEL_DEFAULT_WIDTH, e as SLIDER_DEFAULT_WIDTH, T as TD_COLOR_USED_COLORS_MAX_SIZE } from '../../_chunks/dep-92a02ffb.js';
export { D as Draggable } from '../../_chunks/dep-0af62898.js';
export { a as getColorFormatInputs, b as getColorFormatMap, g as getColorFormatOptions, i as initColorFormat } from '../../_chunks/dep-7a0dc9bd.js';
import '@babel/runtime/helpers/slicedToArray';
import '@babel/runtime/helpers/classCallCheck';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/defineProperty';
import 'tinycolor2';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
