// tdesign-vue-next/cjs/calendar/hooks/useCalendarClass.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-449273df.js');
var index = require('../../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-497fc9a5.js');
require('@babel/runtime/helpers/defineProperty');
var calendar_constants_index = require('../constants/index.js');
require('../../_chunks/dep-f9b59444.js');
require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-80827572.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-6e64ef0e.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');

function useCalendarClass(props, state) {
  var prefixClass = index.usePrefixClass(calendar_constants_index.COMPONENT_NAME);
  var body = Vue.computed(function () {
    return ["".concat(prefixClass.value), "".concat(prefixClass.value, "--").concat(props.theme)];
  });
  var panel = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__panel"), "".concat(prefixClass.value, "__panel--").concat(state.curSelectedMode)];
  });
  var control = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__control")];
  });
  var title = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__title")];
  });
  var controlSection = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__control-section")];
  });
  var controlSectionCell = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__control-section-cell")];
  });
  var controlTag = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__control-tag")];
  });
  var table = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table")];
  });
  var tableHead = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table-head")];
  });
  var tableHeadRow = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table-head-row")];
  });
  var tableHeadCell = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table-head-cell")];
  });
  var tableBody = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table-body")];
  });
  var tableBodyRow = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table-body-row")];
  });
  return {
    body: body,
    panel: panel,
    control: control,
    title: title,
    controlSection: controlSection,
    controlSectionCell: controlSectionCell,
    controlTag: controlTag,
    table: table,
    tableHead: tableHead,
    tableHeadRow: tableHeadRow,
    tableHeadCell: tableHeadCell,
    tableBody: tableBody,
    tableBodyRow: tableBodyRow
  };
}
function useCalendarCellClass() {
  var prefixClass = index.usePrefixClass(calendar_constants_index.COMPONENT_NAME);
  var tableBodyCell = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table-body-cell")];
  });
  var tableBodyCell4Now = Vue.computed(function () {
    return "".concat(prefixClass.value, "__table-body-cell--now");
  });
  var tableBodyCellDisplay = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table-body-cell-display")];
  });
  var tableBodyCellCsontent = Vue.computed(function () {
    return ["".concat(prefixClass.value, "__table-body-cell-content")];
  });
  return {
    tableBodyCell: tableBodyCell,
    tableBodyCell4Now: tableBodyCell4Now,
    tableBodyCellDisplay: tableBodyCellDisplay,
    tableBodyCellCsontent: tableBodyCellCsontent
  };
}

exports.useCalendarCellClass = useCalendarCellClass;
exports.useCalendarClass = useCalendarClass;
