// tdesign-vue-next/esm/date-picker/components/panel/ExtraContent.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode } from 'vue';
import TDateFooter from '../base/Footer.js';
import '@babel/runtime/helpers/defineProperty';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../../../_chunks/dep-d9440b5f.js';
import '../../../_chunks/dep-a6042a25.js';
import '../../../_chunks/dep-b3575781.js';
import '../../../_chunks/dep-41cafcc6.js';
import '../../../_chunks/dep-43a3baa0.js';
import '../../../_chunks/dep-4d7f902f.js';
import '../../../config-provider/hooks/useConfig.js';
import '../../../config-provider/utils/context.js';
import '../../../_chunks/dep-854d2777.js';
import 'dayjs';
import '@babel/runtime/helpers/slicedToArray';
import '../../../_chunks/dep-dfeea6f5.js';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../../../button/index.js';
import '../../../button/button.js';
import '../../../loading/index.js';
import '../../../loading/directive.js';
import '../../../loading/plugin.js';
import '../../../_chunks/dep-c9b67f18.js';
import '../../../loading/icon/gradient.js';
import '../../../_chunks/dep-cdc7f78f.js';
import '../../../_chunks/dep-9e860115.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../../_chunks/dep-9e2a2dff.js';
import '../../../loading/props.js';
import '../../../_chunks/dep-c53cb022.js';
import 'tdesign-vue-next/esm/common/style/web/components/loading/_index.less';
import '../../../button/props.js';
import '../../../_chunks/dep-a7c4f6fa.js';
import '../../../_chunks/dep-735205fa.js';
import 'tdesign-vue-next/esm/common/style/web/components/button/_index.less';

var TExtraContent = defineComponent({
  name: "TExtraContent",
  props: {
    presets: Object,
    enableTimePicker: Boolean,
    presetsPlacement: String,
    needConfirm: Boolean,
    onPresetClick: Function,
    onConfirmClick: Function,
    selectedValue: [String, Number, Array, Date]
  },
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var showPanelFooter = computed(function () {
      return props.enableTimePicker && props.needConfirm || props.presets || !!slots.presets;
    });
    return function () {
      return showPanelFooter.value ? createVNode(TDateFooter, {
        "presets": props.presets,
        "onPresetClick": props.onPresetClick,
        "enableTimePicker": props.enableTimePicker,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "selectedValue": props.selectedValue,
        "needConfirm": props.needConfirm
      }, {
        presets: slots.presets
      }) : null;
    };
  }
});

export { TExtraContent as default };
