// tdesign-vue-next/cjs/menu/components/popup-overflow-content.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-449273df.js');
var index = require('../../_chunks/dep-95cb1381.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-497fc9a5.js');
require('@babel/runtime/helpers/defineProperty');
require('../../_chunks/dep-f9b59444.js');
require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-31abb282.js');
require('../../_chunks/dep-c71a9cd7.js');
require('dayjs');
require('../../_chunks/dep-39529ef1.js');
require('../../_chunks/dep-f4c1cb5b.js');
require('../../_chunks/dep-4896d30f.js');
require('../../_chunks/dep-80827572.js');
require('../../_chunks/dep-2610df9f.js');
require('../../_chunks/dep-2c7a67b8.js');
require('../../_chunks/dep-db6459dd.js');
require('../../_chunks/dep-4d7a3a91.js');
require('../../_chunks/dep-6e64ef0e.js');
require('../../_chunks/dep-959f4847.js');
require('../../_chunks/dep-ae380218.js');
require('../../_chunks/dep-fdac7521.js');
require('../../_chunks/dep-df5442c5.js');
require('../../_chunks/dep-141cbae1.js');
require('../../_chunks/dep-c277c7cb.js');
require('../../_chunks/dep-7653f80f.js');
require('../../_chunks/dep-6e7a2100.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');

var PopupOverflowContent = Vue.defineComponent({
  name: "PopupOverflowContent",
  props: {
    foldIndex: {
      type: Number,
      required: true
    }
  },
  setup: function setup(props, ctx) {
    var classPrefix = index.usePrefixClass();
    var wrapperRef = Vue.ref();
    var hideBeforeFoldIndex = function hideBeforeFoldIndex() {
      if (!wrapperRef.value) return;
      var menuItemClass = "".concat(classPrefix.value, "-menu__item");
      var submenuClass = "".concat(classPrefix.value, "-submenu");
      var items = [];
      var _collect = function collect(parent, depth) {
        if (depth > 3) return;
        Array.from(parent.children).forEach(function (el) {
          if (!(el instanceof HTMLElement)) return;
          if (el.classList.contains(menuItemClass) || el.classList.contains(submenuClass)) {
            items.push(el);
          } else {
            _collect(el, depth + 1);
          }
        });
      };
      _collect(wrapperRef.value, 0);
      items.forEach(function (el, index) {
        el.style.display = index < props.foldIndex ? "none" : "";
      });
    };
    Vue.onMounted(hideBeforeFoldIndex);
    Vue.onUpdated(hideBeforeFoldIndex);
    return function () {
      var _ctx$slots$default, _ctx$slots;
      return Vue.createVNode("div", {
        "ref": wrapperRef
      }, [(_ctx$slots$default = (_ctx$slots = ctx.slots)["default"]) === null || _ctx$slots$default === void 0 ? void 0 : _ctx$slots$default.call(_ctx$slots)]);
    };
  }
});

exports["default"] = PopupOverflowContent;
