// tdesign-vue-next/lib/cascader/cascader-panel.js
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, createVNode } from 'vue';
import TCascaderSubPanel from './components/Panel.js';
import props from './props.js';
import { useCascaderContext } from './hooks/index.js';
import '@babel/runtime/helpers/defineProperty';
import '@babel/runtime/helpers/slicedToArray';
import './components/Item.js';
import 'tdesign-icons-vue-next';
import './utils/className.js';
import '@babel/runtime/helpers/toConsumableArray';
import 'lodash-es';
import './utils/helper.js';
import '../checkbox/index.js';
import '../checkbox/checkbox.js';
import '../checkbox/props.js';
import '@babel/runtime/helpers/typeof';
import '../_chunks/dep-d9440b5f.js';
import '../_chunks/dep-a6042a25.js';
import '../_chunks/dep-b3575781.js';
import '../_chunks/dep-41cafcc6.js';
import '../_chunks/dep-43a3baa0.js';
import '../_chunks/dep-9e2a2dff.js';
import '../config-provider/hooks/useConfig.js';
import '../config-provider/utils/context.js';
import '../_chunks/dep-854d2777.js';
import 'dayjs';
import '../_chunks/dep-4d7f902f.js';
import '../_chunks/dep-a7c4f6fa.js';
import '../_chunks/dep-dfeea6f5.js';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../_chunks/dep-8079b4fa.js';
import '../_chunks/dep-735205fa.js';
import '../_chunks/dep-cdc7f78f.js';
import '../_chunks/dep-cc2cf248.js';
import '../checkbox/constants/index.js';
import '../checkbox/hooks/useCheckboxLazyLoad.js';
import '../_chunks/dep-1b49e47a.js';
import '../checkbox/hooks/useKeyboardEvent.js';
import '../_chunks/dep-d19ff2c3.js';
import '../checkbox/group.js';
import '../checkbox/checkbox-group-props.js';
import '../_chunks/dep-02e12eec.js';
import '../_chunks/dep-c53cb022.js';
import '../loading/index.js';
import '../loading/directive.js';
import '../loading/plugin.js';
import '../_chunks/dep-c9b67f18.js';
import '../loading/icon/gradient.js';
import '../_chunks/dep-9e860115.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../loading/props.js';
import '../_chunks/dep-b4a20622.js';
import './utils/effect.js';
import '../_chunks/dep-ca3ed7c4.js';
import 'mitt';
import '../_chunks/dep-0a9c5f13.js';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/regenerator';
import '../_chunks/dep-2721b80a.js';
import '../_chunks/dep-b6de9ec6.js';

var _CascaderPanel = defineComponent({
  name: "TCascaderPanel",
  props: props,
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var _useCascaderContext = useCascaderContext(props2),
      cascaderContext = _useCascaderContext.cascaderContext;
    return function () {
      return createVNode(TCascaderSubPanel, {
        "trigger": props2.trigger,
        "cascaderContext": cascaderContext.value,
        "empty": props2.empty
      }, {
        empty: slots.empty,
        option: slots.option,
        loadingText: slots.loadingText
      });
    };
  }
});

export { _CascaderPanel as default };
