// tdesign-vue-next/es/date-picker/components/panel/ExtraContent.mjs
/**
 * tdesign v1.20.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode } from 'vue';
import TDateFooter from '../base/Footer.mjs';
import '../../../_chunks/dep-98dac781.mjs';
import '../../../_chunks/dep-f038e2b4.mjs';
import 'lodash-es';
import '../../../_chunks/dep-9ce910b6.mjs';
import '../../../_chunks/dep-557b68dd.mjs';
import '../../../_chunks/dep-26d65784.mjs';
import '../../../_chunks/dep-05bddb06.mjs';
import '../../../_chunks/dep-57027bda.mjs';
import '../../../config-provider/hooks/useConfig.mjs';
import '../../../config-provider/utils/context.mjs';
import '../../../_chunks/dep-57949dcd.mjs';
import '../../../_chunks/dep-cafb5eac.mjs';
import '../../../_chunks/dep-f1da7269.mjs';
import '../../../_chunks/dep-ea5fbe21.mjs';
import '../../../_chunks/dep-192d0c88.mjs';
import '../../../_chunks/dep-652c99fe.mjs';
import '../../../_chunks/dep-5b369203.mjs';
import '../../../_chunks/dep-a4f55e3b.mjs';
import '../../../_chunks/dep-920eff5b.mjs';
import '../../../button/index.mjs';
import '../../../button/button.mjs';
import '../../../loading/index.mjs';
import '../../../loading/directive.mjs';
import '../../../loading/plugin.mjs';
import '../../../loading/loading.mjs';
import '../../../loading/icon/gradient.mjs';
import '../../../_chunks/dep-9b3b77bb.mjs';
import '../../../_chunks/dep-4de66dd0.mjs';
import '../../../loading/props.mjs';
import './style/css.mjs';
import '../../../button/props.mjs';

var TExtraContent = defineComponent({
  name: "TExtraContent",
  props: {
    presets: Object,
    enableTimePicker: Boolean,
    presetsPlacement: String,
    needConfirm: Boolean,
    onPresetClick: Function,
    onConfirmClick: Function,
    selectedValue: [String, Number, Array, Date]
  },
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var showPanelFooter = computed(function () {
      return props.enableTimePicker && props.needConfirm || props.presets || !!slots.presets;
    });
    return function () {
      return showPanelFooter.value ? createVNode(TDateFooter, {
        "presets": props.presets,
        "onPresetClick": props.onPresetClick,
        "enableTimePicker": props.enableTimePicker,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "selectedValue": props.selectedValue,
        "needConfirm": props.needConfirm
      }, {
        presets: slots.presets
      }) : null;
    };
  }
});

export { TExtraContent as default };
